{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from Bio import SeqIO\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TODO:\n",
    "1. Сдлеать рисунки для по geo3918, geo3817\n",
    "2. Сделать колинеарный анализ на основании сравнения геномов гео в seed\n",
    "3. Сделать подобное сравненеи и для Jay 535"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Создание файлов для CIRCOS geo3918\n",
    "\n",
    "**karyotype** -- содержит иформацию о хромосомах или контигах имеет следующее строение:\n",
    "\n",
    "chr - ID LABEL START END COLOR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/data/karyotype.txt'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[1;32m/home/aster/Project_Geoarchaeota_28.11/results/lab_journals/22.04.23_CIRCOS_notebook.ipynb Cell 4\u001b[0m in \u001b[0;36m<cell line: 2>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      <a href='vscode-notebook-cell:/home/aster/Project_Geoarchaeota_28.11/results/lab_journals/22.04.23_CIRCOS_notebook.ipynb#W3sZmlsZQ%3D%3D?line=0'>1</a>\u001b[0m \u001b[39m#Для этого можно прочитать файл с контигами и записать все необходимые данные из него.\u001b[39;00m\n\u001b[1;32m      <a href='vscode-notebook-cell:/home/aster/Project_Geoarchaeota_28.11/results/lab_journals/22.04.23_CIRCOS_notebook.ipynb#W3sZmlsZQ%3D%3D?line=1'>2</a>\u001b[0m \u001b[39mfor\u001b[39;00m num, seq \u001b[39min\u001b[39;00m \u001b[39menumerate\u001b[39m(SeqIO\u001b[39m.\u001b[39mparse(\u001b[39mopen\u001b[39m(\u001b[39m'\u001b[39m\u001b[39m/home/aster/Project_Geoarchaeota_28.11/data/sequences_annotation/1448933.3.contigs.fa\u001b[39m\u001b[39m'\u001b[39m), \u001b[39m'\u001b[39m\u001b[39mfasta\u001b[39m\u001b[39m'\u001b[39m), start\u001b[39m=\u001b[39m\u001b[39m1\u001b[39m):\n\u001b[0;32m----> <a href='vscode-notebook-cell:/home/aster/Project_Geoarchaeota_28.11/results/lab_journals/22.04.23_CIRCOS_notebook.ipynb#W3sZmlsZQ%3D%3D?line=2'>3</a>\u001b[0m     \u001b[39mwith\u001b[39;00m \u001b[39mopen\u001b[39;49m(\u001b[39m'\u001b[39;49m\u001b[39m/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/data/karyotype.txt\u001b[39;49m\u001b[39m'\u001b[39;49m, \u001b[39m'\u001b[39;49m\u001b[39ma\u001b[39;49m\u001b[39m'\u001b[39;49m) \u001b[39mas\u001b[39;00m write_file:\n\u001b[1;32m      <a href='vscode-notebook-cell:/home/aster/Project_Geoarchaeota_28.11/results/lab_journals/22.04.23_CIRCOS_notebook.ipynb#W3sZmlsZQ%3D%3D?line=3'>4</a>\u001b[0m         write_file\u001b[39m.\u001b[39mwrite(\u001b[39mf\u001b[39m\u001b[39m\"\u001b[39m\u001b[39mchr - \u001b[39m\u001b[39m{\u001b[39;00mseq\u001b[39m.\u001b[39mname\u001b[39m.\u001b[39msplit(\u001b[39m'\u001b[39m\u001b[39m_\u001b[39m\u001b[39m'\u001b[39m)[\u001b[39m0\u001b[39m]\u001b[39m}\u001b[39;00m\u001b[39m cn_\u001b[39m\u001b[39m{\u001b[39;00mnum\u001b[39m}\u001b[39;00m\u001b[39m 0 \u001b[39m\u001b[39m{\u001b[39;00m\u001b[39mlen\u001b[39m(seq)\u001b[39m}\u001b[39;00m\u001b[39m green\u001b[39m\u001b[39m\"\u001b[39m)\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/data/karyotype.txt'"
     ]
    }
   ],
   "source": [
    "#Для этого можно прочитать файл с контигами и записать все необходимые данные из него.\n",
    "for num, seq in enumerate(SeqIO.parse(open('/home/aster/Project_Geoarchaeota_28.11/data/sequences_annotation/1448933.3.contigs.fa'), 'fasta'), start=1):\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/geo3918/data/karyotype_geo3918.txt', 'a') as write_file:\n",
    "        write_file.write(f\"chr - {seq.name.split('_')[0]} cn_{num} 0 {len(seq)} green\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**high light** -- Используется для создания секторов разных цевтов, например для подсветки определенных генов\n",
    "\n",
    "Хорошо сочитается если с COG нотацией генов\n",
    "\n",
    "Файл имеет следующую структуру:\n",
    "\n",
    "ID START END COLOR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_geo3918 = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/data/sequences_annotation/1448933.3.gff', sep='\\t', header=None)\n",
    "gff_geo3918_plus_strand = gff_geo3918[gff_geo3918[6] == '+']\n",
    "gff_geo3918_minus_strand = gff_geo3918[gff_geo3918[6] == '-']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in gff_geo3918_plus_strand.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/data/highlight_plus_strand.txt', 'a') as write_file:\n",
    "        write_file.write(f\"contig {row[3]} {row[4]} fill_color=red\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in gff_geo3918_minus_strand.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/data/highlight_minus_strand.txt', 'a') as write_file:\n",
    "        write_file.write(f\"contig {row[3]} {row[4]} fill_color=blue\\n\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Тоже самое но с COG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_geo3918 = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/seq_data/gff_files/geo_3918.gff', sep='\\t', header=None)\n",
    "eggNOG_annot = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/results/eggNog_mapper/MM_k15bh4a0.emapper.annotations(1).tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2</td>\n",
       "      <td>2227</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>ID=fig|1448933.3.peg.1;Name=beta-glucosidase (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2259</td>\n",
       "      <td>3155</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>ID=fig|1448933.3.peg.2;Name=hypothetical protein</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            0    1    2     3     4  5  6  7  \\\n",
       "0  contig_1_pilon_pilon_pilon  FIG  CDS     2  2227  .  -  2   \n",
       "1  contig_1_pilon_pilon_pilon  FIG  CDS  2259  3155  .  -  0   \n",
       "\n",
       "                                                   8  \n",
       "0  ID=fig|1448933.3.peg.1;Name=beta-glucosidase (...  \n",
       "1   ID=fig|1448933.3.peg.2;Name=hypothetical protein  "
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff_geo3918.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>seed_ortholog</th>\n",
       "      <th>evalue</th>\n",
       "      <th>score</th>\n",
       "      <th>eggNOG_OGs</th>\n",
       "      <th>max_annot_lvl</th>\n",
       "      <th>COG_category</th>\n",
       "      <th>Description</th>\n",
       "      <th>Preferred_name</th>\n",
       "      <th>GOs</th>\n",
       "      <th>...</th>\n",
       "      <th>KEGG_ko</th>\n",
       "      <th>KEGG_Pathway</th>\n",
       "      <th>KEGG_Module</th>\n",
       "      <th>KEGG_Reaction</th>\n",
       "      <th>KEGG_rclass</th>\n",
       "      <th>BRITE</th>\n",
       "      <th>KEGG_TC</th>\n",
       "      <th>CAZy</th>\n",
       "      <th>BiGG_Reaction</th>\n",
       "      <th>PFAMs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>fig|1448933.3.peg.1</td>\n",
       "      <td>930945.SiRe_2208</td>\n",
       "      <td>8.040000e-276</td>\n",
       "      <td>784.0</td>\n",
       "      <td>COG1472@1|root,arCOG04634@2157|Archaea,2XPMS@2...</td>\n",
       "      <td>28889|Crenarchaeota</td>\n",
       "      <td>G</td>\n",
       "      <td>Fibronectin type III-like domain</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>ko:K05349</td>\n",
       "      <td>ko00460,ko00500,ko00940,ko01100,ko01110,map004...</td>\n",
       "      <td>-</td>\n",
       "      <td>R00026,R02558,R02887,R02985,R03527,R04949,R049...</td>\n",
       "      <td>RC00049,RC00059,RC00171,RC00262,RC00397,RC0045...</td>\n",
       "      <td>ko00000,ko00001,ko01000</td>\n",
       "      <td>-</td>\n",
       "      <td>GH3</td>\n",
       "      <td>-</td>\n",
       "      <td>Fn3-like,Glyco_hydro_3,Glyco_hydro_3_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>fig|1448933.3.peg.5</td>\n",
       "      <td>1459636.NTE_01989</td>\n",
       "      <td>2.680000e-26</td>\n",
       "      <td>104.0</td>\n",
       "      <td>COG1832@1|root,arCOG04227@2157|Archaea,41SMT@6...</td>\n",
       "      <td>651137|Thaumarchaeota</td>\n",
       "      <td>S</td>\n",
       "      <td>CoA binding domain</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>ko:K06929</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00000</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>CoA_binding_2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    ID      seed_ortholog         evalue  score  \\\n",
       "0  fig|1448933.3.peg.1   930945.SiRe_2208  8.040000e-276  784.0   \n",
       "1  fig|1448933.3.peg.5  1459636.NTE_01989   2.680000e-26  104.0   \n",
       "\n",
       "                                          eggNOG_OGs          max_annot_lvl  \\\n",
       "0  COG1472@1|root,arCOG04634@2157|Archaea,2XPMS@2...    28889|Crenarchaeota   \n",
       "1  COG1832@1|root,arCOG04227@2157|Archaea,41SMT@6...  651137|Thaumarchaeota   \n",
       "\n",
       "  COG_category                       Description Preferred_name GOs  ...  \\\n",
       "0            G  Fibronectin type III-like domain              -   -  ...   \n",
       "1            S                CoA binding domain              -   -  ...   \n",
       "\n",
       "     KEGG_ko                                       KEGG_Pathway KEGG_Module  \\\n",
       "0  ko:K05349  ko00460,ko00500,ko00940,ko01100,ko01110,map004...           -   \n",
       "1  ko:K06929                                                  -           -   \n",
       "\n",
       "                                       KEGG_Reaction  \\\n",
       "0  R00026,R02558,R02887,R02985,R03527,R04949,R049...   \n",
       "1                                                  -   \n",
       "\n",
       "                                         KEGG_rclass                    BRITE  \\\n",
       "0  RC00049,RC00059,RC00171,RC00262,RC00397,RC0045...  ko00000,ko00001,ko01000   \n",
       "1                                                  -                  ko00000   \n",
       "\n",
       "  KEGG_TC CAZy BiGG_Reaction                                   PFAMs  \n",
       "0       -  GH3             -  Fn3-like,Glyco_hydro_3,Glyco_hydro_3_C  \n",
       "1       -    -             -                           CoA_binding_2  \n",
       "\n",
       "[2 rows x 21 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eggNOG_annot.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "#color dict\n",
    "d = {'J':'ff0000', 'A':'c2af58', 'K':'ff9900', 'L':'ffff00', 'B':'ffc600',\n",
    "    'D':'99ff00', 'Y':'493126', 'V':'ff008a', 'T':'0000ff', 'M':'9ec928',\n",
    "    'N':'006633', 'Z':'660099', 'W':'336699', 'U':'33cc99', 'O':'00ffff', \n",
    "    'C':'9900ff', 'G':'805642', 'E':'99334d', 'H':'727dcc', 'I':'5c5a1b',\n",
    "    'P':'0099ff', 'Q':'ffcc99', 'R':'ff9999','S':'d6aadf', 'F':'99334d',\n",
    "    'IQ':'ffcc99',\n",
    "    '-':'white',\n",
    "    'EGP':'b8b100',\n",
    "    'CP':'9900ff',\n",
    "    'JM':'ff0000',\n",
    "    'EP':'99334d',\n",
    "    'NU':'006633',\n",
    "    'HP':'727dcc',\n",
    "    'GM':'805642',\n",
    "    ################\n",
    "    'LT':'ffff00',\n",
    "    'KL':'ff9900',\n",
    "    'BQ':'ffc600',\n",
    "    'EH':'99334d',\n",
    "    'FG':'99334d',\n",
    "    'OU':'00ffff',\n",
    "    'CE':'9900ff'\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in gff_geo3918.iterrows():\n",
    "    for idx2, row2, in eggNOG_annot.iterrows():\n",
    "        if row2['ID'] in row[8]:\n",
    "            gff_geo3918.loc[idx, 'color'] = f\"fill_color={d[row2['COG_category']]}\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>color</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2</td>\n",
       "      <td>2227</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>ID=fig|1448933.3.peg.1;Name=beta-glucosidase (...</td>\n",
       "      <td>fill_color=805642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2259</td>\n",
       "      <td>3155</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>ID=fig|1448933.3.peg.2;Name=hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>3265</td>\n",
       "      <td>4527</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>ID=fig|1448933.3.peg.3;Name=hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4530</td>\n",
       "      <td>4667</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>ID=fig|1448933.3.peg.4;Name=hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4690</td>\n",
       "      <td>5208</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>ID=fig|1448933.3.peg.5;Name=hypothetical protein</td>\n",
       "      <td>fill_color=d6aadf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1550</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>1124017</td>\n",
       "      <td>1124091</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=fig|1448933.3.rna.43;Name=tRNA-Gly-TCC</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1551</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>1124017</td>\n",
       "      <td>1124091</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=fig|1448933.3.rna.43;Name=tRNA-Gly-TCC</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1552</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>1169723</td>\n",
       "      <td>1169797</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=fig|1448933.3.rna.44;Name=tRNA-Thr-TGT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1553</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>1169723</td>\n",
       "      <td>1169797</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=fig|1448933.3.rna.44;Name=tRNA-Thr-TGT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1554</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>1169723</td>\n",
       "      <td>1169797</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=fig|1448933.3.rna.44;Name=tRNA-Thr-TGT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1555 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               0    1     2        3        4  5  6  7  \\\n",
       "0     contig_1_pilon_pilon_pilon  FIG   CDS        2     2227  .  -  2   \n",
       "1     contig_1_pilon_pilon_pilon  FIG   CDS     2259     3155  .  -  0   \n",
       "2     contig_1_pilon_pilon_pilon  FIG   CDS     3265     4527  .  -  1   \n",
       "3     contig_1_pilon_pilon_pilon  FIG   CDS     4530     4667  .  -  0   \n",
       "4     contig_1_pilon_pilon_pilon  FIG   CDS     4690     5208  .  +  1   \n",
       "...                          ...  ...   ...      ...      ... .. .. ..   \n",
       "1550  contig_1_pilon_pilon_pilon  FIG  tRNA  1124017  1124091  .  -  .   \n",
       "1551  contig_1_pilon_pilon_pilon  FIG  tRNA  1124017  1124091  .  -  .   \n",
       "1552  contig_1_pilon_pilon_pilon  FIG  tRNA  1169723  1169797  .  +  .   \n",
       "1553  contig_1_pilon_pilon_pilon  FIG  tRNA  1169723  1169797  .  +  .   \n",
       "1554  contig_1_pilon_pilon_pilon  FIG  tRNA  1169723  1169797  .  +  .   \n",
       "\n",
       "                                                      8              color  \n",
       "0     ID=fig|1448933.3.peg.1;Name=beta-glucosidase (...  fill_color=805642  \n",
       "1      ID=fig|1448933.3.peg.2;Name=hypothetical protein                NaN  \n",
       "2      ID=fig|1448933.3.peg.3;Name=hypothetical protein                NaN  \n",
       "3      ID=fig|1448933.3.peg.4;Name=hypothetical protein                NaN  \n",
       "4      ID=fig|1448933.3.peg.5;Name=hypothetical protein  fill_color=d6aadf  \n",
       "...                                                 ...                ...  \n",
       "1550          ID=fig|1448933.3.rna.43;Name=tRNA-Gly-TCC                NaN  \n",
       "1551          ID=fig|1448933.3.rna.43;Name=tRNA-Gly-TCC                NaN  \n",
       "1552          ID=fig|1448933.3.rna.44;Name=tRNA-Thr-TGT                NaN  \n",
       "1553          ID=fig|1448933.3.rna.44;Name=tRNA-Thr-TGT                NaN  \n",
       "1554          ID=fig|1448933.3.rna.44;Name=tRNA-Thr-TGT                NaN  \n",
       "\n",
       "[1555 rows x 10 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff_geo3918"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "rna_cas = gff_geo3918[(gff_geo3918[8].str.contains('Cass'))|(gff_geo3918[8].str.contains('Name=tRNA-'))|(gff_geo3918[8].str.contains('rRNA_'))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in rna_cas.iterrows():\n",
    "    if 'Name=tRNA-' in row[8]:\n",
    "        rna_cas.loc[idx, 'color'] = 'fill_color=vvdblue'\n",
    "for idx, row in rna_cas.iterrows():\n",
    "    if 'Cass' in row[8]:\n",
    "        rna_cas.loc[idx, 'color'] = 'fill_color=vdred'\n",
    "for idx, row in rna_cas.iterrows():\n",
    "    if 'rRNA_' in row[8]:\n",
    "        rna_cas.loc[idx, 'color'] = 'fill_color=vvdgreen'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0       False\n",
       "1       False\n",
       "2       False\n",
       "3       False\n",
       "4       False\n",
       "        ...  \n",
       "1550     True\n",
       "1551     True\n",
       "1552     True\n",
       "1553     True\n",
       "1554     True\n",
       "Name: 8, Length: 1555, dtype: bool"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff_geo3918[8].isin(rna_cas[8])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_geo3918 =gff_geo3918[~gff_geo3918[8].isin(rna_cas[8])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "gff_geo3918_plus_strand = gff_geo3918[gff_geo3918[6] == '+']\n",
    "gff_geo3918_minus_strand = gff_geo3918[gff_geo3918[6] == '-']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>color</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4690</td>\n",
       "      <td>5208</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>ID=fig|1448933.3.peg.5;Name=hypothetical protein</td>\n",
       "      <td>fill_color=d6aadf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>15761</td>\n",
       "      <td>15955</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>ID=fig|1448933.3.peg.16;Name=hypothetical protein</td>\n",
       "      <td>fill_color=805642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>15989</td>\n",
       "      <td>17098</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>ID=fig|1448933.3.peg.17;Name=N-acyl-L-amino ac...</td>\n",
       "      <td>fill_color=99334d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>19106</td>\n",
       "      <td>19795</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>ID=fig|1448933.3.peg.20;Name=Sugar fermentatio...</td>\n",
       "      <td>fill_color=d6aadf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>19792</td>\n",
       "      <td>20514</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>ID=fig|1448933.3.peg.21;Name=Oxidoreductase%2C...</td>\n",
       "      <td>fill_color=5c5a1b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1408</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>1410616</td>\n",
       "      <td>1411353</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>ID=fig|1448933.3.peg.1409;Name=hypothetical pr...</td>\n",
       "      <td>fill_color=5c5a1b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1410</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>1413740</td>\n",
       "      <td>1414174</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>ID=fig|1448933.3.peg.1411;Name=hypothetical pr...</td>\n",
       "      <td>fill_color=805642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1417</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>1421814</td>\n",
       "      <td>1423343</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>ID=fig|1448933.3.peg.1418;Name=hypothetical pr...</td>\n",
       "      <td>fill_color=d6aadf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1418</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>1423380</td>\n",
       "      <td>1424819</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>ID=fig|1448933.3.peg.1419;Name=hypothetical pr...</td>\n",
       "      <td>fill_color=805642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1422</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>1427650</td>\n",
       "      <td>1429827</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>ID=fig|1448933.3.peg.1423;Name=alpha-xylosidas...</td>\n",
       "      <td>fill_color=805642</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>695 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               0    1    2        3        4  5  6  7  \\\n",
       "4     contig_1_pilon_pilon_pilon  FIG  CDS     4690     5208  .  +  1   \n",
       "15    contig_1_pilon_pilon_pilon  FIG  CDS    15761    15955  .  +  2   \n",
       "16    contig_1_pilon_pilon_pilon  FIG  CDS    15989    17098  .  +  2   \n",
       "19    contig_1_pilon_pilon_pilon  FIG  CDS    19106    19795  .  +  2   \n",
       "20    contig_1_pilon_pilon_pilon  FIG  CDS    19792    20514  .  +  1   \n",
       "...                          ...  ...  ...      ...      ... .. .. ..   \n",
       "1408  contig_1_pilon_pilon_pilon  FIG  CDS  1410616  1411353  .  +  1   \n",
       "1410  contig_1_pilon_pilon_pilon  FIG  CDS  1413740  1414174  .  +  2   \n",
       "1417  contig_1_pilon_pilon_pilon  FIG  CDS  1421814  1423343  .  +  0   \n",
       "1418  contig_1_pilon_pilon_pilon  FIG  CDS  1423380  1424819  .  +  0   \n",
       "1422  contig_1_pilon_pilon_pilon  FIG  CDS  1427650  1429827  .  +  1   \n",
       "\n",
       "                                                      8              color  \n",
       "4      ID=fig|1448933.3.peg.5;Name=hypothetical protein  fill_color=d6aadf  \n",
       "15    ID=fig|1448933.3.peg.16;Name=hypothetical protein  fill_color=805642  \n",
       "16    ID=fig|1448933.3.peg.17;Name=N-acyl-L-amino ac...  fill_color=99334d  \n",
       "19    ID=fig|1448933.3.peg.20;Name=Sugar fermentatio...  fill_color=d6aadf  \n",
       "20    ID=fig|1448933.3.peg.21;Name=Oxidoreductase%2C...  fill_color=5c5a1b  \n",
       "...                                                 ...                ...  \n",
       "1408  ID=fig|1448933.3.peg.1409;Name=hypothetical pr...  fill_color=5c5a1b  \n",
       "1410  ID=fig|1448933.3.peg.1411;Name=hypothetical pr...  fill_color=805642  \n",
       "1417  ID=fig|1448933.3.peg.1418;Name=hypothetical pr...  fill_color=d6aadf  \n",
       "1418  ID=fig|1448933.3.peg.1419;Name=hypothetical pr...  fill_color=805642  \n",
       "1422  ID=fig|1448933.3.peg.1423;Name=alpha-xylosidas...  fill_color=805642  \n",
       "\n",
       "[695 rows x 10 columns]"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff_geo3918_plus_strand"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in gff_geo3918_plus_strand.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/geo3918/data/highlight_plus_strand.txt', 'a') as write_file:\n",
    "        write_file.write(f\"contig {row[3]} {row[4]} {row['color']}\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in gff_geo3918_minus_strand.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/geo3918/data/highlight_minus_strand.txt', 'a') as write_file:\n",
    "        write_file.write(f\"contig {row[3]} {row[4]} {row['color']}\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in rna_cas.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/geo3918/data/rna_cas_3_ring.txt', 'a') as write_file:\n",
    "        write_file.write(f\"contig {row[3]} {row[4]} {row['color']}\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### CG_skew"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def CG_cont(lis:str) -> float:\n",
    "    lis = lis.upper()\n",
    "    GC = lis.count('G') + lis.count('C')\n",
    "    return GC/len(lis)\n",
    "\n",
    "def CG_skew(lis:str) -> float:\n",
    "    lis = lis.upper()\n",
    "    G = lis.count('G')\n",
    "    C = lis.count('C')\n",
    "    return (G - C)/(G + C)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def slider(seq,sliding_window, step_size):\n",
    "    #sliding_window = 1000\n",
    "    step = 0\n",
    "    #step_size = 1000\n",
    "    r_bords = []\n",
    "    GCskew = []\n",
    "    #GCcont = []\n",
    "    l_bords = []\n",
    "    for i in range(0,len(seq)):\n",
    "\n",
    "        l_bord = step\n",
    "        r_bord = sliding_window + l_bord\n",
    "        step += step_size\n",
    "        window = seq[l_bord:r_bord]\n",
    "\n",
    "        l_bords.append(l_bord+1)\n",
    "        r_bords.append(r_bord-sliding_window+1)\n",
    "        GCskew.append(CG_skew(window))\n",
    "        #GCcont.append(CG_cont(window))\n",
    "\n",
    "        if r_bord >= len(seq):\n",
    "            break\n",
    "\n",
    "    GCskew.append((GCskew[0] + GCskew[-1])/2) #Замыкание кольца гистограммы\n",
    "    #GCcont.append((GCcont[0] + GCcont[-1])/2)\n",
    "    l_bords.append(l_bords[-1] + sliding_window-step_size )\n",
    "    r_bords.append(r_bords[-1]  + sliding_window )\n",
    "    return list(zip(l_bords,r_bords,GCskew))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "for sequence in SeqIO.parse(open('/home/aster/Project_Geoarchaeota_28.11/data/sequences_annotation/1448933.3.contigs.fa','r'), 'fasta'):\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/geo3918/data/geo3918_GC_skew.txt','a') as skew_file:\n",
    "\n",
    "        if len(sequence.seq) >= 5000:\n",
    "            lst = slider(str(sequence.seq), sliding_window=5000,step_size=1000)\n",
    "            for i,j,k in lst:\n",
    "                if k > 0:\n",
    "                    skew_file.write(f\"{sequence.name.split('_')[0]} {i} {j} {k} fill_color=green\\n\")\n",
    "                else:\n",
    "                    skew_file.write(f\"{sequence.name.split('_')[0]} {i} {j} {k} fill_color=red\\n\")\n",
    "\n",
    "        else:\n",
    "            lst = slider(str(sequence.seq), sliding_window=1000,step_size=1000)\n",
    "            for i,j,k in lst:\n",
    "                if k > 0:\n",
    "                    skew_file.write(f\"{sequence.name.split('_')[0]} {i} {j} {k} fill_color=green\\n\")\n",
    "                else:\n",
    "                    skew_file.write(f\"{sequence.name.split('_')[0]} {i} {j} {k} fill_color=red\\n\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Создание того же но для geo3817"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "for num, seq in enumerate(SeqIO.parse(open('/home/aster/Project_Geoarchaeota_28.11/3817-70/data/sequence_annotation/1448933.6.contigs.fa'), 'fasta'), start=1):\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/geo3817/data/geo3817_karyotype.txt', 'a') as write_file:\n",
    "        write_file.write(f\"chr - {seq.name} contig_{num} 0 {len(seq)} blue\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_geo3817 = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/seq_data/gff_files/geo_3817.gff', sep='\\t', header=None)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "eggNOG_annot = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/seq_data/eggNOG_annot/geo3817_eggNOG.tsv', sep='\\t', header=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "rna_cas = gff_geo3817[(gff_geo3817[8].str.contains('Cass'))|(gff_geo3817[8].str.contains('Name=tRNA-'))|(gff_geo3817[8].str.contains('rRNA_'))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in gff_geo3817.iterrows():\n",
    "    for idx2, row2, in eggNOG_annot.iterrows():\n",
    "        if row2['Name'] in row[8]:\n",
    "            gff_geo3817.loc[idx, 'color'] = f\"fill_color={d[row2['COG_category']]}\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>color</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NODE_10_length_11968_cov_535.958785</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2</td>\n",
       "      <td>208</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>ID=fig|1448933.6.peg.1;Name=hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NODE_10_length_11968_cov_535.958785</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>266</td>\n",
       "      <td>2443</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>ID=fig|1448933.6.peg.2;Name=alpha-xylosidase (...</td>\n",
       "      <td>fill_color=805642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NODE_10_length_11968_cov_535.958785</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2465</td>\n",
       "      <td>2680</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>ID=fig|1448933.6.peg.3;Name=hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NODE_10_length_11968_cov_535.958785</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2729</td>\n",
       "      <td>3592</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>ID=fig|1448933.6.peg.4;Name=hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NODE_10_length_11968_cov_535.958785</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>3615</td>\n",
       "      <td>4316</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>ID=fig|1448933.6.peg.5;Name=7-cyano-7-deazagua...</td>\n",
       "      <td>fill_color=727dcc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1627</th>\n",
       "      <td>NODE_8_length_17721_cov_583.980471</td>\n",
       "      <td>FIG</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>10958</td>\n",
       "      <td>11042</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=fig|1448933.6.rna.42;Name=tRNA-Ser-CGA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1628</th>\n",
       "      <td>NODE_8_length_17721_cov_583.980471</td>\n",
       "      <td>FIG</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>10958</td>\n",
       "      <td>11042</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=fig|1448933.6.rna.42;Name=tRNA-Ser-CGA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1629</th>\n",
       "      <td>NODE_8_length_17721_cov_583.980471</td>\n",
       "      <td>FIG</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>16531</td>\n",
       "      <td>16630</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=fig|1448933.6.rna.43;Name=tRNA-Ser-GGA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1630</th>\n",
       "      <td>NODE_8_length_17721_cov_583.980471</td>\n",
       "      <td>FIG</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>16531</td>\n",
       "      <td>16630</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=fig|1448933.6.rna.43;Name=tRNA-Ser-GGA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1631</th>\n",
       "      <td>NODE_8_length_17721_cov_583.980471</td>\n",
       "      <td>FIG</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>16531</td>\n",
       "      <td>16630</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=fig|1448933.6.rna.43;Name=tRNA-Ser-GGA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1632 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        0    1     2      3      4  5  6  7  \\\n",
       "0     NODE_10_length_11968_cov_535.958785  FIG   CDS      2    208  .  +  2   \n",
       "1     NODE_10_length_11968_cov_535.958785  FIG   CDS    266   2443  .  -  2   \n",
       "2     NODE_10_length_11968_cov_535.958785  FIG   CDS   2465   2680  .  -  2   \n",
       "3     NODE_10_length_11968_cov_535.958785  FIG   CDS   2729   3592  .  +  2   \n",
       "4     NODE_10_length_11968_cov_535.958785  FIG   CDS   3615   4316  .  +  0   \n",
       "...                                   ...  ...   ...    ...    ... .. .. ..   \n",
       "1627   NODE_8_length_17721_cov_583.980471  FIG  tRNA  10958  11042  .  -  .   \n",
       "1628   NODE_8_length_17721_cov_583.980471  FIG  tRNA  10958  11042  .  -  .   \n",
       "1629   NODE_8_length_17721_cov_583.980471  FIG  tRNA  16531  16630  .  +  .   \n",
       "1630   NODE_8_length_17721_cov_583.980471  FIG  tRNA  16531  16630  .  +  .   \n",
       "1631   NODE_8_length_17721_cov_583.980471  FIG  tRNA  16531  16630  .  +  .   \n",
       "\n",
       "                                                      8              color  \n",
       "0      ID=fig|1448933.6.peg.1;Name=hypothetical protein                NaN  \n",
       "1     ID=fig|1448933.6.peg.2;Name=alpha-xylosidase (...  fill_color=805642  \n",
       "2      ID=fig|1448933.6.peg.3;Name=hypothetical protein                NaN  \n",
       "3      ID=fig|1448933.6.peg.4;Name=hypothetical protein                NaN  \n",
       "4     ID=fig|1448933.6.peg.5;Name=7-cyano-7-deazagua...  fill_color=727dcc  \n",
       "...                                                 ...                ...  \n",
       "1627          ID=fig|1448933.6.rna.42;Name=tRNA-Ser-CGA                NaN  \n",
       "1628          ID=fig|1448933.6.rna.42;Name=tRNA-Ser-CGA                NaN  \n",
       "1629          ID=fig|1448933.6.rna.43;Name=tRNA-Ser-GGA                NaN  \n",
       "1630          ID=fig|1448933.6.rna.43;Name=tRNA-Ser-GGA                NaN  \n",
       "1631          ID=fig|1448933.6.rna.43;Name=tRNA-Ser-GGA                NaN  \n",
       "\n",
       "[1632 rows x 10 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff_geo3817"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_69016/1153070891.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  rna_cas.loc[idx, 'color'] = 'fill_color=vvdblue'\n"
     ]
    }
   ],
   "source": [
    "for idx, row in rna_cas.iterrows():\n",
    "    if 'Name=tRNA-' in row[8]:\n",
    "        rna_cas.loc[idx, 'color'] = 'fill_color=vvdblue'\n",
    "for idx, row in rna_cas.iterrows():\n",
    "    if 'Cass' in row[8]:\n",
    "        rna_cas.loc[idx, 'color'] = 'fill_color=vdred'\n",
    "for idx, row in rna_cas.iterrows():\n",
    "    if 'rRNA_' in row[8]:\n",
    "        rna_cas.loc[idx, 'color'] = 'fill_color=vvdgreen'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_geo3817 = gff_geo3817[~gff_geo3817[8].isin(rna_cas[8])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_geo3817_plus_strand = gff_geo3817[gff_geo3817[6] == '+']\n",
    "gff_geo3817_minus_strand = gff_geo3817[gff_geo3817[6] == '-']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in gff_geo3817_plus_strand.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/geo3817/data/highlight_plus_strand.txt', 'a') as write_file:\n",
    "        write_file.write(f\"{row[0]} {row[3]} {row[4]} {row['color']}\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in gff_geo3817_minus_strand.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/geo3817/data/highlight_minus_strand.txt', 'a') as write_file:\n",
    "        write_file.write(f\"{row[0]} {row[3]} {row[4]} {row['color']}\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in rna_cas.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/geo3817/data/rna_cas_3_ring.txt', 'a') as write_file:\n",
    "        write_file.write(f\"{row[0]} {row[3]} {row[4]} {row['color']}\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "for sequence in SeqIO.parse(open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/seq_data/fna_files/geo3817_contigs.fa','r'), 'fasta'):\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/geo3817/data/geo3918_GC_skew.txt','a') as skew_file:\n",
    "\n",
    "        if len(sequence.seq) >= 5000:\n",
    "            lst = slider(str(sequence.seq), sliding_window=5000,step_size=1000)\n",
    "            for i,j,k in lst:\n",
    "                if k > 0:\n",
    "                    skew_file.write(f\"{sequence.name} {i} {j} {k} fill_color=green\\n\")\n",
    "                else:\n",
    "                    skew_file.write(f\"{sequence.name} {i} {j} {k} fill_color=red\\n\")\n",
    "\n",
    "        else:\n",
    "            lst = slider(str(sequence.seq), sliding_window=1000,step_size=1000)\n",
    "            for i,j,k in lst:\n",
    "                if k > 0:\n",
    "                    skew_file.write(f\"{sequence.name} {i} {j} {k} fill_color=green\\n\")\n",
    "                else:\n",
    "                    skew_file.write(f\"{sequence.name} {i} {j} {k} fill_color=red\\n\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## geo both"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "for num, seq in enumerate(SeqIO.parse(open('/home/aster/Project_Geoarchaeota_28.11/data/sequences_annotation/1448933.3.contigs.fa'), 'fasta'), start=1):\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/geo3918_geo3817/data/karyotype_both.txt', 'a') as write_file:\n",
    "        write_file.write(f\"chr - {seq.name} cn_{num} 0 {len(seq)} green\\n\")\n",
    "        \n",
    "for num, seq in enumerate(SeqIO.parse(open('/home/aster/Project_Geoarchaeota_28.11/3817-70/data/sequence_annotation/1448933.6.contigs.fa'), 'fasta'), start=1):\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/geo3918_geo3817/data/karyotype_both.txt', 'a') as write_file:\n",
    "        write_file.write(f\"chr - {seq.name} contig_{num} 0 {len(seq)} blue\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = {'J':'ff0000', 'A':'c2af58', 'K':'ff9900', 'L':'ffff00', 'B':'ffc600',\n",
    "    'D':'99ff00', 'Y':'493126', 'V':'ff008a', 'T':'0000ff', 'M':'9ec928',\n",
    "    'N':'006633', 'Z':'660099', 'W':'336699', 'U':'33cc99', 'O':'00ffff', \n",
    "    'C':'9900ff', 'G':'805642', 'E':'99334d', 'H':'727dcc', 'I':'5c5a1b',\n",
    "    'P':'0099ff', 'Q':'ffcc99', 'R':'ff9999','S':'d6aadf', 'F':'99334d',\n",
    "    'IQ':'ffcc99',\n",
    "    '-':'white',\n",
    "    'EGP':'b8b100',\n",
    "    'CP':'9900ff',\n",
    "    'JM':'ff0000',\n",
    "    'EP':'99334d',\n",
    "    'NU':'006633',\n",
    "    'HP':'727dcc',\n",
    "    'GM':'805642',\n",
    "    ################\n",
    "    'LT':'ffff00',\n",
    "    'KL':'ff9900',\n",
    "    'BQ':'ffc600',\n",
    "    'EH':'99334d',\n",
    "    'FG':'99334d',\n",
    "    'OU':'00ffff',\n",
    "    'CE':'9900ff'\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_geo3918 = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/data/sequences_annotation/1448933.3.gff', sep='\\t', header=None)\n",
    "eggNOG_annot_geo3918 = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/results/eggNog_mapper/MM_k15bh4a0.emapper.annotations(1).tsv', sep='\\t')\n",
    "gff_geo3817 = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/3817-70/data/sequence_annotation/1448933.6.gff', sep='\\t', header=None)\n",
    "eggNOG_annot_geo3817 = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/3817-70/results/eggNOG5/MM_69m96w52.emapper.annotations.tsv', sep='\\t', header=4)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in gff_geo3918.iterrows():\n",
    "    for idx2, row2, in eggNOG_annot_geo3918.iterrows():\n",
    "        if row2['ID'] in row[8]:\n",
    "            gff_geo3918.loc[idx, 'color'] = d[row2['COG_category']]\n",
    "\n",
    "for idx, row in gff_geo3817.iterrows():\n",
    "    for idx2, row2, in eggNOG_annot_geo3817.iterrows():\n",
    "        if row2['Name'] in row[8]:\n",
    "            gff_geo3817.loc[idx, 'color'] = d[row2['COG_category']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_geo3918_plus_strand = gff_geo3918[gff_geo3918[6] == '+']\n",
    "gff_geo3918_minus_strand = gff_geo3918[gff_geo3918[6] == '-']\n",
    "gff_geo3817_plus_strand = gff_geo3817[gff_geo3817[6] == '+']\n",
    "gff_geo3817_minus_strand = gff_geo3817[gff_geo3817[6] == '-']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in gff_geo3918_plus_strand.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/geo3918_geo3817/data/highlight_plus_strand.txt', 'a') as write_file:\n",
    "        write_file.write(f\"{row[0]} {row[3]} {row[4]} {row['color']}\\n\")\n",
    "\n",
    "for idx, row in gff_geo3918_minus_strand.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/geo3918_geo3817/data/highlight_minus_strand.txt', 'a') as write_file:\n",
    "        write_file.write(f\"{row[0]} {row[3]} {row[4]} {row['color']}\\n\")\n",
    "\n",
    "for idx, row in gff_geo3817_plus_strand.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/geo3918_geo3817/data/highlight_plus_strand.txt', 'a') as write_file:\n",
    "        write_file.write(f\"{row[0]} {row[3]} {row[4]} {row['color']}\\n\")\n",
    "\n",
    "for idx, row in gff_geo3817_minus_strand.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/geo3918_geo3817/data/highlight_minus_strand.txt', 'a') as write_file:\n",
    "        write_file.write(f\"{row[0]} {row[3]} {row[4]} {row['color']}\\n\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Coliniary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [],
   "source": [
    "seed_compare = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/results/SEED/geo2geo_compare/Geo3918_toGeo3817_seq_compar.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [],
   "source": [
    "seed_compare_parsed = seed_compare[['Gene id', 'Gene id.1']]\n",
    "seed_compare_parsed = seed_compare_parsed[seed_compare_parsed['Gene id.1'] != ' ']\n",
    "gff_geo3918['color'] = gff_geo3918['color'].fillna('gray_a2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>color</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2</td>\n",
       "      <td>2227</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>ID=fig|1448933.3.peg.1;Name=beta-glucosidase (...</td>\n",
       "      <td>805642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2259</td>\n",
       "      <td>3155</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>ID=fig|1448933.3.peg.2;Name=hypothetical protein</td>\n",
       "      <td>gray_a2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>3265</td>\n",
       "      <td>4527</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>ID=fig|1448933.3.peg.3;Name=hypothetical protein</td>\n",
       "      <td>gray_a2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4530</td>\n",
       "      <td>4667</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>ID=fig|1448933.3.peg.4;Name=hypothetical protein</td>\n",
       "      <td>gray_a2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4690</td>\n",
       "      <td>5208</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>ID=fig|1448933.3.peg.5;Name=hypothetical protein</td>\n",
       "      <td>d6aadf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1550</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>1124017</td>\n",
       "      <td>1124091</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=fig|1448933.3.rna.43;Name=tRNA-Gly-TCC</td>\n",
       "      <td>gray_a2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1551</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>1124017</td>\n",
       "      <td>1124091</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=fig|1448933.3.rna.43;Name=tRNA-Gly-TCC</td>\n",
       "      <td>gray_a2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1552</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>1169723</td>\n",
       "      <td>1169797</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=fig|1448933.3.rna.44;Name=tRNA-Thr-TGT</td>\n",
       "      <td>gray_a2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1553</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>1169723</td>\n",
       "      <td>1169797</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=fig|1448933.3.rna.44;Name=tRNA-Thr-TGT</td>\n",
       "      <td>gray_a2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1554</th>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>FIG</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>1169723</td>\n",
       "      <td>1169797</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=fig|1448933.3.rna.44;Name=tRNA-Thr-TGT</td>\n",
       "      <td>gray_a2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1555 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               0    1     2        3        4  5  6  7  \\\n",
       "0     contig_1_pilon_pilon_pilon  FIG   CDS        2     2227  .  -  2   \n",
       "1     contig_1_pilon_pilon_pilon  FIG   CDS     2259     3155  .  -  0   \n",
       "2     contig_1_pilon_pilon_pilon  FIG   CDS     3265     4527  .  -  1   \n",
       "3     contig_1_pilon_pilon_pilon  FIG   CDS     4530     4667  .  -  0   \n",
       "4     contig_1_pilon_pilon_pilon  FIG   CDS     4690     5208  .  +  1   \n",
       "...                          ...  ...   ...      ...      ... .. .. ..   \n",
       "1550  contig_1_pilon_pilon_pilon  FIG  tRNA  1124017  1124091  .  -  .   \n",
       "1551  contig_1_pilon_pilon_pilon  FIG  tRNA  1124017  1124091  .  -  .   \n",
       "1552  contig_1_pilon_pilon_pilon  FIG  tRNA  1169723  1169797  .  +  .   \n",
       "1553  contig_1_pilon_pilon_pilon  FIG  tRNA  1169723  1169797  .  +  .   \n",
       "1554  contig_1_pilon_pilon_pilon  FIG  tRNA  1169723  1169797  .  +  .   \n",
       "\n",
       "                                                      8    color  \n",
       "0     ID=fig|1448933.3.peg.1;Name=beta-glucosidase (...   805642  \n",
       "1      ID=fig|1448933.3.peg.2;Name=hypothetical protein  gray_a2  \n",
       "2      ID=fig|1448933.3.peg.3;Name=hypothetical protein  gray_a2  \n",
       "3      ID=fig|1448933.3.peg.4;Name=hypothetical protein  gray_a2  \n",
       "4      ID=fig|1448933.3.peg.5;Name=hypothetical protein   d6aadf  \n",
       "...                                                 ...      ...  \n",
       "1550          ID=fig|1448933.3.rna.43;Name=tRNA-Gly-TCC  gray_a2  \n",
       "1551          ID=fig|1448933.3.rna.43;Name=tRNA-Gly-TCC  gray_a2  \n",
       "1552          ID=fig|1448933.3.rna.44;Name=tRNA-Thr-TGT  gray_a2  \n",
       "1553          ID=fig|1448933.3.rna.44;Name=tRNA-Thr-TGT  gray_a2  \n",
       "1554          ID=fig|1448933.3.rna.44;Name=tRNA-Thr-TGT  gray_a2  \n",
       "\n",
       "[1555 rows x 10 columns]"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff_geo3918"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in seed_compare_parsed.iterrows():\n",
    "    for idx2, row2 in gff_geo3918.iterrows():\n",
    "        if row['Gene id'] in row2[8]:\n",
    "            seed_compare_parsed.loc[idx,'start1'] = row2[3]\n",
    "            seed_compare_parsed.loc[idx,'stop1'] = row2[4]\n",
    "            seed_compare_parsed.loc[idx,'link_id_1'] = row2[0]\n",
    "            seed_compare_parsed.loc[idx,'color'] = row2['color']\n",
    "\n",
    "for idx, row in seed_compare_parsed.iterrows():\n",
    "    for idx2, row2 in gff_geo3817.iterrows():\n",
    "        if row['Gene id.1'] in row2[8]:\n",
    "            seed_compare_parsed.loc[idx,'start2'] = row2[3]\n",
    "            seed_compare_parsed.loc[idx,'stop2'] = row2[4]\n",
    "            seed_compare_parsed.loc[idx,'link_id_2'] = row2[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene id</th>\n",
       "      <th>Gene id.1</th>\n",
       "      <th>start1</th>\n",
       "      <th>stop1</th>\n",
       "      <th>link_id_1</th>\n",
       "      <th>color</th>\n",
       "      <th>start2</th>\n",
       "      <th>stop2</th>\n",
       "      <th>link_id_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>fig|1448933.3.peg.1</td>\n",
       "      <td>fig|1448933.6.peg.627</td>\n",
       "      <td>1427650.0</td>\n",
       "      <td>1429827.0</td>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>805642</td>\n",
       "      <td>603733.0</td>\n",
       "      <td>605979.0</td>\n",
       "      <td>NODE_1_length_696707_cov_578.980505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>fig|1448933.3.peg.2</td>\n",
       "      <td>fig|1448933.6.peg.626</td>\n",
       "      <td>311988.0</td>\n",
       "      <td>312881.0</td>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>ff0000</td>\n",
       "      <td>602805.0</td>\n",
       "      <td>603701.0</td>\n",
       "      <td>NODE_1_length_696707_cov_578.980505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>fig|1448933.3.peg.3</td>\n",
       "      <td>fig|1448933.6.peg.625</td>\n",
       "      <td>402182.0</td>\n",
       "      <td>402880.0</td>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>d6aadf</td>\n",
       "      <td>601330.0</td>\n",
       "      <td>602553.0</td>\n",
       "      <td>NODE_1_length_696707_cov_578.980505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>fig|1448933.3.peg.5</td>\n",
       "      <td>fig|1448933.6.peg.624</td>\n",
       "      <td>584564.0</td>\n",
       "      <td>584941.0</td>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>9900ff</td>\n",
       "      <td>600737.0</td>\n",
       "      <td>601276.0</td>\n",
       "      <td>NODE_1_length_696707_cov_578.980505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>fig|1448933.3.peg.6</td>\n",
       "      <td>fig|1448933.6.peg.623</td>\n",
       "      <td>686210.0</td>\n",
       "      <td>687178.0</td>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>white</td>\n",
       "      <td>599843.0</td>\n",
       "      <td>600688.0</td>\n",
       "      <td>NODE_1_length_696707_cov_578.980505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1418</th>\n",
       "      <td>fig|1448933.3.peg.1419</td>\n",
       "      <td>fig|1448933.6.peg.8</td>\n",
       "      <td>1423380.0</td>\n",
       "      <td>1424819.0</td>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>805642</td>\n",
       "      <td>181850.0</td>\n",
       "      <td>182095.0</td>\n",
       "      <td>NODE_2_length_425441_cov_628.256478</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1419</th>\n",
       "      <td>fig|1448933.3.peg.1420</td>\n",
       "      <td>fig|1448933.6.peg.6</td>\n",
       "      <td>1425086.0</td>\n",
       "      <td>1425847.0</td>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>805642</td>\n",
       "      <td>687629.0</td>\n",
       "      <td>688402.0</td>\n",
       "      <td>NODE_1_length_696707_cov_578.980505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1420</th>\n",
       "      <td>fig|1448933.3.peg.1421</td>\n",
       "      <td>fig|1448933.6.peg.5</td>\n",
       "      <td>1425844.0</td>\n",
       "      <td>1426545.0</td>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>727dcc</td>\n",
       "      <td>574697.0</td>\n",
       "      <td>575809.0</td>\n",
       "      <td>NODE_1_length_696707_cov_578.980505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1421</th>\n",
       "      <td>fig|1448933.3.peg.1422</td>\n",
       "      <td>fig|1448933.6.peg.4</td>\n",
       "      <td>1426568.0</td>\n",
       "      <td>1427266.0</td>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>d6aadf</td>\n",
       "      <td>478568.0</td>\n",
       "      <td>479374.0</td>\n",
       "      <td>NODE_1_length_696707_cov_578.980505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1422</th>\n",
       "      <td>fig|1448933.3.peg.1423</td>\n",
       "      <td>fig|1448933.6.peg.2</td>\n",
       "      <td>1427650.0</td>\n",
       "      <td>1429827.0</td>\n",
       "      <td>contig_1_pilon_pilon_pilon</td>\n",
       "      <td>805642</td>\n",
       "      <td>258917.0</td>\n",
       "      <td>260410.0</td>\n",
       "      <td>NODE_1_length_696707_cov_578.980505</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1222 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Gene id              Gene id.1     start1      stop1  \\\n",
       "0        fig|1448933.3.peg.1  fig|1448933.6.peg.627  1427650.0  1429827.0   \n",
       "1        fig|1448933.3.peg.2  fig|1448933.6.peg.626   311988.0   312881.0   \n",
       "2        fig|1448933.3.peg.3  fig|1448933.6.peg.625   402182.0   402880.0   \n",
       "4        fig|1448933.3.peg.5  fig|1448933.6.peg.624   584564.0   584941.0   \n",
       "5        fig|1448933.3.peg.6  fig|1448933.6.peg.623   686210.0   687178.0   \n",
       "...                      ...                    ...        ...        ...   \n",
       "1418  fig|1448933.3.peg.1419    fig|1448933.6.peg.8  1423380.0  1424819.0   \n",
       "1419  fig|1448933.3.peg.1420    fig|1448933.6.peg.6  1425086.0  1425847.0   \n",
       "1420  fig|1448933.3.peg.1421    fig|1448933.6.peg.5  1425844.0  1426545.0   \n",
       "1421  fig|1448933.3.peg.1422    fig|1448933.6.peg.4  1426568.0  1427266.0   \n",
       "1422  fig|1448933.3.peg.1423    fig|1448933.6.peg.2  1427650.0  1429827.0   \n",
       "\n",
       "                       link_id_1   color    start2     stop2  \\\n",
       "0     contig_1_pilon_pilon_pilon  805642  603733.0  605979.0   \n",
       "1     contig_1_pilon_pilon_pilon  ff0000  602805.0  603701.0   \n",
       "2     contig_1_pilon_pilon_pilon  d6aadf  601330.0  602553.0   \n",
       "4     contig_1_pilon_pilon_pilon  9900ff  600737.0  601276.0   \n",
       "5     contig_1_pilon_pilon_pilon   white  599843.0  600688.0   \n",
       "...                          ...     ...       ...       ...   \n",
       "1418  contig_1_pilon_pilon_pilon  805642  181850.0  182095.0   \n",
       "1419  contig_1_pilon_pilon_pilon  805642  687629.0  688402.0   \n",
       "1420  contig_1_pilon_pilon_pilon  727dcc  574697.0  575809.0   \n",
       "1421  contig_1_pilon_pilon_pilon  d6aadf  478568.0  479374.0   \n",
       "1422  contig_1_pilon_pilon_pilon  805642  258917.0  260410.0   \n",
       "\n",
       "                                link_id_2  \n",
       "0     NODE_1_length_696707_cov_578.980505  \n",
       "1     NODE_1_length_696707_cov_578.980505  \n",
       "2     NODE_1_length_696707_cov_578.980505  \n",
       "4     NODE_1_length_696707_cov_578.980505  \n",
       "5     NODE_1_length_696707_cov_578.980505  \n",
       "...                                   ...  \n",
       "1418  NODE_2_length_425441_cov_628.256478  \n",
       "1419  NODE_1_length_696707_cov_578.980505  \n",
       "1420  NODE_1_length_696707_cov_578.980505  \n",
       "1421  NODE_1_length_696707_cov_578.980505  \n",
       "1422  NODE_1_length_696707_cov_578.980505  \n",
       "\n",
       "[1222 rows x 9 columns]"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seed_compare_parsed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in seed_compare_parsed.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/geo3918_geo3817/data/links.txt', 'a') as link_file:\n",
    "        link_file.write(f\"rep{idx} {row['link_id_1']} {int(row['start1'])} {int(row['stop1'])}\\nrep{idx} {row['link_id_2']} {int(row['start2'])} {int(row['stop2'])}\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in seed_compare_parsed.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/geo3918_geo3817/data/links.txt', 'a') as link_file:\n",
    "        link_file.write(f\"{row['link_id_1']} {int(row['start1'])} {int(row['stop1'])} {row['link_id_2']} {int(row['start2'])} {int(row['stop2'])} color={row['color']}\\n\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Связи но с Jay 535 и Geo3918"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Еще сделать цветак как в COG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "for num, seq in enumerate(SeqIO.parse(open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/seq_data/fna_files/Jay_535_contigs.fa'), 'fasta'), start=1):\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/Jay_535_geo3918/data/karyotype_both.txt', 'a') as write_file:\n",
    "        write_file.write(f\"chr - {seq.name} cn_{num} 0 {len(seq)} blue\\n\")\n",
    "        \n",
    "for num, seq in enumerate(SeqIO.parse(open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/seq_data/fna_files/geo3918_contigs.fa'), 'fasta'), start=1):\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/Jay_535_geo3918/data/karyotype_both.txt', 'a') as write_file:\n",
    "        write_file.write(f\"chr - {seq.name} contig_{num} 0 {len(seq)} green\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = {'J':'ff0000', 'A':'c2af58', 'K':'ff9900', 'L':'ffff00', 'B':'ffc600',\n",
    "    'D':'99ff00', 'Y':'493126', 'V':'ff008a', 'T':'0000ff', 'M':'9ec928',\n",
    "    'N':'006633', 'Z':'660099', 'W':'336699', 'U':'33cc99', 'O':'00ffff', \n",
    "    'C':'9900ff', 'G':'805642', 'E':'99334d', 'H':'727dcc', 'I':'5c5a1b',\n",
    "    'P':'0099ff', 'Q':'ffcc99', 'R':'ff9999','S':'d6aadf', 'F':'99334d',\n",
    "    'IQ':'ffcc99',\n",
    "    '-':'white',\n",
    "    'EGP':'b8b100',\n",
    "    'CP':'9900ff',\n",
    "    'JM':'ff0000',\n",
    "    'EP':'99334d',\n",
    "    'NU':'006633',\n",
    "    'HP':'727dcc',\n",
    "    'GM':'805642',\n",
    "    ################\n",
    "    'LT':'ffff00',\n",
    "    'KL':'ff9900',\n",
    "    'BQ':'ffc600',\n",
    "    'EH':'99334d',\n",
    "    'FG':'99334d',\n",
    "    'OU':'00ffff',\n",
    "    'CE':'9900ff',\n",
    "    'EQ':'99334d',\n",
    "    'GJM':'805642',\n",
    "    'EU':'99334d',\n",
    "    'FJ':'99334d'\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_geo3918 = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/seq_data/gff_files/geo_3918.gff', sep='\\t', header=None)\n",
    "eggNOG_annot_geo3918 = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/seq_data/eggNOG_annot/geo3918_eggNOG.tsv', sep='\\t')\n",
    "gff_Jay535 = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/seq_data/gff_files/Jay_535.gff', sep='\\t', header=None)\n",
    "eggNOG_annot_Jay535 = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/seq_data/eggNOG_annot/Jay_535_eggNOG.tsv', sep='\\t', header = 4)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "eggNOG_annot_Jay535['Unnamed: 6'] = eggNOG_annot_Jay535['Unnamed: 6'].apply(lambda x:x[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in gff_geo3918.iterrows():\n",
    "    for idx2, row2, in eggNOG_annot_geo3918.iterrows():\n",
    "        if row2['ID'] in row[8]:\n",
    "            gff_geo3918.loc[idx, 'color'] = d[row2['COG_category']]\n",
    "\n",
    "for idx, row in gff_Jay535.iterrows():\n",
    "    for idx2, row2, in eggNOG_annot_Jay535.iterrows():\n",
    "        if row2['Unnamed: 0'] in row[8]:\n",
    "            gff_Jay535.loc[idx, 'color'] = d[row2['Unnamed: 6']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_geo3918_plus_strand = gff_geo3918[gff_geo3918[6] == '+']\n",
    "gff_geo3918_minus_strand = gff_geo3918[gff_geo3918[6] == '-']\n",
    "gff_Jay535_plus_strand = gff_Jay535[gff_Jay535[6] == '+']\n",
    "gff_Jay535_minus_strand = gff_Jay535[gff_Jay535[6] == '-']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in gff_geo3918_plus_strand.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/Jay_535_geo3918/data/highlight_plus_strand.txt', 'a') as write_file:\n",
    "        write_file.write(f\"{row[0]} {row[3]} {row[4]} {row['color']}\\n\")\n",
    "\n",
    "for idx, row in gff_geo3918_minus_strand.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/Jay_535_geo3918/data/highlight_minus_strand.txt', 'a') as write_file:\n",
    "        write_file.write(f\"{row[0]} {row[3]} {row[4]} {row['color']}\\n\")\n",
    "\n",
    "for idx, row in gff_Jay535_plus_strand.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/Jay_535_geo3918/data/highlight_plus_strand.txt', 'a') as write_file:\n",
    "        write_file.write(f\"{row[0]} {row[3]} {row[4]} {row['color']}\\n\")\n",
    "\n",
    "for idx, row in gff_Jay535_minus_strand.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/Jay_535_geo3918/data/highlight_minus_strand.txt', 'a') as write_file:\n",
    "        write_file.write(f\"{row[0]} {row[3]} {row[4]} {row['color']}\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "seed_compare = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/seq_data/SEED_compare/geo3918_Jay_535.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "seed_compare_parsed = seed_compare[['Gene id', 'Gene id.1']]\n",
    "seed_compare_parsed = seed_compare_parsed[seed_compare_parsed['Gene id.1'] != ' ']\n",
    "gff_geo3918['color'] = gff_geo3918['color'].fillna('gray_a2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in seed_compare_parsed.iterrows():\n",
    "    for idx2, row2 in gff_geo3918.iterrows():\n",
    "        if row['Gene id'] in row2[8]:\n",
    "            seed_compare_parsed.loc[idx,'start1'] = row2[3]\n",
    "            seed_compare_parsed.loc[idx,'stop1'] = row2[4]\n",
    "            seed_compare_parsed.loc[idx,'link_id_1'] = row2[0]\n",
    "            seed_compare_parsed.loc[idx,'color'] = row2['color']\n",
    "\n",
    "for idx, row in seed_compare_parsed.iterrows():\n",
    "    for idx2, row2 in gff_Jay535.iterrows():\n",
    "        if row['Gene id.1'] in row2[8]:\n",
    "            seed_compare_parsed.loc[idx,'start2'] = row2[3]\n",
    "            seed_compare_parsed.loc[idx,'stop2'] = row2[4]\n",
    "            seed_compare_parsed.loc[idx,'link_id_2'] = row2[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in seed_compare_parsed.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/geo3918_geo3817/data/links.txt', 'a') as link_file:\n",
    "        link_file.write(f\"rep{idx} {row['link_id_1']} {int(row['start1'])} {int(row['stop1'])}\\nrep{idx} {row['link_id_2']} {int(row['start2'])} {int(row['stop2'])}\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in seed_compare_parsed.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/novel_circos_diag/Jay_535_geo3918/data/links.txt', 'a') as link_file:\n",
    "        link_file.write(f\"{row['link_id_1']} {int(row['start1'])} {int(row['stop1'])} {row['link_id_2']} {int(row['start2'])} {int(row['stop2'])} color={row['color']}\\n\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "bioinf",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "5c7c0e940973d1f7f557ece26bca6704213ece01202fcdffb2249dc40fb254cf"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
