{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_data = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/results/circos_diagramms/data/1448933.3.gff', delimiter='\\t', header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [],
   "source": [
    "tRNA_gff_data = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/results/circos_diagramms/data/tRNA.gff', delimiter='\\t', header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_data = pd.concat([gff_data, tRNA_gff_data])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "tRNA_gff_data = tRNA_gff_data.iloc[:,[3,4,6,8]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_data = gff_data.iloc[:,[3,4,6,8]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>6</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>2227</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=fig|1448933.3.peg.1;Name=beta-glucosidase (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2259</td>\n",
       "      <td>3155</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=fig|1448933.3.peg.2;Name=hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3265</td>\n",
       "      <td>4527</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=fig|1448933.3.peg.3;Name=hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4530</td>\n",
       "      <td>4667</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=fig|1448933.3.peg.4;Name=hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4690</td>\n",
       "      <td>5208</td>\n",
       "      <td>+</td>\n",
       "      <td>ID=fig|1448933.3.peg.5;Name=hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>1124017</td>\n",
       "      <td>1124091</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=fig|1448933.3.rna.43;Name=tRNA-Gly-TCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>1124017</td>\n",
       "      <td>1124091</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=fig|1448933.3.rna.43;Name=tRNA-Gly-TCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>1169723</td>\n",
       "      <td>1169797</td>\n",
       "      <td>+</td>\n",
       "      <td>ID=fig|1448933.3.rna.44;Name=tRNA-Thr-TGT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>1169723</td>\n",
       "      <td>1169797</td>\n",
       "      <td>+</td>\n",
       "      <td>ID=fig|1448933.3.rna.44;Name=tRNA-Thr-TGT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>1169723</td>\n",
       "      <td>1169797</td>\n",
       "      <td>+</td>\n",
       "      <td>ID=fig|1448933.3.rna.44;Name=tRNA-Thr-TGT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1555 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           3        4  6                                                  8\n",
       "0          2     2227  -  ID=fig|1448933.3.peg.1;Name=beta-glucosidase (...\n",
       "1       2259     3155  -   ID=fig|1448933.3.peg.2;Name=hypothetical protein\n",
       "2       3265     4527  -   ID=fig|1448933.3.peg.3;Name=hypothetical protein\n",
       "3       4530     4667  -   ID=fig|1448933.3.peg.4;Name=hypothetical protein\n",
       "4       4690     5208  +   ID=fig|1448933.3.peg.5;Name=hypothetical protein\n",
       "..       ...      ... ..                                                ...\n",
       "127  1124017  1124091  -          ID=fig|1448933.3.rna.43;Name=tRNA-Gly-TCC\n",
       "128  1124017  1124091  -          ID=fig|1448933.3.rna.43;Name=tRNA-Gly-TCC\n",
       "129  1169723  1169797  +          ID=fig|1448933.3.rna.44;Name=tRNA-Thr-TGT\n",
       "130  1169723  1169797  +          ID=fig|1448933.3.rna.44;Name=tRNA-Thr-TGT\n",
       "131  1169723  1169797  +          ID=fig|1448933.3.rna.44;Name=tRNA-Thr-TGT\n",
       "\n",
       "[1555 rows x 4 columns]"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_data[8] = [i.split('|')[1].split(';')[0] for i in gff_data[8]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_143024/826490612.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tRNA_gff_data[8] = [i.split('|')[1].split(';')[0] for i in tRNA_gff_data[8]]\n"
     ]
    }
   ],
   "source": [
    "tRNA_gff_data[8] = [i.split('|')[1].split(';')[0] for i in tRNA_gff_data[8]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>6</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>244124</td>\n",
       "      <td>244196</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>244124</td>\n",
       "      <td>244196</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>244124</td>\n",
       "      <td>244196</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>274841</td>\n",
       "      <td>274925</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>274841</td>\n",
       "      <td>274925</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>1124017</td>\n",
       "      <td>1124091</td>\n",
       "      <td>-</td>\n",
       "      <td>1448933.3.rna.43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>1124017</td>\n",
       "      <td>1124091</td>\n",
       "      <td>-</td>\n",
       "      <td>1448933.3.rna.43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>1169723</td>\n",
       "      <td>1169797</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>1169723</td>\n",
       "      <td>1169797</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>1169723</td>\n",
       "      <td>1169797</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.44</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>132 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           3        4  6                 8\n",
       "0     244124   244196  +   1448933.3.rna.1\n",
       "1     244124   244196  +   1448933.3.rna.1\n",
       "2     244124   244196  +   1448933.3.rna.1\n",
       "3     274841   274925  +   1448933.3.rna.2\n",
       "4     274841   274925  +   1448933.3.rna.2\n",
       "..       ...      ... ..               ...\n",
       "127  1124017  1124091  -  1448933.3.rna.43\n",
       "128  1124017  1124091  -  1448933.3.rna.43\n",
       "129  1169723  1169797  +  1448933.3.rna.44\n",
       "130  1169723  1169797  +  1448933.3.rna.44\n",
       "131  1169723  1169797  +  1448933.3.rna.44\n",
       "\n",
       "[132 rows x 4 columns]"
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tRNA_gff_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_eggnog = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/results/eggNog_mapper/MM_k15bh4a0.emapper.annotations(1).tsv', delimiter='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>seed_ortholog</th>\n",
       "      <th>evalue</th>\n",
       "      <th>score</th>\n",
       "      <th>eggNOG_OGs</th>\n",
       "      <th>max_annot_lvl</th>\n",
       "      <th>COG_category</th>\n",
       "      <th>Description</th>\n",
       "      <th>Preferred_name</th>\n",
       "      <th>GOs</th>\n",
       "      <th>...</th>\n",
       "      <th>KEGG_ko</th>\n",
       "      <th>KEGG_Pathway</th>\n",
       "      <th>KEGG_Module</th>\n",
       "      <th>KEGG_Reaction</th>\n",
       "      <th>KEGG_rclass</th>\n",
       "      <th>BRITE</th>\n",
       "      <th>KEGG_TC</th>\n",
       "      <th>CAZy</th>\n",
       "      <th>BiGG_Reaction</th>\n",
       "      <th>PFAMs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>fig|1448933.3.peg.1</td>\n",
       "      <td>930945.SiRe_2208</td>\n",
       "      <td>8.040000e-276</td>\n",
       "      <td>784.0</td>\n",
       "      <td>COG1472@1|root,arCOG04634@2157|Archaea,2XPMS@2...</td>\n",
       "      <td>28889|Crenarchaeota</td>\n",
       "      <td>G</td>\n",
       "      <td>Fibronectin type III-like domain</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>ko:K05349</td>\n",
       "      <td>ko00460,ko00500,ko00940,ko01100,ko01110,map004...</td>\n",
       "      <td>-</td>\n",
       "      <td>R00026,R02558,R02887,R02985,R03527,R04949,R049...</td>\n",
       "      <td>RC00049,RC00059,RC00171,RC00262,RC00397,RC0045...</td>\n",
       "      <td>ko00000,ko00001,ko01000</td>\n",
       "      <td>-</td>\n",
       "      <td>GH3</td>\n",
       "      <td>-</td>\n",
       "      <td>Fn3-like,Glyco_hydro_3,Glyco_hydro_3_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>fig|1448933.3.peg.5</td>\n",
       "      <td>1459636.NTE_01989</td>\n",
       "      <td>2.680000e-26</td>\n",
       "      <td>104.0</td>\n",
       "      <td>COG1832@1|root,arCOG04227@2157|Archaea,41SMT@6...</td>\n",
       "      <td>651137|Thaumarchaeota</td>\n",
       "      <td>S</td>\n",
       "      <td>CoA binding domain</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>ko:K06929</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00000</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>CoA_binding_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>fig|1448933.3.peg.6</td>\n",
       "      <td>1209989.TepiRe1_0483</td>\n",
       "      <td>9.240000e-75</td>\n",
       "      <td>239.0</td>\n",
       "      <td>COG1082@1|root,COG1082@2|Bacteria,1UITG@1239|F...</td>\n",
       "      <td>186801|Clostridia</td>\n",
       "      <td>G</td>\n",
       "      <td>Xylose isomerase-like TIM barrel</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>ko:K01805</td>\n",
       "      <td>ko00040,ko00051,ko01100,map00040,map00051,map0...</td>\n",
       "      <td>-</td>\n",
       "      <td>R00878,R01432</td>\n",
       "      <td>RC00376,RC00516</td>\n",
       "      <td>ko00000,ko00001,ko01000</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>AP_endonuc_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>fig|1448933.3.peg.8</td>\n",
       "      <td>1343739.PAP_09645</td>\n",
       "      <td>2.200000e-33</td>\n",
       "      <td>121.0</td>\n",
       "      <td>COG0432@1|root,arCOG04214@2157|Archaea,2XYW0@2...</td>\n",
       "      <td>183968|Thermococci</td>\n",
       "      <td>S</td>\n",
       "      <td>Uncharacterised protein family UPF0047</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>UPF0047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>fig|1448933.3.peg.9</td>\n",
       "      <td>1353529.M899_1098</td>\n",
       "      <td>7.730000e-08</td>\n",
       "      <td>61.2</td>\n",
       "      <td>COG0530@1|root,COG0530@2|Bacteria,1MU3R@1224|P...</td>\n",
       "      <td>28221|Deltaproteobacteria</td>\n",
       "      <td>P</td>\n",
       "      <td>Sodium/calcium exchanger protein</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>ko:K07301</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00000,ko02000</td>\n",
       "      <td>2.A.19.5</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Na_Ca_ex</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    ID         seed_ortholog         evalue  score  \\\n",
       "0  fig|1448933.3.peg.1      930945.SiRe_2208  8.040000e-276  784.0   \n",
       "1  fig|1448933.3.peg.5     1459636.NTE_01989   2.680000e-26  104.0   \n",
       "2  fig|1448933.3.peg.6  1209989.TepiRe1_0483   9.240000e-75  239.0   \n",
       "3  fig|1448933.3.peg.8     1343739.PAP_09645   2.200000e-33  121.0   \n",
       "4  fig|1448933.3.peg.9     1353529.M899_1098   7.730000e-08   61.2   \n",
       "\n",
       "                                          eggNOG_OGs  \\\n",
       "0  COG1472@1|root,arCOG04634@2157|Archaea,2XPMS@2...   \n",
       "1  COG1832@1|root,arCOG04227@2157|Archaea,41SMT@6...   \n",
       "2  COG1082@1|root,COG1082@2|Bacteria,1UITG@1239|F...   \n",
       "3  COG0432@1|root,arCOG04214@2157|Archaea,2XYW0@2...   \n",
       "4  COG0530@1|root,COG0530@2|Bacteria,1MU3R@1224|P...   \n",
       "\n",
       "               max_annot_lvl COG_category  \\\n",
       "0        28889|Crenarchaeota            G   \n",
       "1      651137|Thaumarchaeota            S   \n",
       "2          186801|Clostridia            G   \n",
       "3         183968|Thermococci            S   \n",
       "4  28221|Deltaproteobacteria            P   \n",
       "\n",
       "                              Description Preferred_name GOs  ...    KEGG_ko  \\\n",
       "0        Fibronectin type III-like domain              -   -  ...  ko:K05349   \n",
       "1                      CoA binding domain              -   -  ...  ko:K06929   \n",
       "2        Xylose isomerase-like TIM barrel              -   -  ...  ko:K01805   \n",
       "3  Uncharacterised protein family UPF0047              -   -  ...          -   \n",
       "4        Sodium/calcium exchanger protein              -   -  ...  ko:K07301   \n",
       "\n",
       "                                        KEGG_Pathway KEGG_Module  \\\n",
       "0  ko00460,ko00500,ko00940,ko01100,ko01110,map004...           -   \n",
       "1                                                  -           -   \n",
       "2  ko00040,ko00051,ko01100,map00040,map00051,map0...           -   \n",
       "3                                                  -           -   \n",
       "4                                                  -           -   \n",
       "\n",
       "                                       KEGG_Reaction  \\\n",
       "0  R00026,R02558,R02887,R02985,R03527,R04949,R049...   \n",
       "1                                                  -   \n",
       "2                                      R00878,R01432   \n",
       "3                                                  -   \n",
       "4                                                  -   \n",
       "\n",
       "                                         KEGG_rclass                    BRITE  \\\n",
       "0  RC00049,RC00059,RC00171,RC00262,RC00397,RC0045...  ko00000,ko00001,ko01000   \n",
       "1                                                  -                  ko00000   \n",
       "2                                    RC00376,RC00516  ko00000,ko00001,ko01000   \n",
       "3                                                  -                        -   \n",
       "4                                                  -          ko00000,ko02000   \n",
       "\n",
       "    KEGG_TC CAZy BiGG_Reaction                                   PFAMs  \n",
       "0         -  GH3             -  Fn3-like,Glyco_hydro_3,Glyco_hydro_3_C  \n",
       "1         -    -             -                           CoA_binding_2  \n",
       "2         -    -             -                            AP_endonuc_2  \n",
       "3         -    -             -                                 UPF0047  \n",
       "4  2.A.19.5    -             -                                Na_Ca_ex  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_eggnog.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_eggnog['ID'] = [i.split('|')[1] for i in data_eggnog['ID']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>seed_ortholog</th>\n",
       "      <th>evalue</th>\n",
       "      <th>score</th>\n",
       "      <th>eggNOG_OGs</th>\n",
       "      <th>max_annot_lvl</th>\n",
       "      <th>COG_category</th>\n",
       "      <th>Description</th>\n",
       "      <th>Preferred_name</th>\n",
       "      <th>GOs</th>\n",
       "      <th>...</th>\n",
       "      <th>KEGG_ko</th>\n",
       "      <th>KEGG_Pathway</th>\n",
       "      <th>KEGG_Module</th>\n",
       "      <th>KEGG_Reaction</th>\n",
       "      <th>KEGG_rclass</th>\n",
       "      <th>BRITE</th>\n",
       "      <th>KEGG_TC</th>\n",
       "      <th>CAZy</th>\n",
       "      <th>BiGG_Reaction</th>\n",
       "      <th>PFAMs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1448933.3.peg.1</td>\n",
       "      <td>930945.SiRe_2208</td>\n",
       "      <td>8.040000e-276</td>\n",
       "      <td>784.0</td>\n",
       "      <td>COG1472@1|root,arCOG04634@2157|Archaea,2XPMS@2...</td>\n",
       "      <td>28889|Crenarchaeota</td>\n",
       "      <td>G</td>\n",
       "      <td>Fibronectin type III-like domain</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>ko:K05349</td>\n",
       "      <td>ko00460,ko00500,ko00940,ko01100,ko01110,map004...</td>\n",
       "      <td>-</td>\n",
       "      <td>R00026,R02558,R02887,R02985,R03527,R04949,R049...</td>\n",
       "      <td>RC00049,RC00059,RC00171,RC00262,RC00397,RC0045...</td>\n",
       "      <td>ko00000,ko00001,ko01000</td>\n",
       "      <td>-</td>\n",
       "      <td>GH3</td>\n",
       "      <td>-</td>\n",
       "      <td>Fn3-like,Glyco_hydro_3,Glyco_hydro_3_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1448933.3.peg.5</td>\n",
       "      <td>1459636.NTE_01989</td>\n",
       "      <td>2.680000e-26</td>\n",
       "      <td>104.0</td>\n",
       "      <td>COG1832@1|root,arCOG04227@2157|Archaea,41SMT@6...</td>\n",
       "      <td>651137|Thaumarchaeota</td>\n",
       "      <td>S</td>\n",
       "      <td>CoA binding domain</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>ko:K06929</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00000</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>CoA_binding_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1448933.3.peg.6</td>\n",
       "      <td>1209989.TepiRe1_0483</td>\n",
       "      <td>9.240000e-75</td>\n",
       "      <td>239.0</td>\n",
       "      <td>COG1082@1|root,COG1082@2|Bacteria,1UITG@1239|F...</td>\n",
       "      <td>186801|Clostridia</td>\n",
       "      <td>G</td>\n",
       "      <td>Xylose isomerase-like TIM barrel</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>ko:K01805</td>\n",
       "      <td>ko00040,ko00051,ko01100,map00040,map00051,map0...</td>\n",
       "      <td>-</td>\n",
       "      <td>R00878,R01432</td>\n",
       "      <td>RC00376,RC00516</td>\n",
       "      <td>ko00000,ko00001,ko01000</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>AP_endonuc_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1448933.3.peg.8</td>\n",
       "      <td>1343739.PAP_09645</td>\n",
       "      <td>2.200000e-33</td>\n",
       "      <td>121.0</td>\n",
       "      <td>COG0432@1|root,arCOG04214@2157|Archaea,2XYW0@2...</td>\n",
       "      <td>183968|Thermococci</td>\n",
       "      <td>S</td>\n",
       "      <td>Uncharacterised protein family UPF0047</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>UPF0047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1448933.3.peg.9</td>\n",
       "      <td>1353529.M899_1098</td>\n",
       "      <td>7.730000e-08</td>\n",
       "      <td>61.2</td>\n",
       "      <td>COG0530@1|root,COG0530@2|Bacteria,1MU3R@1224|P...</td>\n",
       "      <td>28221|Deltaproteobacteria</td>\n",
       "      <td>P</td>\n",
       "      <td>Sodium/calcium exchanger protein</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>ko:K07301</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00000,ko02000</td>\n",
       "      <td>2.A.19.5</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Na_Ca_ex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>830</th>\n",
       "      <td>1448933.3.peg.1417</td>\n",
       "      <td>561175.KB894094_gene1906</td>\n",
       "      <td>5.220000e-104</td>\n",
       "      <td>320.0</td>\n",
       "      <td>COG4948@1|root,COG4948@2|Bacteria,2GJJR@201174...</td>\n",
       "      <td>201174|Actinobacteria</td>\n",
       "      <td>M</td>\n",
       "      <td>Converts 2-succinyl-6-hydroxy-2,4-cyclohexadie...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>ko:K01706</td>\n",
       "      <td>ko00053,ko01100,map00053,map01100</td>\n",
       "      <td>-</td>\n",
       "      <td>R02752,R08056</td>\n",
       "      <td>RC00543</td>\n",
       "      <td>ko00000,ko00001,ko01000</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>MR_MLE_C,MR_MLE_N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>831</th>\n",
       "      <td>1448933.3.peg.1418</td>\n",
       "      <td>397948.Cmaq_1620</td>\n",
       "      <td>7.380000e-187</td>\n",
       "      <td>538.0</td>\n",
       "      <td>COG4697@1|root,arCOG04132@2157|Archaea,2XPPB@2...</td>\n",
       "      <td>28889|Crenarchaeota</td>\n",
       "      <td>S</td>\n",
       "      <td>protein conserved in archaea (DUF2139)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>ko:K02040</td>\n",
       "      <td>ko02010,ko02020,ko05152,map02010,map02020,map0...</td>\n",
       "      <td>M00222</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00000,ko00001,ko00002,ko02000</td>\n",
       "      <td>3.A.1.7</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>DUF2139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>832</th>\n",
       "      <td>1448933.3.peg.1421</td>\n",
       "      <td>1104324.P186_0339</td>\n",
       "      <td>6.070000e-73</td>\n",
       "      <td>229.0</td>\n",
       "      <td>COG0603@1|root,arCOG00039@2157|Archaea</td>\n",
       "      <td>2157|Archaea</td>\n",
       "      <td>H</td>\n",
       "      <td>Catalyzes the ATP-dependent conversion of 7-ca...</td>\n",
       "      <td>queC</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>ko:K06920</td>\n",
       "      <td>ko00790,ko01100,map00790,map01100</td>\n",
       "      <td>-</td>\n",
       "      <td>R09978</td>\n",
       "      <td>RC00959</td>\n",
       "      <td>ko00000,ko00001,ko01000,ko03016</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>QueC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>833</th>\n",
       "      <td>1448933.3.peg.1422</td>\n",
       "      <td>944435.AXAJ01000001_gene502</td>\n",
       "      <td>2.270000e-09</td>\n",
       "      <td>63.5</td>\n",
       "      <td>COG0546@1|root,COG0546@2|Bacteria,1R8YE@1224|P...</td>\n",
       "      <td>28216|Betaproteobacteria</td>\n",
       "      <td>S</td>\n",
       "      <td>Specifically catalyzes the dephosphorylation o...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>ko:K01091</td>\n",
       "      <td>ko00630,ko01100,ko01110,ko01130,map00630,map01...</td>\n",
       "      <td>-</td>\n",
       "      <td>R01334</td>\n",
       "      <td>RC00017</td>\n",
       "      <td>ko00000,ko00001,ko01000</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>HAD_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834</th>\n",
       "      <td>1448933.3.peg.1423</td>\n",
       "      <td>273057.SSO3022</td>\n",
       "      <td>3.290000e-234</td>\n",
       "      <td>677.0</td>\n",
       "      <td>COG1501@1|root,arCOG03663@2157|Archaea,2XRT3@2...</td>\n",
       "      <td>28889|Crenarchaeota</td>\n",
       "      <td>G</td>\n",
       "      <td>Belongs to the glycosyl hydrolase 31 family</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>ko:K01811</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00000,ko01000</td>\n",
       "      <td>-</td>\n",
       "      <td>GH31</td>\n",
       "      <td>-</td>\n",
       "      <td>DUF5110,Gal_mutarotas_2,Glyco_hydro_31</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>835 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     ID                seed_ortholog         evalue  score  \\\n",
       "0       1448933.3.peg.1             930945.SiRe_2208  8.040000e-276  784.0   \n",
       "1       1448933.3.peg.5            1459636.NTE_01989   2.680000e-26  104.0   \n",
       "2       1448933.3.peg.6         1209989.TepiRe1_0483   9.240000e-75  239.0   \n",
       "3       1448933.3.peg.8            1343739.PAP_09645   2.200000e-33  121.0   \n",
       "4       1448933.3.peg.9            1353529.M899_1098   7.730000e-08   61.2   \n",
       "..                  ...                          ...            ...    ...   \n",
       "830  1448933.3.peg.1417     561175.KB894094_gene1906  5.220000e-104  320.0   \n",
       "831  1448933.3.peg.1418             397948.Cmaq_1620  7.380000e-187  538.0   \n",
       "832  1448933.3.peg.1421            1104324.P186_0339   6.070000e-73  229.0   \n",
       "833  1448933.3.peg.1422  944435.AXAJ01000001_gene502   2.270000e-09   63.5   \n",
       "834  1448933.3.peg.1423               273057.SSO3022  3.290000e-234  677.0   \n",
       "\n",
       "                                            eggNOG_OGs  \\\n",
       "0    COG1472@1|root,arCOG04634@2157|Archaea,2XPMS@2...   \n",
       "1    COG1832@1|root,arCOG04227@2157|Archaea,41SMT@6...   \n",
       "2    COG1082@1|root,COG1082@2|Bacteria,1UITG@1239|F...   \n",
       "3    COG0432@1|root,arCOG04214@2157|Archaea,2XYW0@2...   \n",
       "4    COG0530@1|root,COG0530@2|Bacteria,1MU3R@1224|P...   \n",
       "..                                                 ...   \n",
       "830  COG4948@1|root,COG4948@2|Bacteria,2GJJR@201174...   \n",
       "831  COG4697@1|root,arCOG04132@2157|Archaea,2XPPB@2...   \n",
       "832             COG0603@1|root,arCOG00039@2157|Archaea   \n",
       "833  COG0546@1|root,COG0546@2|Bacteria,1R8YE@1224|P...   \n",
       "834  COG1501@1|root,arCOG03663@2157|Archaea,2XRT3@2...   \n",
       "\n",
       "                 max_annot_lvl COG_category  \\\n",
       "0          28889|Crenarchaeota            G   \n",
       "1        651137|Thaumarchaeota            S   \n",
       "2            186801|Clostridia            G   \n",
       "3           183968|Thermococci            S   \n",
       "4    28221|Deltaproteobacteria            P   \n",
       "..                         ...          ...   \n",
       "830      201174|Actinobacteria            M   \n",
       "831        28889|Crenarchaeota            S   \n",
       "832               2157|Archaea            H   \n",
       "833   28216|Betaproteobacteria            S   \n",
       "834        28889|Crenarchaeota            G   \n",
       "\n",
       "                                           Description Preferred_name GOs  \\\n",
       "0                     Fibronectin type III-like domain              -   -   \n",
       "1                                   CoA binding domain              -   -   \n",
       "2                     Xylose isomerase-like TIM barrel              -   -   \n",
       "3               Uncharacterised protein family UPF0047              -   -   \n",
       "4                     Sodium/calcium exchanger protein              -   -   \n",
       "..                                                 ...            ...  ..   \n",
       "830  Converts 2-succinyl-6-hydroxy-2,4-cyclohexadie...              -   -   \n",
       "831             protein conserved in archaea (DUF2139)              -   -   \n",
       "832  Catalyzes the ATP-dependent conversion of 7-ca...           queC   -   \n",
       "833  Specifically catalyzes the dephosphorylation o...              -   -   \n",
       "834        Belongs to the glycosyl hydrolase 31 family              -   -   \n",
       "\n",
       "     ...    KEGG_ko                                       KEGG_Pathway  \\\n",
       "0    ...  ko:K05349  ko00460,ko00500,ko00940,ko01100,ko01110,map004...   \n",
       "1    ...  ko:K06929                                                  -   \n",
       "2    ...  ko:K01805  ko00040,ko00051,ko01100,map00040,map00051,map0...   \n",
       "3    ...          -                                                  -   \n",
       "4    ...  ko:K07301                                                  -   \n",
       "..   ...        ...                                                ...   \n",
       "830  ...  ko:K01706                  ko00053,ko01100,map00053,map01100   \n",
       "831  ...  ko:K02040  ko02010,ko02020,ko05152,map02010,map02020,map0...   \n",
       "832  ...  ko:K06920                  ko00790,ko01100,map00790,map01100   \n",
       "833  ...  ko:K01091  ko00630,ko01100,ko01110,ko01130,map00630,map01...   \n",
       "834  ...  ko:K01811                                                  -   \n",
       "\n",
       "    KEGG_Module                                      KEGG_Reaction  \\\n",
       "0             -  R00026,R02558,R02887,R02985,R03527,R04949,R049...   \n",
       "1             -                                                  -   \n",
       "2             -                                      R00878,R01432   \n",
       "3             -                                                  -   \n",
       "4             -                                                  -   \n",
       "..          ...                                                ...   \n",
       "830           -                                      R02752,R08056   \n",
       "831      M00222                                                  -   \n",
       "832           -                                             R09978   \n",
       "833           -                                             R01334   \n",
       "834           -                                                  -   \n",
       "\n",
       "                                           KEGG_rclass  \\\n",
       "0    RC00049,RC00059,RC00171,RC00262,RC00397,RC0045...   \n",
       "1                                                    -   \n",
       "2                                      RC00376,RC00516   \n",
       "3                                                    -   \n",
       "4                                                    -   \n",
       "..                                                 ...   \n",
       "830                                            RC00543   \n",
       "831                                                  -   \n",
       "832                                            RC00959   \n",
       "833                                            RC00017   \n",
       "834                                                  -   \n",
       "\n",
       "                               BRITE   KEGG_TC  CAZy BiGG_Reaction  \\\n",
       "0            ko00000,ko00001,ko01000         -   GH3             -   \n",
       "1                            ko00000         -     -             -   \n",
       "2            ko00000,ko00001,ko01000         -     -             -   \n",
       "3                                  -         -     -             -   \n",
       "4                    ko00000,ko02000  2.A.19.5     -             -   \n",
       "..                               ...       ...   ...           ...   \n",
       "830          ko00000,ko00001,ko01000         -     -             -   \n",
       "831  ko00000,ko00001,ko00002,ko02000   3.A.1.7     -             -   \n",
       "832  ko00000,ko00001,ko01000,ko03016         -     -             -   \n",
       "833          ko00000,ko00001,ko01000         -     -             -   \n",
       "834                  ko00000,ko01000         -  GH31             -   \n",
       "\n",
       "                                      PFAMs  \n",
       "0    Fn3-like,Glyco_hydro_3,Glyco_hydro_3_C  \n",
       "1                             CoA_binding_2  \n",
       "2                              AP_endonuc_2  \n",
       "3                                   UPF0047  \n",
       "4                                  Na_Ca_ex  \n",
       "..                                      ...  \n",
       "830                       MR_MLE_C,MR_MLE_N  \n",
       "831                                 DUF2139  \n",
       "832                                    QueC  \n",
       "833                                   HAD_2  \n",
       "834  DUF5110,Gal_mutarotas_2,Glyco_hydro_31  \n",
       "\n",
       "[835 rows x 21 columns]"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_eggnog"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = {'J':'ff0000', 'A':'c2af58', 'K':'ff9900', 'L':'ffff00', 'B':'ffc600',\n",
    "    'D':'99ff00', 'Y':'493126', 'V':'ff008a', 'T':'0000ff', 'M':'9ec928',\n",
    "    'N':'006633', 'Z':'660099', 'W':'336699', 'U':'33cc99', 'O':'00ffff', \n",
    "    'C':'9900ff', 'G':'805642', 'E':'ff00ff', 'H':'727dcc', 'I':'5c5a1b',\n",
    "    'P':'0099ff', 'Q':'ffcc99', 'R':'ff9999','S':'d6aadf', 'F':'99334d',\n",
    "    'IQ':'ffcc99',\n",
    "    '-':'white',\n",
    "    'EGP':'ff00ff',\n",
    "    'CP':'9900ff',\n",
    "    'JM':'ff0000',\n",
    "    'EP':'ff00ff',\n",
    "    'NU':'006633',\n",
    "    'HP':'727dcc',\n",
    "    'GM':'805642'\n",
    "    }"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "По идеи можно так сделать так как circos оставит не занятые места белым по умлочанию. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "plus_strand_gff = gff_data[gff_data[6] == '+']\n",
    "minus_strand_gff = gff_data[gff_data[6] == '-']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>6</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4690</td>\n",
       "      <td>5208</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.peg.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>15761</td>\n",
       "      <td>15955</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.peg.16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>15989</td>\n",
       "      <td>17098</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.peg.17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>19106</td>\n",
       "      <td>19795</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.peg.20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>19792</td>\n",
       "      <td>20514</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.peg.21</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        3      4  6                 8\n",
       "4    4690   5208  +   1448933.3.peg.5\n",
       "15  15761  15955  +  1448933.3.peg.16\n",
       "16  15989  17098  +  1448933.3.peg.17\n",
       "19  19106  19795  +  1448933.3.peg.20\n",
       "20  19792  20514  +  1448933.3.peg.21"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "plus_strand_gff.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_169544/2845654062.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  plus_strand_gff.loc[idx,'color'] = f\"fill_color={d[row2['COG_category']]}\"\n"
     ]
    }
   ],
   "source": [
    "for idx,row in plus_strand_gff.iterrows():\n",
    "    for idx2, row2, in data_eggnog.iterrows():\n",
    "        if row[8] == row2['ID']:\n",
    "            plus_strand_gff.loc[idx,'color'] = f\"fill_color={d[row2['COG_category']]}\"\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_169544/559908217.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  minus_strand_gff.loc[idx,'color'] = f\"fill_color={d[row2['COG_category']]}\"\n"
     ]
    }
   ],
   "source": [
    "for idx,row in minus_strand_gff.iterrows():\n",
    "    for idx2, row2, in data_eggnog.iterrows():\n",
    "        if row[8] == row2['ID']:\n",
    "            minus_strand_gff.loc[idx,'color'] = f\"fill_color={d[row2['COG_category']]}\"\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "6700e6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_143024/3639100162.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tRNA_gff_data.loc[idx,'color'] = f\"fill_color=6700e6\"\n"
     ]
    }
   ],
   "source": [
    "for idx,row in tRNA_gff_data.iterrows():\n",
    "    tRNA_gff_data.loc[idx,'color'] = f\"fill_color=6700e6\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>6</th>\n",
       "      <th>8</th>\n",
       "      <th>color</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>244124</td>\n",
       "      <td>244196</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.1</td>\n",
       "      <td>fill_color=6700e6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>244124</td>\n",
       "      <td>244196</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.1</td>\n",
       "      <td>fill_color=6700e6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>244124</td>\n",
       "      <td>244196</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.1</td>\n",
       "      <td>fill_color=6700e6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>274841</td>\n",
       "      <td>274925</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.2</td>\n",
       "      <td>fill_color=6700e6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>274841</td>\n",
       "      <td>274925</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.2</td>\n",
       "      <td>fill_color=6700e6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>1124017</td>\n",
       "      <td>1124091</td>\n",
       "      <td>-</td>\n",
       "      <td>1448933.3.rna.43</td>\n",
       "      <td>fill_color=6700e6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>1124017</td>\n",
       "      <td>1124091</td>\n",
       "      <td>-</td>\n",
       "      <td>1448933.3.rna.43</td>\n",
       "      <td>fill_color=6700e6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>1169723</td>\n",
       "      <td>1169797</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.44</td>\n",
       "      <td>fill_color=6700e6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>1169723</td>\n",
       "      <td>1169797</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.44</td>\n",
       "      <td>fill_color=6700e6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>1169723</td>\n",
       "      <td>1169797</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.44</td>\n",
       "      <td>fill_color=6700e6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>132 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           3        4  6                 8              color\n",
       "0     244124   244196  +   1448933.3.rna.1  fill_color=6700e6\n",
       "1     244124   244196  +   1448933.3.rna.1  fill_color=6700e6\n",
       "2     244124   244196  +   1448933.3.rna.1  fill_color=6700e6\n",
       "3     274841   274925  +   1448933.3.rna.2  fill_color=6700e6\n",
       "4     274841   274925  +   1448933.3.rna.2  fill_color=6700e6\n",
       "..       ...      ... ..               ...                ...\n",
       "127  1124017  1124091  -  1448933.3.rna.43  fill_color=6700e6\n",
       "128  1124017  1124091  -  1448933.3.rna.43  fill_color=6700e6\n",
       "129  1169723  1169797  +  1448933.3.rna.44  fill_color=6700e6\n",
       "130  1169723  1169797  +  1448933.3.rna.44  fill_color=6700e6\n",
       "131  1169723  1169797  +  1448933.3.rna.44  fill_color=6700e6\n",
       "\n",
       "[132 rows x 5 columns]"
      ]
     },
     "execution_count": 140,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tRNA_gff_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>6</th>\n",
       "      <th>8</th>\n",
       "      <th>color</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4690</td>\n",
       "      <td>5208</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.peg.5</td>\n",
       "      <td>fill_color=d6aadf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>15761</td>\n",
       "      <td>15955</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.peg.16</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>15989</td>\n",
       "      <td>17098</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.peg.17</td>\n",
       "      <td>fill_color=99334d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>19106</td>\n",
       "      <td>19795</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.peg.20</td>\n",
       "      <td>fill_color=d6aadf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>19792</td>\n",
       "      <td>20514</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.peg.21</td>\n",
       "      <td>fill_color=5c5a1b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>1050860</td>\n",
       "      <td>1050958</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.41</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>1050860</td>\n",
       "      <td>1050958</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.41</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>1169723</td>\n",
       "      <td>1169797</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.44</td>\n",
       "      <td>fill_color=9ec928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>1169723</td>\n",
       "      <td>1169797</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.44</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>1169723</td>\n",
       "      <td>1169797</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.rna.44</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>789 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           3        4  6                 8              color\n",
       "4       4690     5208  +   1448933.3.peg.5  fill_color=d6aadf\n",
       "15     15761    15955  +  1448933.3.peg.16                NaN\n",
       "16     15989    17098  +  1448933.3.peg.17  fill_color=99334d\n",
       "19     19106    19795  +  1448933.3.peg.20  fill_color=d6aadf\n",
       "20     19792    20514  +  1448933.3.peg.21  fill_color=5c5a1b\n",
       "..       ...      ... ..               ...                ...\n",
       "121  1050860  1050958  +  1448933.3.rna.41                NaN\n",
       "122  1050860  1050958  +  1448933.3.rna.41                NaN\n",
       "129  1169723  1169797  +  1448933.3.rna.44  fill_color=9ec928\n",
       "130  1169723  1169797  +  1448933.3.rna.44                NaN\n",
       "131  1169723  1169797  +  1448933.3.rna.44                NaN\n",
       "\n",
       "[789 rows x 5 columns]"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "plus_strand_gff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>6</th>\n",
       "      <th>8</th>\n",
       "      <th>color</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>2227</td>\n",
       "      <td>-</td>\n",
       "      <td>1448933.3.peg.1</td>\n",
       "      <td>fill_color=805642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2259</td>\n",
       "      <td>3155</td>\n",
       "      <td>-</td>\n",
       "      <td>1448933.3.peg.2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3265</td>\n",
       "      <td>4527</td>\n",
       "      <td>-</td>\n",
       "      <td>1448933.3.peg.3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4530</td>\n",
       "      <td>4667</td>\n",
       "      <td>-</td>\n",
       "      <td>1448933.3.peg.4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5203</td>\n",
       "      <td>6099</td>\n",
       "      <td>-</td>\n",
       "      <td>1448933.3.peg.6</td>\n",
       "      <td>fill_color=805642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>1107407</td>\n",
       "      <td>1107479</td>\n",
       "      <td>-</td>\n",
       "      <td>1448933.3.rna.42</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>1107407</td>\n",
       "      <td>1107479</td>\n",
       "      <td>-</td>\n",
       "      <td>1448933.3.rna.42</td>\n",
       "      <td>fill_color=ffff00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>1124017</td>\n",
       "      <td>1124091</td>\n",
       "      <td>-</td>\n",
       "      <td>1448933.3.rna.43</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>1124017</td>\n",
       "      <td>1124091</td>\n",
       "      <td>-</td>\n",
       "      <td>1448933.3.rna.43</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>1124017</td>\n",
       "      <td>1124091</td>\n",
       "      <td>-</td>\n",
       "      <td>1448933.3.rna.43</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>766 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           3        4  6                 8              color\n",
       "0          2     2227  -   1448933.3.peg.1  fill_color=805642\n",
       "1       2259     3155  -   1448933.3.peg.2                NaN\n",
       "2       3265     4527  -   1448933.3.peg.3                NaN\n",
       "3       4530     4667  -   1448933.3.peg.4                NaN\n",
       "5       5203     6099  -   1448933.3.peg.6  fill_color=805642\n",
       "..       ...      ... ..               ...                ...\n",
       "124  1107407  1107479  -  1448933.3.rna.42                NaN\n",
       "125  1107407  1107479  -  1448933.3.rna.42  fill_color=ffff00\n",
       "126  1124017  1124091  -  1448933.3.rna.43                NaN\n",
       "127  1124017  1124091  -  1448933.3.rna.43                NaN\n",
       "128  1124017  1124091  -  1448933.3.rna.43                NaN\n",
       "\n",
       "[766 rows x 5 columns]"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "minus_strand_gff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in plus_strand_gff.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/circos_diagramms/data/plus_strand_highlights.txt', 'a') as plus_write:\n",
    "        plus_write.write(f\"genome {row[3]} {row[4]} {row['color']} \\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in minus_strand_gff.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/circos_diagramms/data/minus_strand_highlights.txt', 'a') as minus_write:\n",
    "        minus_write.write(f\"genome {row[3]} {row[4]} {row['color']} \\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in tRNA_gff_data.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/circos_diagramms/data/tRNA_highlights.txt', 'a') as trna_write:\n",
    "        trna_write.write(f\"genome {row[3]} {row[4]} {row['color']} \\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## GC content GC_skew"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "def CG_cont(lis:str) -> float:\n",
    "    lis = lis.upper()\n",
    "    GC = lis.count('G') + lis.count('C')\n",
    "    return GC/len(lis)\n",
    "\n",
    "def CG_skew(lis:str) -> float:\n",
    "    lis = lis.upper()\n",
    "    G = lis.count('G')\n",
    "    C = lis.count('C')\n",
    "    return (G - C)/(G + C)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('/home/aster/Project_Geoarchaeota_28.11/data/sequences_annotation/1448933.3.contigs.fa', 'r') as read_file:\n",
    "    genome_data = read_file.read().split('\\n')[1::]\n",
    "    genome_data = ''.join(genome_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "sliding_window = 5000\n",
    "step = 0\n",
    "step_size = 1000\n",
    "r_bords = []\n",
    "GCskew = []\n",
    "GCcont = []\n",
    "l_bords = []\n",
    "for i in range(0,len(genome_data)):\n",
    "\n",
    "    l_bord = step\n",
    "    r_bord = sliding_window + l_bord\n",
    "    step += step_size\n",
    "    window = genome_data[l_bord:r_bord]\n",
    "\n",
    "    l_bords.append(l_bord)\n",
    "    r_bords.append(r_bord-sliding_window)\n",
    "    GCskew.append(CG_skew(window))\n",
    "    GCcont.append(CG_cont(window))\n",
    "\n",
    "    if r_bord >= len(genome_data):\n",
    "        break\n",
    "\n",
    "GCskew.append((GCskew[0] + GCskew[-1])/2) #Замыкание кольца гистограммы\n",
    "GCcont.append((GCcont[0] + GCcont[-1])/2)\n",
    "l_bords.append(l_bords[-1] + sliding_window-step_size )\n",
    "r_bords.append(r_bords[-1]  + sliding_window )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i,j,k, t in list(zip(r_bords, GCskew, GCcont, l_bords)):\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/circos_diagramms/data/GC_skew', 'a') as skew_file:\n",
    "        if j > 0:\n",
    "            skew_file.write(f\"genome {t+1} {i+1} {j} fill_color=green \\n\")\n",
    "        else:\n",
    "            skew_file.write(f\"genome {t+1} {i+1} {j} fill_color=red \\n\")\n",
    "\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/circos_diagramms/data/GC_cont', 'a') as cont_file:\n",
    "            #k2 = (k - min(GCcont))/(max(GCcont)-(min(GCcont))) #minmax norm\n",
    "            if k > 0.5:\n",
    "                cont_file.write(f\"genome {t+1} {i+1} {k} fill_color=blue\\n\")\n",
    "            else:\n",
    "                cont_file.write(f\"genome {t+1} {i+1} {k} fill_color=orange\\n\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# repeats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [],
   "source": [
    "repeats_data = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/results/repeats_ident_in_genome/repeats.tsv', delimiter='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Subject</th>\n",
       "      <th>Query</th>\n",
       "      <th>identity</th>\n",
       "      <th>lenght</th>\n",
       "      <th>missmatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>evalue</th>\n",
       "      <th>coverage</th>\n",
       "      <th>bitscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>fig|1448933.3.peg.1</td>\n",
       "      <td>fig|1448933.3.peg.1203</td>\n",
       "      <td>91.070</td>\n",
       "      <td>1803</td>\n",
       "      <td>146</td>\n",
       "      <td>12</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>81</td>\n",
       "      <td>2423.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>fig|1448933.3.peg.18</td>\n",
       "      <td>fig|1448933.3.peg.1359</td>\n",
       "      <td>97.297</td>\n",
       "      <td>222</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>8.530000e-105</td>\n",
       "      <td>20</td>\n",
       "      <td>375.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>fig|1448933.3.peg.32</td>\n",
       "      <td>fig|1448933.3.peg.1254</td>\n",
       "      <td>87.956</td>\n",
       "      <td>631</td>\n",
       "      <td>65</td>\n",
       "      <td>10</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>71</td>\n",
       "      <td>734.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>fig|1448933.3.peg.34</td>\n",
       "      <td>fig|1448933.3.peg.1229</td>\n",
       "      <td>95.302</td>\n",
       "      <td>149</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>5.180000e-63</td>\n",
       "      <td>11</td>\n",
       "      <td>237.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>fig|1448933.3.peg.46</td>\n",
       "      <td>fig|1448933.3.peg.47</td>\n",
       "      <td>100.000</td>\n",
       "      <td>50</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8.820000e-20</td>\n",
       "      <td>5</td>\n",
       "      <td>93.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Subject                   Query  identity  lenght  missmatch  \\\n",
       "0   fig|1448933.3.peg.1  fig|1448933.3.peg.1203    91.070    1803        146   \n",
       "1  fig|1448933.3.peg.18  fig|1448933.3.peg.1359    97.297     222          5   \n",
       "2  fig|1448933.3.peg.32  fig|1448933.3.peg.1254    87.956     631         65   \n",
       "3  fig|1448933.3.peg.34  fig|1448933.3.peg.1229    95.302     149          7   \n",
       "4  fig|1448933.3.peg.46    fig|1448933.3.peg.47   100.000      50          0   \n",
       "\n",
       "   gapopen         evalue  coverage  bitscore  \n",
       "0       12   0.000000e+00        81    2423.0  \n",
       "1        1  8.530000e-105        20     375.0  \n",
       "2       10   0.000000e+00        71     734.0  \n",
       "3        0   5.180000e-63        11     237.0  \n",
       "4        0   8.820000e-20         5      93.5  "
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "repeats_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "repeats_data = repeats_data[repeats_data['coverage'] >= 60]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Subject</th>\n",
       "      <th>Query</th>\n",
       "      <th>identity</th>\n",
       "      <th>lenght</th>\n",
       "      <th>missmatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>evalue</th>\n",
       "      <th>coverage</th>\n",
       "      <th>bitscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1448933.3.peg.1</td>\n",
       "      <td>1448933.3.peg.1203</td>\n",
       "      <td>91.070</td>\n",
       "      <td>1803</td>\n",
       "      <td>146</td>\n",
       "      <td>12</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>81</td>\n",
       "      <td>2423.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1448933.3.peg.32</td>\n",
       "      <td>1448933.3.peg.1254</td>\n",
       "      <td>87.956</td>\n",
       "      <td>631</td>\n",
       "      <td>65</td>\n",
       "      <td>10</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>71</td>\n",
       "      <td>734.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1448933.3.peg.126</td>\n",
       "      <td>1448933.3.peg.457</td>\n",
       "      <td>84.674</td>\n",
       "      <td>1181</td>\n",
       "      <td>179</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>96</td>\n",
       "      <td>1177.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>1448933.3.peg.166</td>\n",
       "      <td>1448933.3.peg.167</td>\n",
       "      <td>88.046</td>\n",
       "      <td>527</td>\n",
       "      <td>57</td>\n",
       "      <td>6</td>\n",
       "      <td>2.480000e-178</td>\n",
       "      <td>68</td>\n",
       "      <td>619.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>1448933.3.peg.167</td>\n",
       "      <td>1448933.3.peg.166</td>\n",
       "      <td>88.046</td>\n",
       "      <td>527</td>\n",
       "      <td>57</td>\n",
       "      <td>6</td>\n",
       "      <td>2.480000e-178</td>\n",
       "      <td>68</td>\n",
       "      <td>619.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Subject               Query  identity  lenght  missmatch  \\\n",
       "0     1448933.3.peg.1  1448933.3.peg.1203    91.070    1803        146   \n",
       "2    1448933.3.peg.32  1448933.3.peg.1254    87.956     631         65   \n",
       "9   1448933.3.peg.126   1448933.3.peg.457    84.674    1181        179   \n",
       "19  1448933.3.peg.166   1448933.3.peg.167    88.046     527         57   \n",
       "20  1448933.3.peg.167   1448933.3.peg.166    88.046     527         57   \n",
       "\n",
       "    gapopen         evalue  coverage  bitscore  \n",
       "0        12   0.000000e+00        81    2423.0  \n",
       "2        10   0.000000e+00        71     734.0  \n",
       "9         2   0.000000e+00        96    1177.0  \n",
       "19        6  2.480000e-178        68     619.0  \n",
       "20        6  2.480000e-178        68     619.0  "
      ]
     },
     "execution_count": 191,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "repeats_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [],
   "source": [
    "repeats_data['Subject'] = [i.split('|')[1] for i in repeats_data['Subject']]\n",
    "repeats_data['Query'] = [i.split('|')[1] for i in repeats_data['Query']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>6</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>2227</td>\n",
       "      <td>-</td>\n",
       "      <td>1448933.3.peg.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2259</td>\n",
       "      <td>3155</td>\n",
       "      <td>-</td>\n",
       "      <td>1448933.3.peg.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3265</td>\n",
       "      <td>4527</td>\n",
       "      <td>-</td>\n",
       "      <td>1448933.3.peg.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4530</td>\n",
       "      <td>4667</td>\n",
       "      <td>-</td>\n",
       "      <td>1448933.3.peg.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4690</td>\n",
       "      <td>5208</td>\n",
       "      <td>+</td>\n",
       "      <td>1448933.3.peg.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      3     4  6                8\n",
       "0     2  2227  -  1448933.3.peg.1\n",
       "1  2259  3155  -  1448933.3.peg.2\n",
       "2  3265  4527  -  1448933.3.peg.3\n",
       "3  4530  4667  -  1448933.3.peg.4\n",
       "4  4690  5208  +  1448933.3.peg.5"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [],
   "source": [
    "test = pd.DataFrame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx,row in repeats_data.iterrows():\n",
    "    for idx2, row2 in gff_data.iterrows():\n",
    "        if row['Subject'] == row2[8]:\n",
    "            test.loc[idx,'rep1'] = row['Subject']\n",
    "            test.loc[idx,'start1'] = int(row2[3])\n",
    "            test.loc[idx,'stop1'] = int(row2[4])\n",
    "        if row['Query'] == row2[8]:\n",
    "            test.loc[idx,'rep2'] = row['Query']\n",
    "            test.loc[idx,'start2'] = int(row2[3])\n",
    "            test.loc[idx,'stop2'] = int(row2[4])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rep1</th>\n",
       "      <th>start1</th>\n",
       "      <th>stop1</th>\n",
       "      <th>rep2</th>\n",
       "      <th>start2</th>\n",
       "      <th>stop2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1448933.3.peg.1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2227.0</td>\n",
       "      <td>1448933.3.peg.1203</td>\n",
       "      <td>1187190.0</td>\n",
       "      <td>1189421.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1448933.3.peg.32</td>\n",
       "      <td>33288.0</td>\n",
       "      <td>34172.0</td>\n",
       "      <td>1448933.3.peg.1254</td>\n",
       "      <td>1251764.0</td>\n",
       "      <td>1252564.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1448933.3.peg.126</td>\n",
       "      <td>127616.0</td>\n",
       "      <td>128842.0</td>\n",
       "      <td>1448933.3.peg.457</td>\n",
       "      <td>452057.0</td>\n",
       "      <td>453271.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>1448933.3.peg.166</td>\n",
       "      <td>173051.0</td>\n",
       "      <td>173815.0</td>\n",
       "      <td>1448933.3.peg.167</td>\n",
       "      <td>173822.0</td>\n",
       "      <td>174586.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>1448933.3.peg.167</td>\n",
       "      <td>173822.0</td>\n",
       "      <td>174586.0</td>\n",
       "      <td>1448933.3.peg.166</td>\n",
       "      <td>173051.0</td>\n",
       "      <td>173815.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>1448933.3.peg.380</td>\n",
       "      <td>387578.0</td>\n",
       "      <td>388561.0</td>\n",
       "      <td>1448933.3.peg.1347</td>\n",
       "      <td>1340116.0</td>\n",
       "      <td>1341039.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>1448933.3.peg.380</td>\n",
       "      <td>387578.0</td>\n",
       "      <td>388561.0</td>\n",
       "      <td>1448933.3.peg.480</td>\n",
       "      <td>473478.0</td>\n",
       "      <td>474488.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>1448933.3.peg.457</td>\n",
       "      <td>452057.0</td>\n",
       "      <td>453271.0</td>\n",
       "      <td>1448933.3.peg.126</td>\n",
       "      <td>127616.0</td>\n",
       "      <td>128842.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>1448933.3.peg.480</td>\n",
       "      <td>473478.0</td>\n",
       "      <td>474488.0</td>\n",
       "      <td>1448933.3.peg.1347</td>\n",
       "      <td>1340116.0</td>\n",
       "      <td>1341039.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>1448933.3.peg.480</td>\n",
       "      <td>473478.0</td>\n",
       "      <td>474488.0</td>\n",
       "      <td>1448933.3.peg.380</td>\n",
       "      <td>387578.0</td>\n",
       "      <td>388561.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>1448933.3.peg.592</td>\n",
       "      <td>577320.0</td>\n",
       "      <td>577952.0</td>\n",
       "      <td>1448933.3.peg.1123</td>\n",
       "      <td>1103905.0</td>\n",
       "      <td>1104534.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>1448933.3.peg.638</td>\n",
       "      <td>625469.0</td>\n",
       "      <td>626578.0</td>\n",
       "      <td>1448933.3.peg.1193</td>\n",
       "      <td>1178019.0</td>\n",
       "      <td>1178966.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>1448933.3.peg.846</td>\n",
       "      <td>821351.0</td>\n",
       "      <td>821605.0</td>\n",
       "      <td>1448933.3.peg.126</td>\n",
       "      <td>127616.0</td>\n",
       "      <td>128842.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>1448933.3.peg.846</td>\n",
       "      <td>821351.0</td>\n",
       "      <td>821605.0</td>\n",
       "      <td>1448933.3.peg.457</td>\n",
       "      <td>452057.0</td>\n",
       "      <td>453271.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>1448933.3.peg.1040</td>\n",
       "      <td>1010280.0</td>\n",
       "      <td>1012262.0</td>\n",
       "      <td>1448933.3.peg.1053</td>\n",
       "      <td>1026923.0</td>\n",
       "      <td>1028950.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>1448933.3.peg.1040</td>\n",
       "      <td>1010280.0</td>\n",
       "      <td>1012262.0</td>\n",
       "      <td>1448933.3.peg.1410</td>\n",
       "      <td>1411587.0</td>\n",
       "      <td>1413539.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>1448933.3.peg.1053</td>\n",
       "      <td>1026923.0</td>\n",
       "      <td>1028950.0</td>\n",
       "      <td>1448933.3.peg.1040</td>\n",
       "      <td>1010280.0</td>\n",
       "      <td>1012262.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>1448933.3.peg.1053</td>\n",
       "      <td>1026923.0</td>\n",
       "      <td>1028950.0</td>\n",
       "      <td>1448933.3.peg.1410</td>\n",
       "      <td>1411587.0</td>\n",
       "      <td>1413539.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>1448933.3.peg.1123</td>\n",
       "      <td>1103905.0</td>\n",
       "      <td>1104534.0</td>\n",
       "      <td>1448933.3.peg.592</td>\n",
       "      <td>577320.0</td>\n",
       "      <td>577952.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>1448933.3.peg.1193</td>\n",
       "      <td>1178019.0</td>\n",
       "      <td>1178966.0</td>\n",
       "      <td>1448933.3.peg.638</td>\n",
       "      <td>625469.0</td>\n",
       "      <td>626578.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>1448933.3.peg.1203</td>\n",
       "      <td>1187190.0</td>\n",
       "      <td>1189421.0</td>\n",
       "      <td>1448933.3.peg.1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2227.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>1448933.3.peg.1204</td>\n",
       "      <td>1189526.0</td>\n",
       "      <td>1191790.0</td>\n",
       "      <td>1448933.3.peg.1338</td>\n",
       "      <td>1329427.0</td>\n",
       "      <td>1331733.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>1448933.3.peg.1248</td>\n",
       "      <td>1242229.0</td>\n",
       "      <td>1244967.0</td>\n",
       "      <td>1448933.3.peg.1352</td>\n",
       "      <td>1348398.0</td>\n",
       "      <td>1350800.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>1448933.3.peg.1248</td>\n",
       "      <td>1242229.0</td>\n",
       "      <td>1244967.0</td>\n",
       "      <td>1448933.3.peg.1352</td>\n",
       "      <td>1348398.0</td>\n",
       "      <td>1350800.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>1448933.3.peg.1254</td>\n",
       "      <td>1251764.0</td>\n",
       "      <td>1252564.0</td>\n",
       "      <td>1448933.3.peg.32</td>\n",
       "      <td>33288.0</td>\n",
       "      <td>34172.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>1448933.3.peg.1338</td>\n",
       "      <td>1329427.0</td>\n",
       "      <td>1331733.0</td>\n",
       "      <td>1448933.3.peg.1204</td>\n",
       "      <td>1189526.0</td>\n",
       "      <td>1191790.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>1448933.3.peg.1341</td>\n",
       "      <td>1333582.0</td>\n",
       "      <td>1334934.0</td>\n",
       "      <td>1448933.3.peg.896</td>\n",
       "      <td>870341.0</td>\n",
       "      <td>872830.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>1448933.3.peg.1347</td>\n",
       "      <td>1340116.0</td>\n",
       "      <td>1341039.0</td>\n",
       "      <td>1448933.3.peg.480</td>\n",
       "      <td>473478.0</td>\n",
       "      <td>474488.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>1448933.3.peg.1347</td>\n",
       "      <td>1340116.0</td>\n",
       "      <td>1341039.0</td>\n",
       "      <td>1448933.3.peg.380</td>\n",
       "      <td>387578.0</td>\n",
       "      <td>388561.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>1448933.3.peg.1352</td>\n",
       "      <td>1348398.0</td>\n",
       "      <td>1350800.0</td>\n",
       "      <td>1448933.3.peg.1248</td>\n",
       "      <td>1242229.0</td>\n",
       "      <td>1244967.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>1448933.3.peg.1352</td>\n",
       "      <td>1348398.0</td>\n",
       "      <td>1350800.0</td>\n",
       "      <td>1448933.3.peg.1248</td>\n",
       "      <td>1242229.0</td>\n",
       "      <td>1244967.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>1448933.3.peg.1410</td>\n",
       "      <td>1411587.0</td>\n",
       "      <td>1413539.0</td>\n",
       "      <td>1448933.3.peg.1053</td>\n",
       "      <td>1026923.0</td>\n",
       "      <td>1028950.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>1448933.3.peg.1410</td>\n",
       "      <td>1411587.0</td>\n",
       "      <td>1413539.0</td>\n",
       "      <td>1448933.3.peg.1040</td>\n",
       "      <td>1010280.0</td>\n",
       "      <td>1012262.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   rep1     start1      stop1                rep2     start2  \\\n",
       "0       1448933.3.peg.1        2.0     2227.0  1448933.3.peg.1203  1187190.0   \n",
       "2      1448933.3.peg.32    33288.0    34172.0  1448933.3.peg.1254  1251764.0   \n",
       "9     1448933.3.peg.126   127616.0   128842.0   1448933.3.peg.457   452057.0   \n",
       "19    1448933.3.peg.166   173051.0   173815.0   1448933.3.peg.167   173822.0   \n",
       "20    1448933.3.peg.167   173822.0   174586.0   1448933.3.peg.166   173051.0   \n",
       "33    1448933.3.peg.380   387578.0   388561.0  1448933.3.peg.1347  1340116.0   \n",
       "34    1448933.3.peg.380   387578.0   388561.0   1448933.3.peg.480   473478.0   \n",
       "35    1448933.3.peg.457   452057.0   453271.0   1448933.3.peg.126   127616.0   \n",
       "38    1448933.3.peg.480   473478.0   474488.0  1448933.3.peg.1347  1340116.0   \n",
       "39    1448933.3.peg.480   473478.0   474488.0   1448933.3.peg.380   387578.0   \n",
       "44    1448933.3.peg.592   577320.0   577952.0  1448933.3.peg.1123  1103905.0   \n",
       "54    1448933.3.peg.638   625469.0   626578.0  1448933.3.peg.1193  1178019.0   \n",
       "66    1448933.3.peg.846   821351.0   821605.0   1448933.3.peg.126   127616.0   \n",
       "67    1448933.3.peg.846   821351.0   821605.0   1448933.3.peg.457   452057.0   \n",
       "77   1448933.3.peg.1040  1010280.0  1012262.0  1448933.3.peg.1053  1026923.0   \n",
       "78   1448933.3.peg.1040  1010280.0  1012262.0  1448933.3.peg.1410  1411587.0   \n",
       "81   1448933.3.peg.1053  1026923.0  1028950.0  1448933.3.peg.1040  1010280.0   \n",
       "82   1448933.3.peg.1053  1026923.0  1028950.0  1448933.3.peg.1410  1411587.0   \n",
       "89   1448933.3.peg.1123  1103905.0  1104534.0   1448933.3.peg.592   577320.0   \n",
       "90   1448933.3.peg.1193  1178019.0  1178966.0   1448933.3.peg.638   625469.0   \n",
       "91   1448933.3.peg.1203  1187190.0  1189421.0     1448933.3.peg.1        2.0   \n",
       "92   1448933.3.peg.1204  1189526.0  1191790.0  1448933.3.peg.1338  1329427.0   \n",
       "102  1448933.3.peg.1248  1242229.0  1244967.0  1448933.3.peg.1352  1348398.0   \n",
       "103  1448933.3.peg.1248  1242229.0  1244967.0  1448933.3.peg.1352  1348398.0   \n",
       "107  1448933.3.peg.1254  1251764.0  1252564.0    1448933.3.peg.32    33288.0   \n",
       "113  1448933.3.peg.1338  1329427.0  1331733.0  1448933.3.peg.1204  1189526.0   \n",
       "116  1448933.3.peg.1341  1333582.0  1334934.0   1448933.3.peg.896   870341.0   \n",
       "117  1448933.3.peg.1347  1340116.0  1341039.0   1448933.3.peg.480   473478.0   \n",
       "118  1448933.3.peg.1347  1340116.0  1341039.0   1448933.3.peg.380   387578.0   \n",
       "119  1448933.3.peg.1352  1348398.0  1350800.0  1448933.3.peg.1248  1242229.0   \n",
       "120  1448933.3.peg.1352  1348398.0  1350800.0  1448933.3.peg.1248  1242229.0   \n",
       "133  1448933.3.peg.1410  1411587.0  1413539.0  1448933.3.peg.1053  1026923.0   \n",
       "134  1448933.3.peg.1410  1411587.0  1413539.0  1448933.3.peg.1040  1010280.0   \n",
       "\n",
       "         stop2  \n",
       "0    1189421.0  \n",
       "2    1252564.0  \n",
       "9     453271.0  \n",
       "19    174586.0  \n",
       "20    173815.0  \n",
       "33   1341039.0  \n",
       "34    474488.0  \n",
       "35    128842.0  \n",
       "38   1341039.0  \n",
       "39    388561.0  \n",
       "44   1104534.0  \n",
       "54   1178966.0  \n",
       "66    128842.0  \n",
       "67    453271.0  \n",
       "77   1028950.0  \n",
       "78   1413539.0  \n",
       "81   1012262.0  \n",
       "82   1413539.0  \n",
       "89    577952.0  \n",
       "90    626578.0  \n",
       "91      2227.0  \n",
       "92   1331733.0  \n",
       "102  1350800.0  \n",
       "103  1350800.0  \n",
       "107    34172.0  \n",
       "113  1191790.0  \n",
       "116   872830.0  \n",
       "117   474488.0  \n",
       "118   388561.0  \n",
       "119  1244967.0  \n",
       "120  1244967.0  \n",
       "133  1028950.0  \n",
       "134  1012262.0  "
      ]
     },
     "execution_count": 192,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in test.iterrows():\n",
    "    with open('/home/aster/Project_Geoarchaeota_28.11/results/circos_diagramms/data/repeat_links.txt', 'a') as repeat_file:\n",
    "        repeat_file.write(f\"rep{idx} genome {row['start1']}{row['stop1']}\\nrep{idx} genome {row['start2']}{row['stop2']} \\n\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Тоже самое но для А10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Тут все сложнее так как геном поделен на контиги, благо на IMG есть gff поделенны на контиги"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Создание файла кариотипа из файла контигов\n",
    "\n",
    "всего 86 контигов"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "contigs_len = []\n",
    "\n",
    "for sequence in SeqIO.parse(open('/home/aster/Trmosos_files/results/circos_diagramm/data/2893809466.fna','r'), 'fasta'):\n",
    "    with open('/home/aster/Trmosos_files/results/circos_diagramm/data/A10_karyotype.txt', 'a') as write_file:\n",
    "        #chr - ID LABEL START END COLOR\n",
    "        write_file.write(f\"chr - {sequence.name} A10_{sequence.name.split('_')[1]} 0 {len(sequence.seq)} green\\n\")\n",
    "        #print(f'{sequence.name} -- {len(sequence.seq)}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Преобразование gff для circos"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_a10 = pd.read_csv('/home/aster/Trmosos_files/results/circos_diagramm/data/2893809466.gff', delimiter='\\t', header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_a10 = gff_a10.iloc[:,[0,3,4,6,8]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_43304/3745091554.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gff_a10[8] = [i.split(';')[0].split('=')[1] for i in gff_a10[8]]\n"
     ]
    }
   ],
   "source": [
    "gff_a10[8] = [i.split(';')[0].split('=')[1] for i in gff_a10[8]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>6</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ga0454715_01</td>\n",
       "      <td>26</td>\n",
       "      <td>664</td>\n",
       "      <td>-</td>\n",
       "      <td>2893809466</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Ga0454715_01</td>\n",
       "      <td>706</td>\n",
       "      <td>1668</td>\n",
       "      <td>+</td>\n",
       "      <td>2893809467</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              0    3     4  6           8\n",
       "0  Ga0454715_01   26   664  -  2893809466\n",
       "1  Ga0454715_01  706  1668  +  2893809467"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff_a10.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = {'J':'ff0000', 'A':'c2af58', 'K':'ff9900', 'L':'ffff00', 'B':'ffc600',\n",
    "    'D':'99ff00', 'Y':'493126', 'V':'ff008a', 'T':'0000ff', 'M':'9ec928',\n",
    "    'N':'006633', 'Z':'660099', 'W':'336699', 'U':'33cc99', 'O':'00ffff', \n",
    "    'C':'9900ff', 'G':'805642', 'E':'99334d', 'H':'727dcc', 'I':'5c5a1b',\n",
    "    'P':'0099ff', 'Q':'ffcc99', 'R':'ff9999','S':'d6aadf', 'F':'99334d',\n",
    "    'IQ':'ffcc99',\n",
    "    '-':'white',\n",
    "    'EGP':'b8b100',\n",
    "    'CP':'9900ff',\n",
    "    'JM':'ff0000',\n",
    "    'EP':'99334d',\n",
    "    'NU':'006633',\n",
    "    'HP':'727dcc',\n",
    "    'GM':'805642',\n",
    "    ################\n",
    "    'LT':'ffff00',\n",
    "    'KL':'ff9900',\n",
    "    'BQ':'ffc600',\n",
    "    'EH':'99334d',\n",
    "    'FG':'99334d',\n",
    "    'OU':'00ffff',\n",
    "    'CE':'9900ff'\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "eggnog_a10_data = pd.read_csv('/home/aster/Trmosos_files/results/circos_diagramm/data/MM_b8br3mu9.emapper.annotations.tsv', delimiter='\\t', header=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>query</th>\n",
       "      <th>seed_ortholog</th>\n",
       "      <th>evalue</th>\n",
       "      <th>score</th>\n",
       "      <th>eggNOG_OGs</th>\n",
       "      <th>max_annot_lvl</th>\n",
       "      <th>COG_category</th>\n",
       "      <th>Description</th>\n",
       "      <th>Preferred_name</th>\n",
       "      <th>GOs</th>\n",
       "      <th>...</th>\n",
       "      <th>KEGG_ko</th>\n",
       "      <th>KEGG_Pathway</th>\n",
       "      <th>KEGG_Module</th>\n",
       "      <th>KEGG_Reaction</th>\n",
       "      <th>KEGG_rclass</th>\n",
       "      <th>BRITE</th>\n",
       "      <th>KEGG_TC</th>\n",
       "      <th>CAZy</th>\n",
       "      <th>BiGG_Reaction</th>\n",
       "      <th>PFAMs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2893809466</td>\n",
       "      <td>330779.Saci_1780</td>\n",
       "      <td>3.440000e-64</td>\n",
       "      <td>204.0</td>\n",
       "      <td>COG2039@1|root,arCOG05850@2157|Archaea,2XQMQ@2...</td>\n",
       "      <td>28889|Crenarchaeota</td>\n",
       "      <td>O</td>\n",
       "      <td>Removes 5-oxoproline from various penultimate ...</td>\n",
       "      <td>pcp</td>\n",
       "      <td>GO:0006508,GO:0006807,GO:0008150,GO:0008152,GO...</td>\n",
       "      <td>...</td>\n",
       "      <td>ko:K01304</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00000,ko01000,ko01002</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Peptidase_C15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2893809467</td>\n",
       "      <td>639282.DEFDS_1234</td>\n",
       "      <td>5.990000e-118</td>\n",
       "      <td>350.0</td>\n",
       "      <td>COG1775@1|root,COG1775@2|Bacteria,2GEXY@200930...</td>\n",
       "      <td>200930|Deferribacteres</td>\n",
       "      <td>E</td>\n",
       "      <td>2-hydroxyglutaryl-CoA dehydratase, D-component</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>HGD-D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2893809468</td>\n",
       "      <td>1220534.B655_0450</td>\n",
       "      <td>8.010000e-49</td>\n",
       "      <td>164.0</td>\n",
       "      <td>COG1014@1|root,arCOG01602@2157|Archaea,2XXK2@2...</td>\n",
       "      <td>183925|Methanobacteria</td>\n",
       "      <td>C</td>\n",
       "      <td>indolepyruvate ferredoxin oxidoreductase, beta...</td>\n",
       "      <td>iorB</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>ko:K00180</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>br01601,ko00000,ko01000</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>POR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2893809469</td>\n",
       "      <td>673860.AciM339_0762</td>\n",
       "      <td>8.790000e-262</td>\n",
       "      <td>739.0</td>\n",
       "      <td>COG4231@1|root,arCOG01609@2157|Archaea,2XUNR@2...</td>\n",
       "      <td>28890|Euryarchaeota</td>\n",
       "      <td>C</td>\n",
       "      <td>Catalyzes the ferredoxin-dependent oxidative d...</td>\n",
       "      <td>iorA</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>ko:K00179</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>br01601,ko00000,ko01000</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Fer4,Fer4_7,POR_N,TPP_enzyme_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2893809470</td>\n",
       "      <td>673860.AciM339_0763</td>\n",
       "      <td>1.310000e-158</td>\n",
       "      <td>451.0</td>\n",
       "      <td>COG1013@1|root,arCOG01601@2157|Archaea,2XT3E@2...</td>\n",
       "      <td>28890|Euryarchaeota</td>\n",
       "      <td>C</td>\n",
       "      <td>Thiamine pyrophosphate enzyme, C-terminal TPP ...</td>\n",
       "      <td>vorB</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>ko:K00170,ko:K00187</td>\n",
       "      <td>ko00010,ko00020,ko00280,ko00620,ko00633,ko0064...</td>\n",
       "      <td>M00173,M00307,M00374,M00620</td>\n",
       "      <td>R01196,R01199,R07160,R08034,R08566,R08567</td>\n",
       "      <td>RC00004,RC00250,RC02742,RC02833,RC02856</td>\n",
       "      <td>br01601,ko00000,ko00001,ko00002,ko01000</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>TPP_enzyme_C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        query        seed_ortholog         evalue  score  \\\n",
       "0  2893809466     330779.Saci_1780   3.440000e-64  204.0   \n",
       "1  2893809467    639282.DEFDS_1234  5.990000e-118  350.0   \n",
       "2  2893809468    1220534.B655_0450   8.010000e-49  164.0   \n",
       "3  2893809469  673860.AciM339_0762  8.790000e-262  739.0   \n",
       "4  2893809470  673860.AciM339_0763  1.310000e-158  451.0   \n",
       "\n",
       "                                          eggNOG_OGs           max_annot_lvl  \\\n",
       "0  COG2039@1|root,arCOG05850@2157|Archaea,2XQMQ@2...     28889|Crenarchaeota   \n",
       "1  COG1775@1|root,COG1775@2|Bacteria,2GEXY@200930...  200930|Deferribacteres   \n",
       "2  COG1014@1|root,arCOG01602@2157|Archaea,2XXK2@2...  183925|Methanobacteria   \n",
       "3  COG4231@1|root,arCOG01609@2157|Archaea,2XUNR@2...     28890|Euryarchaeota   \n",
       "4  COG1013@1|root,arCOG01601@2157|Archaea,2XT3E@2...     28890|Euryarchaeota   \n",
       "\n",
       "  COG_category                                        Description  \\\n",
       "0            O  Removes 5-oxoproline from various penultimate ...   \n",
       "1            E     2-hydroxyglutaryl-CoA dehydratase, D-component   \n",
       "2            C  indolepyruvate ferredoxin oxidoreductase, beta...   \n",
       "3            C  Catalyzes the ferredoxin-dependent oxidative d...   \n",
       "4            C  Thiamine pyrophosphate enzyme, C-terminal TPP ...   \n",
       "\n",
       "  Preferred_name                                                GOs  ...  \\\n",
       "0            pcp  GO:0006508,GO:0006807,GO:0008150,GO:0008152,GO...  ...   \n",
       "1              -                                                  -  ...   \n",
       "2           iorB                                                  -  ...   \n",
       "3           iorA                                                  -  ...   \n",
       "4           vorB                                                  -  ...   \n",
       "\n",
       "               KEGG_ko                                       KEGG_Pathway  \\\n",
       "0            ko:K01304                                                  -   \n",
       "1                    -                                                  -   \n",
       "2            ko:K00180                                                  -   \n",
       "3            ko:K00179                                                  -   \n",
       "4  ko:K00170,ko:K00187  ko00010,ko00020,ko00280,ko00620,ko00633,ko0064...   \n",
       "\n",
       "                   KEGG_Module                              KEGG_Reaction  \\\n",
       "0                            -                                          -   \n",
       "1                            -                                          -   \n",
       "2                            -                                          -   \n",
       "3                            -                                          -   \n",
       "4  M00173,M00307,M00374,M00620  R01196,R01199,R07160,R08034,R08566,R08567   \n",
       "\n",
       "                               KEGG_rclass  \\\n",
       "0                                        -   \n",
       "1                                        -   \n",
       "2                                        -   \n",
       "3                                        -   \n",
       "4  RC00004,RC00250,RC02742,RC02833,RC02856   \n",
       "\n",
       "                                     BRITE KEGG_TC CAZy BiGG_Reaction  \\\n",
       "0                  ko00000,ko01000,ko01002       -    -             -   \n",
       "1                                        -       -    -             -   \n",
       "2                  br01601,ko00000,ko01000       -    -             -   \n",
       "3                  br01601,ko00000,ko01000       -    -             -   \n",
       "4  br01601,ko00000,ko00001,ko00002,ko01000       -    -             -   \n",
       "\n",
       "                            PFAMs  \n",
       "0                   Peptidase_C15  \n",
       "1                           HGD-D  \n",
       "2                             POR  \n",
       "3  Fer4,Fer4_7,POR_N,TPP_enzyme_C  \n",
       "4                    TPP_enzyme_C  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eggnog_a10_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in eggnog_a10_data.iterrows():\n",
    "    for idx2,row2 in gff_a10.iterrows():\n",
    "        if row['query'] == row2[8]:\n",
    "            gff_a10.loc[idx,'color'] = f\"fill_color={d[row['COG_category']]}\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "plus_strand_a10 = gff_a10[gff_a10[6] == '+']\n",
    "minus_strand_a10 = gff_a10[gff_a10[6] == '-']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>6</th>\n",
       "      <th>8</th>\n",
       "      <th>color</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Ga0454715_01</td>\n",
       "      <td>706</td>\n",
       "      <td>1668</td>\n",
       "      <td>+</td>\n",
       "      <td>2893809467</td>\n",
       "      <td>fill_color=99334d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Ga0454715_01</td>\n",
       "      <td>7199</td>\n",
       "      <td>7480</td>\n",
       "      <td>+</td>\n",
       "      <td>2893809474</td>\n",
       "      <td>fill_color=ffcc99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Ga0454715_01</td>\n",
       "      <td>7486</td>\n",
       "      <td>8079</td>\n",
       "      <td>+</td>\n",
       "      <td>2893809475</td>\n",
       "      <td>fill_color=ff0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Ga0454715_01</td>\n",
       "      <td>8072</td>\n",
       "      <td>8647</td>\n",
       "      <td>+</td>\n",
       "      <td>2893809476</td>\n",
       "      <td>fill_color=ff9900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Ga0454715_01</td>\n",
       "      <td>8662</td>\n",
       "      <td>8820</td>\n",
       "      <td>+</td>\n",
       "      <td>2893809477</td>\n",
       "      <td>fill_color=0000ff</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               0     3     4  6           8              color\n",
       "1   Ga0454715_01   706  1668  +  2893809467  fill_color=99334d\n",
       "8   Ga0454715_01  7199  7480  +  2893809474  fill_color=ffcc99\n",
       "9   Ga0454715_01  7486  8079  +  2893809475  fill_color=ff0000\n",
       "10  Ga0454715_01  8072  8647  +  2893809476  fill_color=ff9900\n",
       "11  Ga0454715_01  8662  8820  +  2893809477  fill_color=0000ff"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "plus_strand_a10.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Создание highlite файлов\n",
    "\n",
    "ID START STOP COLOR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in plus_strand_a10.iterrows():\n",
    "    with open('/home/aster/Trmosos_files/results/circos_diagramm/data/plus_strand_highlights.txt', 'a') as plus_file:\n",
    "        plus_file.write(f\"{row[0]} {row[3]} {row[4]} {row['color']}\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, row in minus_strand_a10.iterrows():\n",
    "    with open('/home/aster/Trmosos_files/results/circos_diagramm/data/minus_strand_highlights.txt', 'a') as minus_file:\n",
    "        minus_file.write(f\"{row[0]} {row[3]} {row[4]} {row['color']}\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CG_skew"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "def CG_cont(lis:str) -> float:\n",
    "    lis = lis.upper()\n",
    "    GC = lis.count('G') + lis.count('C')\n",
    "    return GC/len(lis)\n",
    "\n",
    "def CG_skew(lis:str) -> float:\n",
    "    lis = lis.upper()\n",
    "    G = lis.count('G')\n",
    "    C = lis.count('C')\n",
    "    return (G - C)/(G + C)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "def slider(seq,sliding_window, step_size):\n",
    "    #sliding_window = 1000\n",
    "    step = 0\n",
    "    #step_size = 1000\n",
    "    r_bords = []\n",
    "    GCskew = []\n",
    "    #GCcont = []\n",
    "    l_bords = []\n",
    "    for i in range(0,len(seq)):\n",
    "\n",
    "        l_bord = step\n",
    "        r_bord = sliding_window + l_bord\n",
    "        step += step_size\n",
    "        window = seq[l_bord:r_bord]\n",
    "\n",
    "        l_bords.append(l_bord+1)\n",
    "        r_bords.append(r_bord-sliding_window+1)\n",
    "        GCskew.append(CG_skew(window))\n",
    "        #GCcont.append(CG_cont(window))\n",
    "\n",
    "        if r_bord >= len(seq):\n",
    "            break\n",
    "\n",
    "    GCskew.append((GCskew[0] + GCskew[-1])/2) #Замыкание кольца гистограммы\n",
    "    #GCcont.append((GCcont[0] + GCcont[-1])/2)\n",
    "    l_bords.append(l_bords[-1] + sliding_window-step_size )\n",
    "    r_bords.append(r_bords[-1]  + sliding_window )\n",
    "    return list(zip(l_bords,r_bords,GCskew))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "for sequence in SeqIO.parse(open('/home/aster/Trmosos_files/results/circos_diagramm/data/2893809466.fna','r'), 'fasta'):\n",
    "    with open('/home/aster/Trmosos_files/results/circos_diagramm/data/A10_GC_skew.txt','a') as skew_file:\n",
    "\n",
    "        if len(sequence.seq) >= 5000:\n",
    "            lst = slider(str(sequence.seq), sliding_window=5000,step_size=1000)\n",
    "            for i,j,k in lst:\n",
    "                if k > 0:\n",
    "                    skew_file.write(f\"{sequence.name} {i} {j} {k} fill_color=green\\n\")\n",
    "                else:\n",
    "                    skew_file.write(f\"{sequence.name} {i} {j} {k} fill_color=red\\n\")\n",
    "\n",
    "        else:\n",
    "            lst = slider(str(sequence.seq), sliding_window=1000,step_size=1000)\n",
    "            for i,j,k in lst:\n",
    "                if k > 0:\n",
    "                    skew_file.write(f\"{sequence.name} {i} {j} {k} fill_color=green\\n\")\n",
    "                else:\n",
    "                    skew_file.write(f\"{sequence.name} {i} {j} {k} fill_color=red\\n\")\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "bioinf",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "5c7c0e940973d1f7f557ece26bca6704213ece01202fcdffb2249dc40fb254cf"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
