{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "import pandas as pd\n",
    "import os\n",
    "import numpy as np\n",
    "from tqdm import tqdm\n",
    "import matplotlib.pyplot as plt\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Making tablets from fasta\n",
    "\n",
    "Make table from thous fasta files obtained from GeneBank. \n",
    "\n",
    "reference my older script that make table (Maybe I should rewrite it in future.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read fasts file:\n",
    "\n",
    "path = '/home/aster/Project_Geoarchaeota_28.11/pgap_annot/other_arch/genomes/folder_with_proteomes/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "tables_lst = []\n",
    "for file in os.listdir(path):\n",
    "    fpath = os.path.join(path, file)\n",
    "\n",
    "    accesion = file.split('.')[0]\n",
    "    pre_table = pd.DataFrame()\n",
    "\n",
    "    for idx, seq in enumerate(SeqIO.parse(fpath, 'fasta')):\n",
    "        \n",
    "        \n",
    "        protein_id = seq.description.split(' ')[0]\n",
    "\n",
    "        protein_name = seq.description.split('[')[0].split(',')[0].split(' ')\n",
    "        protein_name = ' '.join([i for i in protein_name if '.1' not in i and '_' not in i]) \n",
    "\n",
    "        locus_tag = seq.description.split('[')[0].split(',')[0].split(' ')\n",
    "        locus_tag = ''.join([i.lstrip('[').rstrip(']') for i in locus_tag if '_' in i ])\n",
    "\n",
    "        if len(locus_tag) == 0:\n",
    "            locus_tag = np.NaN\n",
    "\n",
    "        org_name = set()\n",
    "        org_name.add(str(seq.description.split('[')[1].rstrip(']')))\n",
    "        org_name = list(org_name)\n",
    "\n",
    "        sequence = str(seq.seq)\n",
    "        prot_len = len(sequence)\n",
    "\n",
    "        pre_table.loc[idx, 'Gene id'] = protein_id\n",
    "        pre_table.loc[idx, 'Name'] = protein_name\n",
    "        pre_table.loc[idx, 'Locus Tag'] = locus_tag\n",
    "        pre_table['Organism Name'] = org_name[0]\n",
    "        pre_table.loc[idx, 'Protein lenght'] = prot_len\n",
    "        pre_table.loc[idx, 'AAs'] = sequence\n",
    "    pre_table['Accession'] = accesion\n",
    "\n",
    "    tables_lst.append(pre_table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene id</th>\n",
       "      <th>Name</th>\n",
       "      <th>Locus Tag</th>\n",
       "      <th>Organism Name</th>\n",
       "      <th>Protein lenght</th>\n",
       "      <th>AAs</th>\n",
       "      <th>Accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>gnl|extdb|tmp_000288</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>gnl|extdb|tmp_000288tmp_000288</td>\n",
       "      <td>archaeon UBA168</td>\n",
       "      <td>150.0</td>\n",
       "      <td>MRKKVVIAGAIALVIGVVLFLVSPYMFNATLSLDSLLPFKQEQSLN...</td>\n",
       "      <td>GCA_002507085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>gnl|extdb|tmp_000289</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>gnl|extdb|tmp_000289tmp_000289</td>\n",
       "      <td>archaeon UBA168</td>\n",
       "      <td>261.0</td>\n",
       "      <td>MITFMAVSPHGDELLSPSDAESVKLNEAMNAVVRERQSVRVDTWVV...</td>\n",
       "      <td>GCA_002507085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>gnl|extdb|tmp_000290</td>\n",
       "      <td>DUF4910 domain-containing protein</td>\n",
       "      <td>gnl|extdb|tmp_000290</td>\n",
       "      <td>archaeon UBA168</td>\n",
       "      <td>482.0</td>\n",
       "      <td>MQAFSPYRAFDHVYNISKFHRIQGSQGLLDAANYIYDHLNVREKEM...</td>\n",
       "      <td>GCA_002507085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>gnl|extdb|tmp_000291</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>gnl|extdb|tmp_000291tmp_000291</td>\n",
       "      <td>archaeon UBA168</td>\n",
       "      <td>51.0</td>\n",
       "      <td>MIKKGVIDSACVFVANQSGAAKRLTEIGVQLHPIGERLSYYVNQGY...</td>\n",
       "      <td>GCA_002507085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>gnl|extdb|tmp_000292</td>\n",
       "      <td>FAD/NAD(P)-binding oxidoreductase</td>\n",
       "      <td>gnl|extdb|tmp_000292</td>\n",
       "      <td>archaeon UBA168</td>\n",
       "      <td>383.0</td>\n",
       "      <td>MKTVVIAGGGVGGVVAAKRLSERLKGKANLVLISDNEKYVFPPYFT...</td>\n",
       "      <td>GCA_002507085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>918</th>\n",
       "      <td>gnl|extdb|tmp_000322</td>\n",
       "      <td>DUF655 domain-containing protein</td>\n",
       "      <td>gnl|extdb|tmp_000322</td>\n",
       "      <td>archaeon UBA168</td>\n",
       "      <td>200.0</td>\n",
       "      <td>MSERDRGRRGHIRINEEYAIVLDYLPYGNPYSNRQSPTPIIQAMGE...</td>\n",
       "      <td>GCA_002507085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>919</th>\n",
       "      <td>gnl|extdb|tmp_000323</td>\n",
       "      <td>NAD(+)/NADH kinase</td>\n",
       "      <td>gnl|extdb|tmp_000323</td>\n",
       "      <td>archaeon UBA168</td>\n",
       "      <td>269.0</td>\n",
       "      <td>MENYARWSSGVEVLYLSHRPGSRAAQEAENKVASQAESMGIEVVRL...</td>\n",
       "      <td>GCA_002507085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>920</th>\n",
       "      <td>gnl|extdb|tmp_000775</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>gnl|extdb|tmp_000775tmp_000775</td>\n",
       "      <td>archaeon UBA168</td>\n",
       "      <td>244.0</td>\n",
       "      <td>FRRFLRLTGLSAEEGEPVYKAYLDAGGDDVDALAELVDRVKSYDPA...</td>\n",
       "      <td>GCA_002507085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>921</th>\n",
       "      <td>gnl|extdb|tmp_000776</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>gnl|extdb|tmp_000776tmp_000776</td>\n",
       "      <td>archaeon UBA168</td>\n",
       "      <td>298.0</td>\n",
       "      <td>VSLVGRMPDKARESMRITIPGMIKELEEQTHGVINKDDWYSVPYTG...</td>\n",
       "      <td>GCA_002507085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>922</th>\n",
       "      <td>gnl|extdb|tmp_000777</td>\n",
       "      <td>radical SAM protein</td>\n",
       "      <td>gnl|extdb|tmp_000777</td>\n",
       "      <td>archaeon UBA168</td>\n",
       "      <td>196.0</td>\n",
       "      <td>MSKLQYNGRGLENPNYIPDTLGANCPFDCGLCQRHLSHTALANVVV...</td>\n",
       "      <td>GCA_002507085</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>923 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Gene id                                Name  \\\n",
       "0    gnl|extdb|tmp_000288                hypothetical protein   \n",
       "1    gnl|extdb|tmp_000289               hypothetical protein    \n",
       "2    gnl|extdb|tmp_000290  DUF4910 domain-containing protein    \n",
       "3    gnl|extdb|tmp_000291               hypothetical protein    \n",
       "4    gnl|extdb|tmp_000292  FAD/NAD(P)-binding oxidoreductase    \n",
       "..                    ...                                 ...   \n",
       "918  gnl|extdb|tmp_000322   DUF655 domain-containing protein    \n",
       "919  gnl|extdb|tmp_000323                 NAD(+)/NADH kinase    \n",
       "920  gnl|extdb|tmp_000775                hypothetical protein   \n",
       "921  gnl|extdb|tmp_000776                hypothetical protein   \n",
       "922  gnl|extdb|tmp_000777                 radical SAM protein   \n",
       "\n",
       "                          Locus Tag    Organism Name  Protein lenght  \\\n",
       "0    gnl|extdb|tmp_000288tmp_000288  archaeon UBA168           150.0   \n",
       "1    gnl|extdb|tmp_000289tmp_000289  archaeon UBA168           261.0   \n",
       "2              gnl|extdb|tmp_000290  archaeon UBA168           482.0   \n",
       "3    gnl|extdb|tmp_000291tmp_000291  archaeon UBA168            51.0   \n",
       "4              gnl|extdb|tmp_000292  archaeon UBA168           383.0   \n",
       "..                              ...              ...             ...   \n",
       "918            gnl|extdb|tmp_000322  archaeon UBA168           200.0   \n",
       "919            gnl|extdb|tmp_000323  archaeon UBA168           269.0   \n",
       "920  gnl|extdb|tmp_000775tmp_000775  archaeon UBA168           244.0   \n",
       "921  gnl|extdb|tmp_000776tmp_000776  archaeon UBA168           298.0   \n",
       "922            gnl|extdb|tmp_000777  archaeon UBA168           196.0   \n",
       "\n",
       "                                                   AAs      Accession  \n",
       "0    MRKKVVIAGAIALVIGVVLFLVSPYMFNATLSLDSLLPFKQEQSLN...  GCA_002507085  \n",
       "1    MITFMAVSPHGDELLSPSDAESVKLNEAMNAVVRERQSVRVDTWVV...  GCA_002507085  \n",
       "2    MQAFSPYRAFDHVYNISKFHRIQGSQGLLDAANYIYDHLNVREKEM...  GCA_002507085  \n",
       "3    MIKKGVIDSACVFVANQSGAAKRLTEIGVQLHPIGERLSYYVNQGY...  GCA_002507085  \n",
       "4    MKTVVIAGGGVGGVVAAKRLSERLKGKANLVLISDNEKYVFPPYFT...  GCA_002507085  \n",
       "..                                                 ...            ...  \n",
       "918  MSERDRGRRGHIRINEEYAIVLDYLPYGNPYSNRQSPTPIIQAMGE...  GCA_002507085  \n",
       "919  MENYARWSSGVEVLYLSHRPGSRAAQEAENKVASQAESMGIEVVRL...  GCA_002507085  \n",
       "920  FRRFLRLTGLSAEEGEPVYKAYLDAGGDDVDALAELVDRVKSYDPA...  GCA_002507085  \n",
       "921  VSLVGRMPDKARESMRITIPGMIKELEEQTHGVINKDDWYSVPYTG...  GCA_002507085  \n",
       "922  MSKLQYNGRGLENPNYIPDTLGANCPFDCGLCQRHLSHTALANVVV...  GCA_002507085  \n",
       "\n",
       "[923 rows x 7 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tables_lst[2] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "for tab in tables_lst:\n",
    "    if '085' in tab.loc[0,'Accession']:\n",
    "        tab['Locus Tag'] = tab['Locus Tag'].apply(lambda x: f\"temp_{x.split('_')[-1]}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene id</th>\n",
       "      <th>Name</th>\n",
       "      <th>Locus Tag</th>\n",
       "      <th>Organism Name</th>\n",
       "      <th>Protein lenght</th>\n",
       "      <th>AAs</th>\n",
       "      <th>Accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>gnl|extdb|tmp_000288</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>temp_000288</td>\n",
       "      <td>archaeon UBA168</td>\n",
       "      <td>150.0</td>\n",
       "      <td>MRKKVVIAGAIALVIGVVLFLVSPYMFNATLSLDSLLPFKQEQSLN...</td>\n",
       "      <td>GCA_002507085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>gnl|extdb|tmp_000289</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>temp_000289</td>\n",
       "      <td>archaeon UBA168</td>\n",
       "      <td>261.0</td>\n",
       "      <td>MITFMAVSPHGDELLSPSDAESVKLNEAMNAVVRERQSVRVDTWVV...</td>\n",
       "      <td>GCA_002507085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>gnl|extdb|tmp_000290</td>\n",
       "      <td>DUF4910 domain-containing protein</td>\n",
       "      <td>temp_000290</td>\n",
       "      <td>archaeon UBA168</td>\n",
       "      <td>482.0</td>\n",
       "      <td>MQAFSPYRAFDHVYNISKFHRIQGSQGLLDAANYIYDHLNVREKEM...</td>\n",
       "      <td>GCA_002507085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>gnl|extdb|tmp_000291</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>temp_000291</td>\n",
       "      <td>archaeon UBA168</td>\n",
       "      <td>51.0</td>\n",
       "      <td>MIKKGVIDSACVFVANQSGAAKRLTEIGVQLHPIGERLSYYVNQGY...</td>\n",
       "      <td>GCA_002507085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>gnl|extdb|tmp_000292</td>\n",
       "      <td>FAD/NAD(P)-binding oxidoreductase</td>\n",
       "      <td>temp_000292</td>\n",
       "      <td>archaeon UBA168</td>\n",
       "      <td>383.0</td>\n",
       "      <td>MKTVVIAGGGVGGVVAAKRLSERLKGKANLVLISDNEKYVFPPYFT...</td>\n",
       "      <td>GCA_002507085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>918</th>\n",
       "      <td>gnl|extdb|tmp_000322</td>\n",
       "      <td>DUF655 domain-containing protein</td>\n",
       "      <td>temp_000322</td>\n",
       "      <td>archaeon UBA168</td>\n",
       "      <td>200.0</td>\n",
       "      <td>MSERDRGRRGHIRINEEYAIVLDYLPYGNPYSNRQSPTPIIQAMGE...</td>\n",
       "      <td>GCA_002507085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>919</th>\n",
       "      <td>gnl|extdb|tmp_000323</td>\n",
       "      <td>NAD(+)/NADH kinase</td>\n",
       "      <td>temp_000323</td>\n",
       "      <td>archaeon UBA168</td>\n",
       "      <td>269.0</td>\n",
       "      <td>MENYARWSSGVEVLYLSHRPGSRAAQEAENKVASQAESMGIEVVRL...</td>\n",
       "      <td>GCA_002507085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>920</th>\n",
       "      <td>gnl|extdb|tmp_000775</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>temp_000775</td>\n",
       "      <td>archaeon UBA168</td>\n",
       "      <td>244.0</td>\n",
       "      <td>FRRFLRLTGLSAEEGEPVYKAYLDAGGDDVDALAELVDRVKSYDPA...</td>\n",
       "      <td>GCA_002507085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>921</th>\n",
       "      <td>gnl|extdb|tmp_000776</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>temp_000776</td>\n",
       "      <td>archaeon UBA168</td>\n",
       "      <td>298.0</td>\n",
       "      <td>VSLVGRMPDKARESMRITIPGMIKELEEQTHGVINKDDWYSVPYTG...</td>\n",
       "      <td>GCA_002507085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>922</th>\n",
       "      <td>gnl|extdb|tmp_000777</td>\n",
       "      <td>radical SAM protein</td>\n",
       "      <td>temp_000777</td>\n",
       "      <td>archaeon UBA168</td>\n",
       "      <td>196.0</td>\n",
       "      <td>MSKLQYNGRGLENPNYIPDTLGANCPFDCGLCQRHLSHTALANVVV...</td>\n",
       "      <td>GCA_002507085</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>923 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Gene id                                Name    Locus Tag  \\\n",
       "0    gnl|extdb|tmp_000288                hypothetical protein  temp_000288   \n",
       "1    gnl|extdb|tmp_000289               hypothetical protein   temp_000289   \n",
       "2    gnl|extdb|tmp_000290  DUF4910 domain-containing protein   temp_000290   \n",
       "3    gnl|extdb|tmp_000291               hypothetical protein   temp_000291   \n",
       "4    gnl|extdb|tmp_000292  FAD/NAD(P)-binding oxidoreductase   temp_000292   \n",
       "..                    ...                                 ...          ...   \n",
       "918  gnl|extdb|tmp_000322   DUF655 domain-containing protein   temp_000322   \n",
       "919  gnl|extdb|tmp_000323                 NAD(+)/NADH kinase   temp_000323   \n",
       "920  gnl|extdb|tmp_000775                hypothetical protein  temp_000775   \n",
       "921  gnl|extdb|tmp_000776                hypothetical protein  temp_000776   \n",
       "922  gnl|extdb|tmp_000777                 radical SAM protein  temp_000777   \n",
       "\n",
       "       Organism Name  Protein lenght  \\\n",
       "0    archaeon UBA168           150.0   \n",
       "1    archaeon UBA168           261.0   \n",
       "2    archaeon UBA168           482.0   \n",
       "3    archaeon UBA168            51.0   \n",
       "4    archaeon UBA168           383.0   \n",
       "..               ...             ...   \n",
       "918  archaeon UBA168           200.0   \n",
       "919  archaeon UBA168           269.0   \n",
       "920  archaeon UBA168           244.0   \n",
       "921  archaeon UBA168           298.0   \n",
       "922  archaeon UBA168           196.0   \n",
       "\n",
       "                                                   AAs      Accession  \n",
       "0    MRKKVVIAGAIALVIGVVLFLVSPYMFNATLSLDSLLPFKQEQSLN...  GCA_002507085  \n",
       "1    MITFMAVSPHGDELLSPSDAESVKLNEAMNAVVRERQSVRVDTWVV...  GCA_002507085  \n",
       "2    MQAFSPYRAFDHVYNISKFHRIQGSQGLLDAANYIYDHLNVREKEM...  GCA_002507085  \n",
       "3    MIKKGVIDSACVFVANQSGAAKRLTEIGVQLHPIGERLSYYVNQGY...  GCA_002507085  \n",
       "4    MKTVVIAGGGVGGVVAAKRLSERLKGKANLVLISDNEKYVFPPYFT...  GCA_002507085  \n",
       "..                                                 ...            ...  \n",
       "918  MSERDRGRRGHIRINEEYAIVLDYLPYGNPYSNRQSPTPIIQAMGE...  GCA_002507085  \n",
       "919  MENYARWSSGVEVLYLSHRPGSRAAQEAENKVASQAESMGIEVVRL...  GCA_002507085  \n",
       "920  FRRFLRLTGLSAEEGEPVYKAYLDAGGDDVDALAELVDRVKSYDPA...  GCA_002507085  \n",
       "921  VSLVGRMPDKARESMRITIPGMIKELEEQTHGVINKDDWYSVPYTG...  GCA_002507085  \n",
       "922  MSKLQYNGRGLENPNYIPDTLGANCPFDCGLCQRHLSHTALANVVV...  GCA_002507085  \n",
       "\n",
       "[923 rows x 7 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tables_lst[2]"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Adding CAZyms information to tables\n",
    "First of all make a separate folder for all CAZy  data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "caz_path = '/home/aster/Project_Geoarchaeota_28.11/pgap_annot/other_arch/genomes/folder_with_cazy/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "9"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(tables_lst)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  0%|          | 0/18 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003019595.1_dbsub.out\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2359it [00:05, 396.66it/s]\n",
      "  6%|▌         | 1/18 [00:05<01:41,  5.96s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003019595.1_CAZ_overview.txt\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 11%|█         | 2/18 [00:12<01:37,  6.07s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003019545.1_CAZ_overview.txt\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 17%|█▋        | 3/18 [00:20<01:48,  7.25s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCF_902812505.1_dbsub.out\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1444it [00:01, 740.76it/s]\n",
      " 22%|██▏       | 4/18 [00:22<01:12,  5.16s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002507085.1_dbsub.out\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "923it [00:00, 1003.11it/s]\n",
      " 28%|██▊       | 5/18 [00:23<00:47,  3.63s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_011055275.1_CAZ_overview.txt\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 33%|███▎      | 6/18 [00:24<00:32,  2.73s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003019535.1_CAZ_overview.txt\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 39%|███▉      | 7/18 [00:30<00:42,  3.88s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002507085.1_CAZ_overview.txt\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 44%|████▍     | 8/18 [00:31<00:29,  2.96s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003019515.1_dbsub.out\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2487it [00:05, 434.36it/s]\n",
      " 50%|█████     | 9/18 [00:37<00:34,  3.83s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003019615.1_dbsub.out\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2129it [00:03, 666.86it/s]\n",
      " 56%|█████▌    | 10/18 [00:40<00:29,  3.64s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003019615.1_CAZ_overview.txt\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 61%|██████    | 11/18 [00:44<00:25,  3.63s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003019475.1_dbsub.out\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2405it [00:05, 432.71it/s]\n",
      " 67%|██████▋   | 12/18 [00:49<00:25,  4.22s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCF_902812505.1_CAZ_overview.txt\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 72%|███████▏  | 13/18 [00:52<00:18,  3.61s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003019545.1_dbsub.out\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2862it [00:08, 337.97it/s]\n",
      " 78%|███████▊  | 14/18 [01:00<00:20,  5.08s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003019535.1_dbsub.out\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2470it [00:06, 385.20it/s]\n",
      " 83%|████████▎ | 15/18 [01:07<00:16,  5.48s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003019515.1_CAZ_overview.txt\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 89%|████████▉ | 16/18 [01:13<00:11,  5.69s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_011055275.1_dbsub.out\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "868it [00:00, 1045.44it/s]\n",
      " 94%|█████████▍| 17/18 [01:14<00:04,  4.23s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003019475.1_CAZ_overview.txt\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 18/18 [01:19<00:00,  4.44s/it]\n"
     ]
    }
   ],
   "source": [
    "for file in tqdm(os.listdir(caz_path)):\n",
    "    fpath = os.path.join(caz_path, file)\n",
    "    for table in tables_lst:\n",
    "        #print(table.iloc[1,-1])\n",
    "        if table.loc[0, 'Accession'] in file:\n",
    "            print(file)\n",
    "            if 'dbsub' in file:\n",
    "                dbsub_tab = pd.read_csv(fpath, sep='\\t')\n",
    "\n",
    "                for idx, row in tqdm(table.iterrows()):\n",
    "                    for idx2, row2 in dbsub_tab.iterrows():\n",
    "                        if row['Gene id'] == row2['Gene ID']:\n",
    "                            table.loc[idx, 'dbCAN subfam'] = row2['dbCAN subfam']\n",
    "                            table.loc[idx, 'Subfam Composition'] = row2['Subfam Composition']\n",
    "                            table.loc[idx, 'Subfam EC'] = row2['Subfam EC']\n",
    "                            table.loc[idx, 'Substrate'] = row2['Substrate']\n",
    "                            table.loc[idx, 'Coverage'] = row2['Coverage']\n",
    "                            table.loc[idx, 'E Value'] = row2['E Value']\n",
    "\n",
    "            if 'CAZ' in file:\n",
    "                CAZ_over_tab = pd.read_csv(fpath, sep='\\t')\n",
    "                \n",
    "                for idx, row in table.iterrows():\n",
    "                    for idx2, row2 in CAZ_over_tab.iterrows():\n",
    "                        if row['Gene id'] == row2['Gene ID']:\n",
    "                            table.loc[idx, 'HMMER'] = row2['HMMER']\n",
    "                            table.loc[idx, 'dbCAN_sub'] = row2['dbCAN_sub']\n",
    "                            table.loc[idx, 'DIAMOND'] = row2['DIAMOND']\n",
    "                            table.loc[idx, '#ofTools'] = row2['#ofTools']"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Checking/Validation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>EC#</th>\n",
       "      <th>HMMER</th>\n",
       "      <th>dbCAN_sub</th>\n",
       "      <th>DIAMOND</th>\n",
       "      <th>Signalp</th>\n",
       "      <th>#ofTools</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PSN97385.1</td>\n",
       "      <td>3.2.1.-|3.2.1.21|3.2.1.23</td>\n",
       "      <td>GH1(8-464)</td>\n",
       "      <td>GH1_e66</td>\n",
       "      <td>GH1</td>\n",
       "      <td>N</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PSN97781.1</td>\n",
       "      <td>N</td>\n",
       "      <td>CE4(15-154)</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PSN97922.1</td>\n",
       "      <td>3.2.1.172</td>\n",
       "      <td>GH105(32-326)</td>\n",
       "      <td>GH105_e0</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PSN98170.1</td>\n",
       "      <td>-</td>\n",
       "      <td>GT4(188-335)</td>\n",
       "      <td>GT4_e3552</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PSN98177.1</td>\n",
       "      <td>-</td>\n",
       "      <td>GT4(220-376)</td>\n",
       "      <td>GT4_e3992</td>\n",
       "      <td>GT4</td>\n",
       "      <td>N</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene ID                        EC#          HMMER  dbCAN_sub DIAMOND  \\\n",
       "0  PSN97385.1  3.2.1.-|3.2.1.21|3.2.1.23     GH1(8-464)    GH1_e66     GH1   \n",
       "1  PSN97781.1                          N    CE4(15-154)          N       N   \n",
       "2  PSN97922.1                  3.2.1.172  GH105(32-326)   GH105_e0       N   \n",
       "3  PSN98170.1                          -   GT4(188-335)  GT4_e3552       N   \n",
       "4  PSN98177.1                          -   GT4(220-376)  GT4_e3992     GT4   \n",
       "\n",
       "  Signalp  #ofTools  \n",
       "0       N         3  \n",
       "1       N         1  \n",
       "2       N         2  \n",
       "3       N         2  \n",
       "4       N         3  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "CAZ_over_tab.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dbCAN subfam</th>\n",
       "      <th>Subfam Composition</th>\n",
       "      <th>Subfam EC</th>\n",
       "      <th>Substrate</th>\n",
       "      <th>Profile Length</th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>Gene Length</th>\n",
       "      <th>E Value</th>\n",
       "      <th>Profile Start</th>\n",
       "      <th>Profile End</th>\n",
       "      <th>Gene Start</th>\n",
       "      <th>Gene End</th>\n",
       "      <th>Coverage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GT4_e1578</td>\n",
       "      <td>GT4:125|GT2:11|GT41:5|GT0:1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>143</td>\n",
       "      <td>HEV50939.1</td>\n",
       "      <td>377</td>\n",
       "      <td>2.000000e-28</td>\n",
       "      <td>2</td>\n",
       "      <td>142</td>\n",
       "      <td>207</td>\n",
       "      <td>344</td>\n",
       "      <td>0.979021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GT2</td>\n",
       "      <td>GT2:245</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>417</td>\n",
       "      <td>HEV50980.1</td>\n",
       "      <td>437</td>\n",
       "      <td>3.000000e-50</td>\n",
       "      <td>2</td>\n",
       "      <td>353</td>\n",
       "      <td>47</td>\n",
       "      <td>401</td>\n",
       "      <td>0.841727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GH133_e2</td>\n",
       "      <td>GH133:464|GH13:1|GH23:1|GT2:1|GT4:1|GT5:1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>358</td>\n",
       "      <td>HEV51048.1</td>\n",
       "      <td>612</td>\n",
       "      <td>7.700000e-17</td>\n",
       "      <td>51</td>\n",
       "      <td>354</td>\n",
       "      <td>285</td>\n",
       "      <td>567</td>\n",
       "      <td>0.846369</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GT2</td>\n",
       "      <td>GT2:8</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>151</td>\n",
       "      <td>HEV51075.1</td>\n",
       "      <td>220</td>\n",
       "      <td>8.000000e-40</td>\n",
       "      <td>1</td>\n",
       "      <td>150</td>\n",
       "      <td>10</td>\n",
       "      <td>156</td>\n",
       "      <td>0.986755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GT4_e2990</td>\n",
       "      <td>GT4:41</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>150</td>\n",
       "      <td>HEV51076.1</td>\n",
       "      <td>168</td>\n",
       "      <td>2.400000e-31</td>\n",
       "      <td>17</td>\n",
       "      <td>145</td>\n",
       "      <td>1</td>\n",
       "      <td>130</td>\n",
       "      <td>0.853333</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  dbCAN subfam                         Subfam Composition Subfam EC Substrate  \\\n",
       "0    GT4_e1578                GT4:125|GT2:11|GT41:5|GT0:1         -         -   \n",
       "1          GT2                                    GT2:245         -         -   \n",
       "2     GH133_e2  GH133:464|GH13:1|GH23:1|GT2:1|GT4:1|GT5:1         -         -   \n",
       "3          GT2                                      GT2:8         -         -   \n",
       "4    GT4_e2990                                     GT4:41         -         -   \n",
       "\n",
       "   Profile Length     Gene ID  Gene Length       E Value  Profile Start  \\\n",
       "0             143  HEV50939.1          377  2.000000e-28              2   \n",
       "1             417  HEV50980.1          437  3.000000e-50              2   \n",
       "2             358  HEV51048.1          612  7.700000e-17             51   \n",
       "3             151  HEV51075.1          220  8.000000e-40              1   \n",
       "4             150  HEV51076.1          168  2.400000e-31             17   \n",
       "\n",
       "   Profile End  Gene Start  Gene End  Coverage  \n",
       "0          142         207       344  0.979021  \n",
       "1          353          47       401  0.841727  \n",
       "2          354         285       567  0.846369  \n",
       "3          150          10       156  0.986755  \n",
       "4          145           1       130  0.853333  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dbsub_tab.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene id</th>\n",
       "      <th>Name</th>\n",
       "      <th>Locus Tag</th>\n",
       "      <th>Organism Name</th>\n",
       "      <th>Protein lenght</th>\n",
       "      <th>AAs</th>\n",
       "      <th>Accession</th>\n",
       "      <th>dbCAN subfam</th>\n",
       "      <th>Subfam Composition</th>\n",
       "      <th>Subfam EC</th>\n",
       "      <th>Substrate</th>\n",
       "      <th>Coverage</th>\n",
       "      <th>E Value</th>\n",
       "      <th>HMMER</th>\n",
       "      <th>dbCAN_sub</th>\n",
       "      <th>DIAMOND</th>\n",
       "      <th>#ofTools</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2237</th>\n",
       "      <td>PSN96656.1</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>B9Q06_00470</td>\n",
       "      <td>Candidatus Marsarchaeota G2 archaeon ECH_B_2</td>\n",
       "      <td>385.0</td>\n",
       "      <td>MDDSLNILHVADNSYPSIGGVERVIHEVAKRQASMGHRVTVVSQLQ...</td>\n",
       "      <td>GCA_003019515</td>\n",
       "      <td>GT4_e3552</td>\n",
       "      <td>GT4:625|GH77:1|GT2:1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.95858</td>\n",
       "      <td>7.800000e-32</td>\n",
       "      <td>GT4(192-354)</td>\n",
       "      <td>GT4_e3552</td>\n",
       "      <td>N</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene id                   Name    Locus Tag  \\\n",
       "2237  PSN96656.1  hypothetical protein   B9Q06_00470   \n",
       "\n",
       "                                     Organism Name  Protein lenght  \\\n",
       "2237  Candidatus Marsarchaeota G2 archaeon ECH_B_2           385.0   \n",
       "\n",
       "                                                    AAs      Accession  \\\n",
       "2237  MDDSLNILHVADNSYPSIGGVERVIHEVAKRQASMGHRVTVVSQLQ...  GCA_003019515   \n",
       "\n",
       "     dbCAN subfam    Subfam Composition Subfam EC Substrate  Coverage  \\\n",
       "2237    GT4_e3552  GT4:625|GH77:1|GT2:1         -         -   0.95858   \n",
       "\n",
       "           E Value         HMMER  dbCAN_sub DIAMOND  #ofTools  \n",
       "2237  7.800000e-32  GT4(192-354)  GT4_e3552       N       2.0  "
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "tables_lst[0][tables_lst[0]['Gene id'] == 'PSN96656.1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene id</th>\n",
       "      <th>Name</th>\n",
       "      <th>Locus Tag</th>\n",
       "      <th>Organism Name</th>\n",
       "      <th>Protein lenght</th>\n",
       "      <th>AAs</th>\n",
       "      <th>Accession</th>\n",
       "      <th>dbCAN subfam</th>\n",
       "      <th>Subfam Composition</th>\n",
       "      <th>Subfam EC</th>\n",
       "      <th>Substrate</th>\n",
       "      <th>Coverage</th>\n",
       "      <th>E Value</th>\n",
       "      <th>HMMER</th>\n",
       "      <th>dbCAN_sub</th>\n",
       "      <th>DIAMOND</th>\n",
       "      <th>#ofTools</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>WP_174569547.1</td>\n",
       "      <td>glycosyltransferase family 39 protein</td>\n",
       "      <td>WP_174569547.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>411.0</td>\n",
       "      <td>MASDNSLNNTSKPPVSKSFNGENVVLISIIFLGVVLRASWMLNPST...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>GT39_e28</td>\n",
       "      <td>GT39:91</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.694323</td>\n",
       "      <td>8.200000e-26</td>\n",
       "      <td>GT39(30-244)</td>\n",
       "      <td>GT39_e28</td>\n",
       "      <td>N</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>WP_174569559.1</td>\n",
       "      <td>glycosyltransferase family 4 protein</td>\n",
       "      <td>WP_174569559.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>385.0</td>\n",
       "      <td>MDDSLNILHVADNSYPSIGGVERVIHEVAKRQASMGHRVTVVSQLQ...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>GT4_e3552</td>\n",
       "      <td>GT4:625|GH77:1|GT2:1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.958580</td>\n",
       "      <td>1.200000e-31</td>\n",
       "      <td>GT4(192-354)</td>\n",
       "      <td>GT4_e3552</td>\n",
       "      <td>N</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>WP_174569660.1</td>\n",
       "      <td>polyprenol monophosphomannose synthase</td>\n",
       "      <td>WP_174569660.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>245.0</td>\n",
       "      <td>MRITSDNPSTKMIGYNEPLVSIVIPTYNEKENVVPIIHNLHSVMQK...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>GT2</td>\n",
       "      <td>GT2:32</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.987730</td>\n",
       "      <td>1.300000e-58</td>\n",
       "      <td>GT2(21-185)</td>\n",
       "      <td>GT2</td>\n",
       "      <td>N</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>243</th>\n",
       "      <td>WP_174569774.1</td>\n",
       "      <td>glycosyltransferase</td>\n",
       "      <td>WP_174569774.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>391.0</td>\n",
       "      <td>MTPLTLGYYALILLGIVDFLAWLILLYRFEAGFRRAPKLPSEYIGQ...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>GT2</td>\n",
       "      <td>GT2:34</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.942238</td>\n",
       "      <td>1.000000e-47</td>\n",
       "      <td>GT2(53-219)</td>\n",
       "      <td>GT2</td>\n",
       "      <td>N</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316</th>\n",
       "      <td>WP_174569847.1</td>\n",
       "      <td>FAD-binding protein</td>\n",
       "      <td>WP_174569847.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>440.0</td>\n",
       "      <td>MTLEEEFSQSGLEVVSNTEKYREDFTHYPVEPRIVVKAKSTEEVVS...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>AA7_e0</td>\n",
       "      <td>AA7:17</td>\n",
       "      <td>1.1.3.-:1</td>\n",
       "      <td>chitooligosaccharide</td>\n",
       "      <td>0.479616</td>\n",
       "      <td>4.100000e-18</td>\n",
       "      <td>AA7(29-208)</td>\n",
       "      <td>AA7_e0</td>\n",
       "      <td>N</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342</th>\n",
       "      <td>WP_174569873.1</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>WP_174569873.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>406.0</td>\n",
       "      <td>MVRVVVGFMRSPGVRPTVVIGACGIGLGHVGRMLPVARRLAGRGYS...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>GT1_e519</td>\n",
       "      <td>GT1:7</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.851282</td>\n",
       "      <td>4.900000e-22</td>\n",
       "      <td>GT1(227-404)</td>\n",
       "      <td>GT1_e519</td>\n",
       "      <td>N</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373</th>\n",
       "      <td>WP_174569904.1</td>\n",
       "      <td>glycosyltransferase</td>\n",
       "      <td>WP_174569904.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>322.0</td>\n",
       "      <td>MVSRERVIHSSNAGGEKRSVLVGIPVYNSEKTITSCLEKVLNQAQR...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>GT2</td>\n",
       "      <td>GT2:6</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.964286</td>\n",
       "      <td>3.000000e-27</td>\n",
       "      <td>GT2(22-157)</td>\n",
       "      <td>GT2</td>\n",
       "      <td>N</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>904</th>\n",
       "      <td>WP_174570445.1</td>\n",
       "      <td>polysaccharide deacetylase family protein</td>\n",
       "      <td>WP_174570445.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>304.0</td>\n",
       "      <td>MVFSVRDFVGYADRSLEFTWPGGRRLALSIVVNFEEGSEHSVVTDG...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>CE4_e210</td>\n",
       "      <td>CE4:182</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.990741</td>\n",
       "      <td>6.200000e-29</td>\n",
       "      <td>CE4(72-181)</td>\n",
       "      <td>CE4_e210</td>\n",
       "      <td>CE4</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>931</th>\n",
       "      <td>WP_174570472.1</td>\n",
       "      <td>glycosyltransferase</td>\n",
       "      <td>WP_174570472.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>299.0</td>\n",
       "      <td>MRDNGYSPQVSVLIPTYKEAGSIRSTLETVSRELEATGVLSYEIIV...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>GT2</td>\n",
       "      <td>GT2:41</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.976331</td>\n",
       "      <td>2.400000e-30</td>\n",
       "      <td>GT2(11-173)</td>\n",
       "      <td>GT2</td>\n",
       "      <td>N</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>964</th>\n",
       "      <td>WP_174570505.1</td>\n",
       "      <td>malto-oligosyltrehalose synthase</td>\n",
       "      <td>WP_174570505.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>788.0</td>\n",
       "      <td>MRVLSTYRLQLNREFPFASVIGLLEYFSELGVTHLYLSPILKARPA...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>GH13_e50</td>\n",
       "      <td>GH13_26:22</td>\n",
       "      <td>-</td>\n",
       "      <td>sucrose</td>\n",
       "      <td>0.992453</td>\n",
       "      <td>4.000000e-100</td>\n",
       "      <td>GH13_26(24-310)</td>\n",
       "      <td>GH13_e50</td>\n",
       "      <td>GH13_26</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>965</th>\n",
       "      <td>WP_174570506.1</td>\n",
       "      <td>glycogen debranching protein GlgX</td>\n",
       "      <td>WP_174570506.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>716.0</td>\n",
       "      <td>MLRYRDRPLRPGYSYPLGSTWFEDDDGVNFVLFSENASSVDLLLYH...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>GH13_e48</td>\n",
       "      <td>GH13_11:3001|CBM48:2819|GH13:35|GH77:23|GH13_10:3</td>\n",
       "      <td>3.2.1.68:15|3.2.1.-:1</td>\n",
       "      <td>sucrose, starch, starch</td>\n",
       "      <td>0.986034</td>\n",
       "      <td>1.300000e-172</td>\n",
       "      <td>CBM48(16-110)+GH13_11(192-545)</td>\n",
       "      <td>CBM48_e3+GH13_e48</td>\n",
       "      <td>CBM48+GH13_11</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>967</th>\n",
       "      <td>WP_174570508.1</td>\n",
       "      <td>malto-oligosyltrehalose trehalohydrolase</td>\n",
       "      <td>WP_174570508.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>574.0</td>\n",
       "      <td>MCANTTHDNGVRFSVWAPNHRSVSLVIYGNHGRRVLPMVRGKRGYF...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>GH13_e36</td>\n",
       "      <td>GH13_10:23|CBM48:16</td>\n",
       "      <td>3.2.1.141:3|3.2.1.1:1</td>\n",
       "      <td>sucrose, starch, starch</td>\n",
       "      <td>0.993528</td>\n",
       "      <td>2.600000e-154</td>\n",
       "      <td>GH13_10(116-430)</td>\n",
       "      <td>CBM48_e0+GH13_e36</td>\n",
       "      <td>CBM48+GH13_10</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>968</th>\n",
       "      <td>WP_174570509.1</td>\n",
       "      <td>DUF3416 domain-containing protein</td>\n",
       "      <td>WP_174570509.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>644.0</td>\n",
       "      <td>MIGREGDIVIQDVHPMIDGGKYPVKRIVGDVLKVDAEVFTTGEKAV...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>GH13_e41</td>\n",
       "      <td>GH13_3:1253|GH0:92|GH13:34|GT4:1</td>\n",
       "      <td>-</td>\n",
       "      <td>sucrose</td>\n",
       "      <td>0.986175</td>\n",
       "      <td>3.800000e-92</td>\n",
       "      <td>GH13_3(219-439)</td>\n",
       "      <td>GH13_e41</td>\n",
       "      <td>GH13_3</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>970</th>\n",
       "      <td>WP_174570511.1</td>\n",
       "      <td>1</td>\n",
       "      <td>WP_174570511.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>767.0</td>\n",
       "      <td>MTDVDVTRVQVDGLNGSDRGISENDVLSIVSFDNDRPDLVLGPHLE...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>GH13_e200</td>\n",
       "      <td>GH13_9:2874|CBM48:2782|GH77:10|GH13:7|GH13_3:5...</td>\n",
       "      <td>2.4.1.18:14</td>\n",
       "      <td>sucrose, alpha-glucan</td>\n",
       "      <td>0.993355</td>\n",
       "      <td>2.000000e-163</td>\n",
       "      <td>CBM48(140-223)+GH13_9(294-593)</td>\n",
       "      <td>CBM48_e2+GH13_e200</td>\n",
       "      <td>CBM48+GH13_9</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>971</th>\n",
       "      <td>WP_174570512.1</td>\n",
       "      <td>maltose alpha-D-glucosyltransferase</td>\n",
       "      <td>WP_174570512.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>551.0</td>\n",
       "      <td>MDSTNRLWYRDAIFYEIPVKSFYDSNNDGVGDFKGLTQKLDYLAWL...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>GH13_e0</td>\n",
       "      <td>GH13_16:815|GH13:8|GH13_19:1</td>\n",
       "      <td>5.4.99.16:7</td>\n",
       "      <td>sucrose, starch</td>\n",
       "      <td>0.997175</td>\n",
       "      <td>8.500000e-185</td>\n",
       "      <td>GH13_16(31-383)</td>\n",
       "      <td>GH13_e0</td>\n",
       "      <td>GH13_16</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>WP_174570692.1</td>\n",
       "      <td>glycosyltransferase family 4 protein</td>\n",
       "      <td>WP_174570692.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>456.0</td>\n",
       "      <td>MPIRICINSQTPFLRFKLSYSELVEKYGILDDPLDIRMLEEGVDYD...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>GT4_e3135</td>\n",
       "      <td>GT4:23</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.937500</td>\n",
       "      <td>2.300000e-51</td>\n",
       "      <td>GT4(253-412)</td>\n",
       "      <td>GT4_e3135</td>\n",
       "      <td>GT4</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1170</th>\n",
       "      <td>WP_174570772.1</td>\n",
       "      <td>glycosyltransferase family 39 protein</td>\n",
       "      <td>WP_174570772.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>694.0</td>\n",
       "      <td>MQPPRIKLSLGFWNYTLTALSLVFVFVIAVIARLGPLKYAFQIAEV...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>GT66_e26</td>\n",
       "      <td>GT66:41</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.987296</td>\n",
       "      <td>4.800000e-89</td>\n",
       "      <td>GT66(19-553)</td>\n",
       "      <td>GT66_e26</td>\n",
       "      <td>N</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1292</th>\n",
       "      <td>WP_174570894.1</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>WP_174570894.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>382.0</td>\n",
       "      <td>MKLLVGVSSVGLGHVKRSLSIVREIVKLRPNTSVGWLCAQPALSYL...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>GT1_e269</td>\n",
       "      <td>GT1:13</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.465565</td>\n",
       "      <td>2.700000e-17</td>\n",
       "      <td>GT28(251-359)</td>\n",
       "      <td>GT1_e269</td>\n",
       "      <td>N</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1303</th>\n",
       "      <td>WP_174570905.1</td>\n",
       "      <td>glycosyltransferase family 4 protein</td>\n",
       "      <td>WP_174570905.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>371.0</td>\n",
       "      <td>MALKVAMFGWEYPPHVVGGLGVHSAELTRRLVSYGVEVDFYKPKID...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>GT4_e2970</td>\n",
       "      <td>GT4:7</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.971831</td>\n",
       "      <td>2.300000e-37</td>\n",
       "      <td>GT4(194-338)</td>\n",
       "      <td>GT4_e2970</td>\n",
       "      <td>N</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1304</th>\n",
       "      <td>WP_174570906.1</td>\n",
       "      <td>alpha-amylase</td>\n",
       "      <td>WP_174570906.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>389.0</td>\n",
       "      <td>MAKHFLLCFEAHQPHRLGRIRFWDGEVSFWESYNNEVLGRVSSKCY...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>GH57_e37</td>\n",
       "      <td>GH57:64</td>\n",
       "      <td>3.2.1.1:1</td>\n",
       "      <td>starch</td>\n",
       "      <td>0.993750</td>\n",
       "      <td>5.300000e-108</td>\n",
       "      <td>GH57(7-308)</td>\n",
       "      <td>GH57_e37</td>\n",
       "      <td>N</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1380</th>\n",
       "      <td>WP_174570982.1</td>\n",
       "      <td>LysM peptidoglycan-binding domain-containing p...</td>\n",
       "      <td>WP_174570982.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>155.0</td>\n",
       "      <td>MGDRLSNLLDGAGNFVGGVIAGGVTWANQANQSHSIQQHRSQQNNS...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>CBM50_e752</td>\n",
       "      <td>CBM50:5</td>\n",
       "      <td>-</td>\n",
       "      <td>chitin, peptidoglycan</td>\n",
       "      <td>0.979592</td>\n",
       "      <td>9.700000e-23</td>\n",
       "      <td>CBM50(105-154)</td>\n",
       "      <td>CBM50_e752</td>\n",
       "      <td>N</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1382</th>\n",
       "      <td>WP_174570984.1</td>\n",
       "      <td>LysM peptidoglycan-binding domain-containing p...</td>\n",
       "      <td>WP_174570984.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>54.0</td>\n",
       "      <td>MLYYTVVSGDTLWGVAARFYGTGLMWQRIYRANIGVIGSNPNLIYV...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>CBM50_e752</td>\n",
       "      <td>CBM50:5</td>\n",
       "      <td>-</td>\n",
       "      <td>chitin, peptidoglycan</td>\n",
       "      <td>0.979592</td>\n",
       "      <td>2.000000e-21</td>\n",
       "      <td>CBM50(4-53)</td>\n",
       "      <td>CBM50_e752</td>\n",
       "      <td>N</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1401</th>\n",
       "      <td>WP_174571003.1</td>\n",
       "      <td>glycosyltransferase family 2 protein</td>\n",
       "      <td>WP_174571003.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>262.0</td>\n",
       "      <td>MFSIQIPTLHGEYLREVLESIRGQTFQDYEVVVVNSGGGFISDLIH...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>GT2</td>\n",
       "      <td>GT2:16</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.985507</td>\n",
       "      <td>2.800000e-37</td>\n",
       "      <td>N</td>\n",
       "      <td>GT2</td>\n",
       "      <td>GT2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1413</th>\n",
       "      <td>WP_174571015.1</td>\n",
       "      <td>LysM peptidoglycan-binding domain-containing p...</td>\n",
       "      <td>WP_174571015.1</td>\n",
       "      <td>archaeon</td>\n",
       "      <td>350.0</td>\n",
       "      <td>QAGAATVAGYMGYGPFGFVQYHAGQTGTNKGYTGQYNDPLSGLDYY...</td>\n",
       "      <td>GCF_902812505</td>\n",
       "      <td>CBM50_e752</td>\n",
       "      <td>CBM50:5</td>\n",
       "      <td>-</td>\n",
       "      <td>chitin, peptidoglycan</td>\n",
       "      <td>0.959184</td>\n",
       "      <td>3.200000e-21</td>\n",
       "      <td>N</td>\n",
       "      <td>CBM50_e752</td>\n",
       "      <td>N</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Gene id                                               Name  \\\n",
       "16    WP_174569547.1             glycosyltransferase family 39 protein    \n",
       "28    WP_174569559.1              glycosyltransferase family 4 protein    \n",
       "129   WP_174569660.1            polyprenol monophosphomannose synthase    \n",
       "243   WP_174569774.1                               glycosyltransferase    \n",
       "316   WP_174569847.1                               FAD-binding protein    \n",
       "342   WP_174569873.1                              hypothetical protein    \n",
       "373   WP_174569904.1                               glycosyltransferase    \n",
       "904   WP_174570445.1         polysaccharide deacetylase family protein    \n",
       "931   WP_174570472.1                               glycosyltransferase    \n",
       "964   WP_174570505.1                  malto-oligosyltrehalose synthase    \n",
       "965   WP_174570506.1                 glycogen debranching protein GlgX    \n",
       "967   WP_174570508.1          malto-oligosyltrehalose trehalohydrolase    \n",
       "968   WP_174570509.1                 DUF3416 domain-containing protein    \n",
       "970   WP_174570511.1                                                  1   \n",
       "971   WP_174570512.1               maltose alpha-D-glucosyltransferase    \n",
       "1097  WP_174570692.1              glycosyltransferase family 4 protein    \n",
       "1170  WP_174570772.1             glycosyltransferase family 39 protein    \n",
       "1292  WP_174570894.1                              hypothetical protein    \n",
       "1303  WP_174570905.1              glycosyltransferase family 4 protein    \n",
       "1304  WP_174570906.1                                     alpha-amylase    \n",
       "1380  WP_174570982.1  LysM peptidoglycan-binding domain-containing p...   \n",
       "1382  WP_174570984.1  LysM peptidoglycan-binding domain-containing p...   \n",
       "1401  WP_174571003.1              glycosyltransferase family 2 protein    \n",
       "1413  WP_174571015.1  LysM peptidoglycan-binding domain-containing p...   \n",
       "\n",
       "           Locus Tag Organism Name  Protein lenght  \\\n",
       "16    WP_174569547.1      archaeon           411.0   \n",
       "28    WP_174569559.1      archaeon           385.0   \n",
       "129   WP_174569660.1      archaeon           245.0   \n",
       "243   WP_174569774.1      archaeon           391.0   \n",
       "316   WP_174569847.1      archaeon           440.0   \n",
       "342   WP_174569873.1      archaeon           406.0   \n",
       "373   WP_174569904.1      archaeon           322.0   \n",
       "904   WP_174570445.1      archaeon           304.0   \n",
       "931   WP_174570472.1      archaeon           299.0   \n",
       "964   WP_174570505.1      archaeon           788.0   \n",
       "965   WP_174570506.1      archaeon           716.0   \n",
       "967   WP_174570508.1      archaeon           574.0   \n",
       "968   WP_174570509.1      archaeon           644.0   \n",
       "970   WP_174570511.1      archaeon           767.0   \n",
       "971   WP_174570512.1      archaeon           551.0   \n",
       "1097  WP_174570692.1      archaeon           456.0   \n",
       "1170  WP_174570772.1      archaeon           694.0   \n",
       "1292  WP_174570894.1      archaeon           382.0   \n",
       "1303  WP_174570905.1      archaeon           371.0   \n",
       "1304  WP_174570906.1      archaeon           389.0   \n",
       "1380  WP_174570982.1      archaeon           155.0   \n",
       "1382  WP_174570984.1      archaeon            54.0   \n",
       "1401  WP_174571003.1      archaeon           262.0   \n",
       "1413  WP_174571015.1      archaeon           350.0   \n",
       "\n",
       "                                                    AAs      Accession  \\\n",
       "16    MASDNSLNNTSKPPVSKSFNGENVVLISIIFLGVVLRASWMLNPST...  GCF_902812505   \n",
       "28    MDDSLNILHVADNSYPSIGGVERVIHEVAKRQASMGHRVTVVSQLQ...  GCF_902812505   \n",
       "129   MRITSDNPSTKMIGYNEPLVSIVIPTYNEKENVVPIIHNLHSVMQK...  GCF_902812505   \n",
       "243   MTPLTLGYYALILLGIVDFLAWLILLYRFEAGFRRAPKLPSEYIGQ...  GCF_902812505   \n",
       "316   MTLEEEFSQSGLEVVSNTEKYREDFTHYPVEPRIVVKAKSTEEVVS...  GCF_902812505   \n",
       "342   MVRVVVGFMRSPGVRPTVVIGACGIGLGHVGRMLPVARRLAGRGYS...  GCF_902812505   \n",
       "373   MVSRERVIHSSNAGGEKRSVLVGIPVYNSEKTITSCLEKVLNQAQR...  GCF_902812505   \n",
       "904   MVFSVRDFVGYADRSLEFTWPGGRRLALSIVVNFEEGSEHSVVTDG...  GCF_902812505   \n",
       "931   MRDNGYSPQVSVLIPTYKEAGSIRSTLETVSRELEATGVLSYEIIV...  GCF_902812505   \n",
       "964   MRVLSTYRLQLNREFPFASVIGLLEYFSELGVTHLYLSPILKARPA...  GCF_902812505   \n",
       "965   MLRYRDRPLRPGYSYPLGSTWFEDDDGVNFVLFSENASSVDLLLYH...  GCF_902812505   \n",
       "967   MCANTTHDNGVRFSVWAPNHRSVSLVIYGNHGRRVLPMVRGKRGYF...  GCF_902812505   \n",
       "968   MIGREGDIVIQDVHPMIDGGKYPVKRIVGDVLKVDAEVFTTGEKAV...  GCF_902812505   \n",
       "970   MTDVDVTRVQVDGLNGSDRGISENDVLSIVSFDNDRPDLVLGPHLE...  GCF_902812505   \n",
       "971   MDSTNRLWYRDAIFYEIPVKSFYDSNNDGVGDFKGLTQKLDYLAWL...  GCF_902812505   \n",
       "1097  MPIRICINSQTPFLRFKLSYSELVEKYGILDDPLDIRMLEEGVDYD...  GCF_902812505   \n",
       "1170  MQPPRIKLSLGFWNYTLTALSLVFVFVIAVIARLGPLKYAFQIAEV...  GCF_902812505   \n",
       "1292  MKLLVGVSSVGLGHVKRSLSIVREIVKLRPNTSVGWLCAQPALSYL...  GCF_902812505   \n",
       "1303  MALKVAMFGWEYPPHVVGGLGVHSAELTRRLVSYGVEVDFYKPKID...  GCF_902812505   \n",
       "1304  MAKHFLLCFEAHQPHRLGRIRFWDGEVSFWESYNNEVLGRVSSKCY...  GCF_902812505   \n",
       "1380  MGDRLSNLLDGAGNFVGGVIAGGVTWANQANQSHSIQQHRSQQNNS...  GCF_902812505   \n",
       "1382  MLYYTVVSGDTLWGVAARFYGTGLMWQRIYRANIGVIGSNPNLIYV...  GCF_902812505   \n",
       "1401  MFSIQIPTLHGEYLREVLESIRGQTFQDYEVVVVNSGGGFISDLIH...  GCF_902812505   \n",
       "1413  QAGAATVAGYMGYGPFGFVQYHAGQTGTNKGYTGQYNDPLSGLDYY...  GCF_902812505   \n",
       "\n",
       "     dbCAN subfam                                 Subfam Composition  \\\n",
       "16       GT39_e28                                            GT39:91   \n",
       "28      GT4_e3552                               GT4:625|GH77:1|GT2:1   \n",
       "129           GT2                                             GT2:32   \n",
       "243           GT2                                             GT2:34   \n",
       "316        AA7_e0                                             AA7:17   \n",
       "342      GT1_e519                                              GT1:7   \n",
       "373           GT2                                              GT2:6   \n",
       "904      CE4_e210                                            CE4:182   \n",
       "931           GT2                                             GT2:41   \n",
       "964      GH13_e50                                         GH13_26:22   \n",
       "965      GH13_e48  GH13_11:3001|CBM48:2819|GH13:35|GH77:23|GH13_10:3   \n",
       "967      GH13_e36                                GH13_10:23|CBM48:16   \n",
       "968      GH13_e41                   GH13_3:1253|GH0:92|GH13:34|GT4:1   \n",
       "970     GH13_e200  GH13_9:2874|CBM48:2782|GH77:10|GH13:7|GH13_3:5...   \n",
       "971       GH13_e0                       GH13_16:815|GH13:8|GH13_19:1   \n",
       "1097    GT4_e3135                                             GT4:23   \n",
       "1170     GT66_e26                                            GT66:41   \n",
       "1292     GT1_e269                                             GT1:13   \n",
       "1303    GT4_e2970                                              GT4:7   \n",
       "1304     GH57_e37                                            GH57:64   \n",
       "1380   CBM50_e752                                            CBM50:5   \n",
       "1382   CBM50_e752                                            CBM50:5   \n",
       "1401          GT2                                             GT2:16   \n",
       "1413   CBM50_e752                                            CBM50:5   \n",
       "\n",
       "                  Subfam EC                Substrate  Coverage        E Value  \\\n",
       "16                        -                        -  0.694323   8.200000e-26   \n",
       "28                        -                        -  0.958580   1.200000e-31   \n",
       "129                       -                        -  0.987730   1.300000e-58   \n",
       "243                       -                        -  0.942238   1.000000e-47   \n",
       "316               1.1.3.-:1     chitooligosaccharide  0.479616   4.100000e-18   \n",
       "342                       -                        -  0.851282   4.900000e-22   \n",
       "373                       -                        -  0.964286   3.000000e-27   \n",
       "904                       -                        -  0.990741   6.200000e-29   \n",
       "931                       -                        -  0.976331   2.400000e-30   \n",
       "964                       -                  sucrose  0.992453  4.000000e-100   \n",
       "965   3.2.1.68:15|3.2.1.-:1  sucrose, starch, starch  0.986034  1.300000e-172   \n",
       "967   3.2.1.141:3|3.2.1.1:1  sucrose, starch, starch  0.993528  2.600000e-154   \n",
       "968                       -                  sucrose  0.986175   3.800000e-92   \n",
       "970             2.4.1.18:14    sucrose, alpha-glucan  0.993355  2.000000e-163   \n",
       "971             5.4.99.16:7          sucrose, starch  0.997175  8.500000e-185   \n",
       "1097                      -                        -  0.937500   2.300000e-51   \n",
       "1170                      -                        -  0.987296   4.800000e-89   \n",
       "1292                      -                        -  0.465565   2.700000e-17   \n",
       "1303                      -                        -  0.971831   2.300000e-37   \n",
       "1304              3.2.1.1:1                   starch  0.993750  5.300000e-108   \n",
       "1380                      -    chitin, peptidoglycan  0.979592   9.700000e-23   \n",
       "1382                      -    chitin, peptidoglycan  0.979592   2.000000e-21   \n",
       "1401                      -                        -  0.985507   2.800000e-37   \n",
       "1413                      -    chitin, peptidoglycan  0.959184   3.200000e-21   \n",
       "\n",
       "                               HMMER           dbCAN_sub        DIAMOND  \\\n",
       "16                      GT39(30-244)            GT39_e28              N   \n",
       "28                      GT4(192-354)           GT4_e3552              N   \n",
       "129                      GT2(21-185)                 GT2              N   \n",
       "243                      GT2(53-219)                 GT2              N   \n",
       "316                      AA7(29-208)              AA7_e0              N   \n",
       "342                     GT1(227-404)            GT1_e519              N   \n",
       "373                      GT2(22-157)                 GT2              N   \n",
       "904                      CE4(72-181)            CE4_e210            CE4   \n",
       "931                      GT2(11-173)                 GT2              N   \n",
       "964                  GH13_26(24-310)            GH13_e50        GH13_26   \n",
       "965   CBM48(16-110)+GH13_11(192-545)   CBM48_e3+GH13_e48  CBM48+GH13_11   \n",
       "967                 GH13_10(116-430)   CBM48_e0+GH13_e36  CBM48+GH13_10   \n",
       "968                  GH13_3(219-439)            GH13_e41         GH13_3   \n",
       "970   CBM48(140-223)+GH13_9(294-593)  CBM48_e2+GH13_e200   CBM48+GH13_9   \n",
       "971                  GH13_16(31-383)             GH13_e0        GH13_16   \n",
       "1097                    GT4(253-412)           GT4_e3135            GT4   \n",
       "1170                    GT66(19-553)            GT66_e26              N   \n",
       "1292                   GT28(251-359)            GT1_e269              N   \n",
       "1303                    GT4(194-338)           GT4_e2970              N   \n",
       "1304                     GH57(7-308)            GH57_e37              N   \n",
       "1380                  CBM50(105-154)          CBM50_e752              N   \n",
       "1382                     CBM50(4-53)          CBM50_e752              N   \n",
       "1401                               N                 GT2            GT2   \n",
       "1413                               N          CBM50_e752              N   \n",
       "\n",
       "      #ofTools  \n",
       "16         2.0  \n",
       "28         2.0  \n",
       "129        2.0  \n",
       "243        2.0  \n",
       "316        2.0  \n",
       "342        2.0  \n",
       "373        2.0  \n",
       "904        3.0  \n",
       "931        2.0  \n",
       "964        3.0  \n",
       "965        3.0  \n",
       "967        3.0  \n",
       "968        3.0  \n",
       "970        3.0  \n",
       "971        3.0  \n",
       "1097       3.0  \n",
       "1170       2.0  \n",
       "1292       2.0  \n",
       "1303       2.0  \n",
       "1304       2.0  \n",
       "1380       2.0  \n",
       "1382       2.0  \n",
       "1401       2.0  \n",
       "1413       1.0  "
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tables_lst[-1].dropna(subset='dbCAN subfam')"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## save_tabs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "for i in tables_lst:\n",
    "    i.to_csv(f\"/home/aster/Project_Geoarchaeota_28.11/pgap_annot/other_arch/tablets/{i.loc[0,'Accession']}.csv\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## making separate CAZY_only"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "for table in tables_lst:\n",
    "    CAZ_only = table.dropna(subset='dbCAN subfam')\n",
    "    CAZ_only.to_csv(f\"/home/aster/Project_Geoarchaeota_28.11/pgap_annot/other_arch/tablets/CAZ_only/{table.loc[0,'Accession']}_CAZ_only.csv\", index=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gather a statistic data. For CAZy fam."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "def caz_gather(table:pd.DataFrame, conc=None):\n",
    "    CAZ_lst = [i.split('_')[0] for i in table['dbCAN subfam'].dropna().to_list()]\n",
    "    #CAZ_lst = [i.split('_')[0] for i in table['dbCAN_sub'].dropna().to_list()]\n",
    "\n",
    "    GH = 0\n",
    "    PL = 0\n",
    "    CE = 0\n",
    "    GT = 0\n",
    "    AA = 0\n",
    "    CBM = 0\n",
    "\n",
    "    for i in CAZ_lst:\n",
    "        if 'GH' in i:\n",
    "            GH += 1\n",
    "        elif 'PL' in i:\n",
    "            PL += 1\n",
    "        elif 'CE' in i:\n",
    "            CE += 1\n",
    "        elif 'GT' in i:\n",
    "            GT += 1\n",
    "        elif 'AA' in i:\n",
    "            AA += 1\n",
    "        elif 'CBM' in i:\n",
    "            CBM += 1\n",
    "    \n",
    "    numbers = [('GH', GH), ('PL', PL), ('CE', CE), ('GT', GT), ('AA', AA), ('CBM', CBM)]\n",
    "\n",
    "    agregate_table = pd.DataFrame({'GH':GH, 'PL':PL, 'CE':CE, 'GT':GT, 'AA':AA, 'CBM':CBM}, index=[0])\n",
    "\n",
    "    test_lst = []\n",
    "    for i in CAZ_lst:\n",
    "        test_lst.append(CAZ_lst.count(i))\n",
    "\n",
    "    all_table = pd.DataFrame({\"CAZ\":CAZ_lst, \"Num\":test_lst}).drop_duplicates(subset='CAZ')\n",
    "\n",
    "    if conc is None:\n",
    "        \n",
    "        return agregate_table, all_table\n",
    "    \n",
    "    else:\n",
    "\n",
    "        agregate_table_t = agregate_table.T\n",
    "        agregate_table_t['CAZ'] = agregate_table_t.index\n",
    "        agregate_table_t = agregate_table_t[['CAZ', 0]]\n",
    "        agregate_table_t = agregate_table_t.rename({0:\"Num\"}, axis=1)\n",
    "        \n",
    "        df_conc = pd.concat([all_table, agregate_table_t], ignore_index=True)\n",
    "        return df_conc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "#for i in tables_lst:\n",
    "#    agregate, all_tab = caz_gather(i)\n",
    "#    agregate.to_csv(f\"/home/aster/Project_Geoarchaeota_28.11/pgap_annot/other_arch/CAZ_stat/{i.loc[0, 'Accession']}_fam_summary.csv\", index=False)\n",
    "#    all_tab.to_csv(f\"/home/aster/Project_Geoarchaeota_28.11/pgap_annot/other_arch/CAZ_stat/{i.loc[0, 'Accession']}_fam_count.csv\", index=False)\n",
    "\n",
    "for i in tables_lst:\n",
    "    concat_tab = caz_gather(i, conc=1)\n",
    "    concat_tab.to_csv(f\"/home/aster/Project_Geoarchaeota_28.11/pgap_annot/other_arch/CAZ_stat/{i.loc[0, 'Accession']}_CAZ_summary.csv\", index=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Add summary to Geo3918 Geo3817"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "CAZ_3918 = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/pgap_annot/geo/Geo3918_МП_table.csv')\n",
    "CAZ_3817 = pd.read_csv('/home/aster/Project_Geoarchaeota_28.11/pgap_annot/geo/geo3817_AK_full_table.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['IDs', 'Description', 'pgap_loc_tag', 'pgap_descr', 'Type', 'TM Type',\n",
       "       'Lenght_AA', 'AAs', 'TM_composition', 'dbCAN subfam',\n",
       "       'Subfam Composition', 'Substrate', 'E Value', 'Coverage', 'Subfam EC',\n",
       "       'HMMER', 'DIAMOND'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "CAZ_3817.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i,j in [(CAZ_3918, 'Geo_3918_МП'),(CAZ_3817, 'Geo_3817_AK')]:\n",
    "    concat_tab = caz_gather(i, conc=1)\n",
    "    concat_tab.to_csv(f\"/home/aster/Project_Geoarchaeota_28.11/pgap_annot/other_arch/CAZ_stat/{j}_CAZ_summary.csv\", index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "bioinf",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "5c7c0e940973d1f7f557ece26bca6704213ece01202fcdffb2249dc40fb254cf"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
